{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "CIF_DIR = '/home/mikhail/bioinformatics/data/NR_3.0/cif'\n",
    "JSON_DIR = '/home/mikhail/bioinformatics/data/NR_3.0/json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 0/1074 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bsb.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 681\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 1/1074 [00:00<09:23,  1.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1ytu.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 837\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 2/1074 [00:01<10:34,  1.69it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k4s.cif1'\n",
      "    total number of nucleotides: 67\n",
      "    total number of amino acids: 924\n",
      "    total number of base pairs: 30\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 64\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 14\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 3/1074 [00:02<12:05,  1.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4o26.cif1'\n",
      "    total number of nucleotides: 94\n",
      "    total number of amino acids: 468\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.544.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.457.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.544.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.457.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.506.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 44\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 5\n",
      "    total number of stems: 8\n",
      "    total number of non-pairing interactions: 93\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 25\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 4/1074 [00:03<13:04,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5l4o.cif1'\n",
      "  1.A.H2U.20 0.120\n",
      "  1.A.U.60 0.121\n",
      "1..A.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 76\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 83\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 5/1074 [00:03<12:30,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xs2.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 88\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 6/1074 [00:03<09:47,  1.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cb3.cif1'\n",
      "  1.A.U.44 0.126\n",
      "    total number of nucleotides: 200\n",
      "    total number of base pairs: 92\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 244\n",
      "    total number of atom-base capping interactions: 18\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 6\n",
      "    total number of phosphate interactions: 44\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 7/1074 [00:04<08:38,  2.06it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ht9.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 182\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 8/1074 [00:04<07:14,  2.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6doy.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 135\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 9/1074 [00:04<06:16,  2.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3avt.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 1203\n",
      "[w] arginine [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.564.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.573.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.612.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.618.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.710.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.711.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.752.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.914.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.937.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.939.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.997.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1088.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1197.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1241.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1249.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2\n",
      "    total number of helices: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 10/1074 [00:06<12:43,  1.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4p3u.cif1'\n",
      "    total number of nucleotides: 86\n",
      "    total number of base pairs: 40\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 71\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 11/1074 [00:06<11:16,  1.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6i0v.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 346\n",
      "[w] arginine [1..A.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.275.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.327.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.275.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.327.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 12/1074 [00:07<10:24,  1.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3jcs.cif1'\n",
      "  1.2.H2U.1404 0.198\n",
      "    total number of nucleotides: 3420\n",
      "    total number of amino acids: 5847\n",
      "[w] arginine [1..A.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.227.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.227.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.341.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.376.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.199.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.172.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.234.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.310.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.172.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.172.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..h.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..h.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..h.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..h.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..h.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..k.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..k.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..l.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..l.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..n.ARG.45.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 1524\n",
      "    total number of multiplets: 195\n",
      "    total number of helices: 105\n",
      "    total number of stems: 184\n",
      "    total number of isolated WC/wobble pairs: 100\n",
      "    total number of non-pairing interactions: 4630\n",
      "    total number of atom-base capping interactions: 250\n",
      "    total number of splayed-apart dinucleotides: 509\n",
      "                        consolidated into units: 306\n",
      "    total number of hairpin loops: 52\n",
      "    total number of bulges: 40\n",
      "    total number of internal loops: 101\n",
      "    total number of junctions: 34\n",
      "    total number of non-loop single-stranded segments: 77\n",
      "    total number of kissing loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 47\n",
      "    total number of eXtended A-minor (type X) motifs: 51\n",
      "    total number of U-turns: 20\n",
      "    total number of ribose zippers: 35\n",
      "    total number of kink turns: 3\n",
      "    total number of phosphate interactions: 1243\n",
      "[w]  number-of-residues=280 -- in shortened form\n",
      "[w]  number-of-residues=280 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|          | 13/1074 [00:39<2:58:54, 10.12s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bbm.cif1'\n",
      "    total number of nucleotides: 61\n",
      "    total number of base pairs: 30\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 70\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 10\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|▏         | 14/1074 [00:40<2:08:37,  7.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1m8y.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 683\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|▏         | 15/1074 [00:40<1:33:11,  5.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ftb.cif1'\n",
      "  *1.A.EPE.404 0.255 (planarity=0.279)\n",
      "  1.A.EPE.404 0.255\n",
      "[i] taken as a pyrimidine ... 1.A.EPE.404\n",
      "  *1.B.EPE.403 0.254 (planarity=0.276)\n",
      "  1.B.EPE.403 0.254\n",
      "[i] taken as a pyrimidine ... 1.B.EPE.403\n",
      "  *1.C.EPE.401 0.256 (planarity=0.281)\n",
      "  1.C.EPE.401 0.256\n",
      "[i] taken as a pyrimidine ... 1.C.EPE.401\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 1001\n",
      "[w] Residue 1..A.EPE.404. missing planar atom [ C4 ]\n",
      "[w] Residue 1..A.EPE.404. missing planar atom [ N3 ]\n",
      "[w] Residue 1..B.EPE.403. missing planar atom [ C4 ]\n",
      "[w] Residue 1..B.EPE.403. missing planar atom [ N3 ]\n",
      "[w] Residue 1..C.EPE.401. missing planar atom [ C4 ]\n",
      "[w] Residue 1..C.EPE.401. missing planar atom [ N3 ]\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  1%|▏         | 16/1074 [00:41<1:08:23,  3.88s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hc9.cif1'\n",
      "    total number of nucleotides: 152\n",
      "    total number of amino acids: 835\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.183.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.235.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.265.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.302.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.314.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.235.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.265.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.302.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.314.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 63\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 183\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 32\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 17/1074 [00:42<53:55,  3.06s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ycp.cif1'\n",
      "    total number of nucleotides: 61\n",
      "    total number of amino acids: 308\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 71\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 18/1074 [00:43<41:00,  2.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4kzd.cif1'\n",
      "    total number of nucleotides: 84\n",
      "    total number of amino acids: 440\n",
      "    total number of base pairs: 41\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 95\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of G-tetrads: 2\n",
      "    total number of G4-helices: 1\n",
      "    total number of G4-stems: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 19/1074 [00:44<34:03,  1.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dp8.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 136\n",
      "[w] arginine [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 20/1074 [00:44<25:35,  1.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v8d.cif1'\n",
      "    total number of nucleotides: 9593\n",
      "    total number of amino acids: 11511\n",
      "[w] arginine [1..AG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AG.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AJ.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AT.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AW.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BD.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BD.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BE.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BE.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BE.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BE.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BG.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B0.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B0.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B1.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B1.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B1.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B3.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B3.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CW.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D4.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D6.ARG.37.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4771\n",
      "    total number of multiplets: 697\n",
      "    total number of helices: 276\n",
      "    total number of stems: 605\n",
      "    total number of isolated WC/wobble pairs: 178\n",
      "    total number of non-pairing interactions: 13869\n",
      "    total number of atom-base capping interactions: 661\n",
      "    total number of splayed-apart dinucleotides: 1232\n",
      "                        consolidated into units: 785\n",
      "    total number of hairpin loops: 241\n",
      "    total number of bulges: 128\n",
      "    total number of internal loops: 231\n",
      "    total number of junctions: 120\n",
      "    total number of non-loop single-stranded segments: 50\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v8d.cif1', hairpin loops (#45/#47) have 2 interactions [#123, #124 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v8d.cif1', hairpin loops (#166/#168) have 2 interactions [#426, #427 ...]\n",
      "    total number of kissing loops: 20\n",
      "    total number of A-minor (types I and II) motifs: 264\n",
      "    total number of eXtended A-minor (type X) motifs: 184\n",
      "    total number of U-turns: 105\n",
      "    total number of ribose zippers: 132\n",
      "    total number of kink turns: 22\n",
      "    total number of phosphate interactions: 3275\n",
      "[w]  number-of-residues=467 -- in shortened form\n",
      "[w]  number-of-residues=467 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:03:44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 21/1074 [04:28<19:57:15, 68.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yz9.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 438\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 22/1074 [04:29<14:00:22, 47.93s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5m3h.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 2195\n",
      "[w] arginine [1..A.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.610.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.706.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.723.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.610.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.706.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.723.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.427.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.737.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 21\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 23/1074 [04:31<10:00:37, 34.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ytv.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 75\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 24/1074 [04:31<7:00:56, 24.05s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wzk.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 513\n",
      "[w] arginine [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.295.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.360.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.585.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.595.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.295.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.360.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.585.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.595.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 25/1074 [04:32<4:57:30, 17.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ix1.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 643\n",
      "[w] arginine [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  2%|▏         | 26/1074 [04:32<3:31:17, 12.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4py5.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 267\n",
      "[w] arginine [1..A.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.222.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.222.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 39\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 27/1074 [04:33<2:29:17,  8.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wt1.cif1'\n",
      "    total number of nucleotides: 137\n",
      "    total number of amino acids: 664\n",
      "[w] arginine [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.313.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.313.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.313.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.313.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 62\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 152\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 22\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 46\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 28/1074 [04:33<1:47:27,  6.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6bhj.cif1'\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 1933\n",
      "[w] arginine [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 30\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 29/1074 [04:35<1:23:31,  4.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6az4.cif1'\n",
      "    total number of nucleotides: 41\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 45\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 30/1074 [04:35<59:33,  3.42s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2nue.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of amino acids: 440\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 50\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 31/1074 [04:36<44:07,  2.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4znp.cif1'\n",
      "    total number of nucleotides: 139\n",
      "    total number of base pairs: 62\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 4\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 176\n",
      "    total number of atom-base capping interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 32/1074 [04:36<33:53,  1.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3knc.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 33/1074 [04:36<24:48,  1.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pkd.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of amino acids: 240\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 46\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 34/1074 [04:37<19:23,  1.12s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6o16.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 1595\n",
      "[w] arginine [1..A.ARG.304.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.340.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.391.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.486.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.641.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.826.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.304.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.340.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.391.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.486.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.641.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.826.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.340.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.391.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.607.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.641.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.696.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.766.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.834.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.885.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 35/1074 [04:38<21:20,  1.23s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6e8u.cif1'\n",
      "    total number of nucleotides: 37\n",
      "    total number of base pairs: 23\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 3\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of G-tetrads: 2\n",
      "    total number of G4-helices: 1\n",
      "    total number of G4-stems: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 36/1074 [04:38<16:03,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3trz.cif1'\n",
      "    total number of nucleotides: 126\n",
      "    total number of amino acids: 780\n",
      "[w] arginine [1..A.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.138.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 36\n",
      "    total number of helices: 6\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 84\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 6\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  3%|▎         | 37/1074 [04:40<20:20,  1.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5o1y.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 163\n",
      "[w] arginine [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▎         | 38/1074 [04:41<15:58,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rj1.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▎         | 39/1074 [04:41<12:04,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5emf.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of isolated WC/wobble pairs: 8\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▎         | 40/1074 [04:41<09:23,  1.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1pgl.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 555\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 41/1074 [04:41<08:37,  1.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3l25.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 492\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 17\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 42/1074 [04:42<08:13,  2.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fdv.cif1'\n",
      "  1.1A.U.12 0.122\n",
      "  1.1A.U.274 0.124\n",
      "  1.1A.U.330 0.126\n",
      "  1.1A.U.880 0.125\n",
      "  1.1A.U.1003 0.124\n",
      "  1.1A.U.1106 0.126\n",
      "  1.1A.U.1112 0.130\n",
      "  1.1A.U.1128 0.133\n",
      "  1.1A.U.1143 0.131\n",
      "1..1A.PSU.1933. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.1939. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.2617. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 8981\n",
      "    total number of amino acids: 11863\n",
      "[w] arginine [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1G.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..1B.A.57. missing planar atom [ C8 ]\n",
      "[w] Residue [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1G.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Q.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1W.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1X.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..10.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..12.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..13.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..14.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2x.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1b.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1b.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1c.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1e.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1h.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1p.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1x.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2O.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..20.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..22.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..23.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..24.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..28.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2h.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2n.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2s.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4640\n",
      "    total number of multiplets: 747\n",
      "    total number of helices: 261\n",
      "    total number of stems: 566\n",
      "    total number of isolated WC/wobble pairs: 168\n",
      "    total number of non-pairing interactions: 12827\n",
      "    total number of atom-base capping interactions: 667\n",
      "    total number of splayed-apart dinucleotides: 1175\n",
      "                        consolidated into units: 733\n",
      "    total number of hairpin loops: 207\n",
      "    total number of bulges: 135\n",
      "    total number of internal loops: 209\n",
      "    total number of junctions: 112\n",
      "    total number of non-loop single-stranded segments: 65\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fdv.cif1', hairpin loops (#1/#3) have 2 interactions [#10, #11 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fdv.cif1', hairpin loops (#105/#107) have 2 interactions [#292, #293 ...]\n",
      "    total number of kissing loops: 14\n",
      "    total number of A-minor (types I and II) motifs: 289\n",
      "    total number of eXtended A-minor (type X) motifs: 202\n",
      "    total number of U-turns: 99\n",
      "    total number of ribose zippers: 127\n",
      "    total number of kink turns: 25\n",
      "    total number of phosphate interactions: 3543\n",
      "[w]  number-of-residues=367 -- in shortened form\n",
      "[w]  number-of-residues=367 -- in shortened form\n",
      "\n",
      "Time used: 00:00:03:30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 43/1074 [08:12<18:10:28, 63.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5z1i.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 44/1074 [08:12<12:43:43, 44.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2dra.cif1'\n",
      "    total number of nucleotides: 35\n",
      "    total number of amino acids: 437\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 45/1074 [08:13<8:57:28, 31.34s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ec0.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of amino acids: 940\n",
      "[w] arginine [1..D.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 32\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 46/1074 [08:14<6:22:51, 22.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4erd.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 216\n",
      "[w] arginine [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.465.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 22\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 47/1074 [08:15<4:30:40, 15.81s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5une.cif1'\n",
      "    total number of nucleotides: 94\n",
      "    total number of base pairs: 44\n",
      "    total number of helices: 1\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 109\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of internal loops: 9\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  4%|▍         | 48/1074 [08:15<3:10:44, 11.15s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5v3f.cif1'\n",
      "    total number of nucleotides: 50\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 65\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of G-tetrads: 6\n",
      "    total number of G4-helices: 1\n",
      "    total number of G4-stems: 2\n",
      "    total number of phosphate interactions: 24\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▍         | 49/1074 [08:16<2:15:47,  7.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1p79.cif1'\r\n",
      "    total number of nucleotides: 5\r\n",
      "    total number of non-pairing interactions: 4\r\n",
      "    total number of splayed-apart dinucleotides: 2\r\n",
      "                        consolidated into units: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▍         | 50/1074 [08:16<1:35:44,  5.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2du3.cif1'\n",
      "    total number of nucleotides: 71\n",
      "    total number of amino acids: 1001\n",
      "[w] arginine [1..A.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.263.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.265.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.270.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.451.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.509.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.263.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.265.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.270.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.451.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.509.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 32\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▍         | 51/1074 [08:19<1:21:37,  4.79s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jxs.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▍         | 52/1074 [08:19<1:00:19,  3.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wtk.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of amino acids: 1215\n",
      "[w] arginine [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.820.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.820.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1278.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▍         | 53/1074 [08:22<56:56,  3.35s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1f1t.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of base pairs: 18\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▌         | 54/1074 [08:23<40:51,  2.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zsb.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 1547\n",
      "[w] arginine [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▌         | 55/1074 [08:25<40:11,  2.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fq5.cif1'\n",
      "    total number of nucleotides: 117\n",
      "    total number of amino acids: 1318\n",
      "[w] arginine [1..B.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 47\n",
      "    total number of helices: 6\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 118\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 57\n",
      "\n",
      "Time used: 00:00:00:05\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▌         | 56/1074 [08:30<56:23,  3.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5id6.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of amino acids: 1209\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.737.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.883.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.737.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.883.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 9\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▌         | 57/1074 [08:32<49:44,  2.93s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5z4d.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 351\n",
      "[w] arginine [1..A.ARG.295.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.295.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▌         | 58/1074 [08:33<39:31,  2.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ev4.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 197\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  5%|▌         | 59/1074 [08:34<28:50,  1.70s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rby.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 60/1074 [08:34<21:02,  1.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1tfw.cif1'\n",
      "    total number of nucleotides: 78\n",
      "    total number of amino acids: 1748\n",
      "    total number of base pairs: 31\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 82\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of phosphate interactions: 25\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 61/1074 [08:36<24:27,  1.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zh5.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 369\n",
      "[w] arginine [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.91.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 62/1074 [08:36<20:00,  1.19s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3u2e.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 515\n",
      "[w] arginine [1..A.ARG.370.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.404.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.407.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.301.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.370.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.404.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.370.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.404.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.407.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.301.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.370.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.404.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.407.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.470.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.487.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 63/1074 [08:37<16:02,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ol9.cif1'\n",
      "    total number of nucleotides: 144\n",
      "    total number of amino acids: 1844\n",
      "    total number of base pairs: 67\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of non-pairing interactions: 144\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of phosphate interactions: 36\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 64/1074 [08:39<25:07,  1.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ts2.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 272\n",
      "[w] arginine [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.138.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 65/1074 [08:40<19:46,  1.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4lj0.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 129\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 66/1074 [08:40<16:01,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1rmv.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 156\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▌         | 67/1074 [08:41<13:45,  1.22it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5bjo.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of base pairs: 41\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 6\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 89\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of G-tetrads: 4\n",
      "    total number of G4-helices: 2\n",
      "    total number of G4-stems: 2\n",
      "    total number of ribose zippers: 4\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▋         | 68/1074 [08:41<10:35,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ozb.cif1'\n",
      "    total number of nucleotides: 66\n",
      "    total number of amino acids: 730\n",
      "[w] arginine [1..B.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.293.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.293.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  6%|▋         | 69/1074 [08:42<11:00,  1.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5aor.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 2001\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 70/1074 [08:46<28:01,  1.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q0n.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 681\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 71/1074 [08:46<22:56,  1.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3d0m.cif1'\r\n",
      "    total number of nucleotides: 16\r\n",
      "    total number of non-pairing interactions: 14\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 72/1074 [08:47<16:49,  1.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5amq.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 1666\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 73/1074 [08:51<34:24,  2.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cgr.cif1'\n",
      "1..A.PSU.7. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 25\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 74/1074 [08:51<25:20,  1.52s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1y27.cif1'\n",
      "  1.X.CCC.81 0.186\n",
      "    total number of nucleotides: 69\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 78\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 75/1074 [08:52<19:43,  1.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wzi.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 533\n",
      "[w] arginine [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.329.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.329.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.619.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 76/1074 [08:52<17:17,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5h3u.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 1332\n",
      "[w] arginine [1..A.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.335.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.682.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.335.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.682.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 77/1074 [08:54<20:01,  1.21s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2gjw.cif1'\n",
      "  1.F.A23.14 0.197\n",
      "  1.J.A23.14 0.217\n",
      "    total number of nucleotides: 75\n",
      "    total number of amino acids: 1223\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 73\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "    total number of bulges: 2\n",
      "    total number of non-loop single-stranded segments: 12\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 78/1074 [08:55<20:49,  1.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5eme.cif1'\n",
      "  1.B.TPN.106 0.120\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of isolated WC/wobble pairs: 8\n",
      "    total number of non-pairing interactions: 16\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 79/1074 [08:56<15:37,  1.06it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3pdr.cif1'\n",
      "    total number of nucleotides: 322\n",
      "    total number of base pairs: 163\n",
      "    total number of multiplets: 23\n",
      "    total number of helices: 7\n",
      "    total number of stems: 20\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 448\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 39\n",
      "                        consolidated into units: 22\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of A-minor (types I and II) motifs: 13\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 69\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  7%|▋         | 80/1074 [08:58<21:13,  1.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jcf.cif1'\n",
      "    total number of nucleotides: 43\n",
      "    total number of amino acids: 1320\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 45\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 81/1074 [08:59<21:02,  1.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1rpu.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 275\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 82/1074 [08:59<16:29,  1.00it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ztm.cif1'\n",
      "    total number of nucleotides: 49\n",
      "    total number of amino acids: 332\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 83/1074 [09:00<13:10,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t2a.cif1'\n",
      "  1.2.U.56 0.124\n",
      "  1.2.U.2046 0.122\n",
      "    total number of nucleotides: 5277\n",
      "    total number of amino acids: 10932\n",
      "[w] arginine [1..J.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..K.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..M.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..O.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..a.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..c.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..d.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..j.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..u.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..u.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..u.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..u.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.14.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2272\n",
      "    total number of multiplets: 291\n",
      "    total number of helices: 174\n",
      "    total number of stems: 310\n",
      "    total number of isolated WC/wobble pairs: 152\n",
      "    total number of non-pairing interactions: 6875\n",
      "    total number of atom-base capping interactions: 280\n",
      "    total number of splayed-apart dinucleotides: 919\n",
      "                        consolidated into units: 510\n",
      "    total number of hairpin loops: 100\n",
      "    total number of bulges: 77\n",
      "    total number of internal loops: 161\n",
      "    total number of junctions: 57\n",
      "    total number of non-loop single-stranded segments: 98\n",
      "    total number of kissing loops: 6\n",
      "    total number of A-minor (types I and II) motifs: 90\n",
      "    total number of eXtended A-minor (type X) motifs: 96\n",
      "    total number of U-turns: 45\n",
      "    total number of ribose zippers: 46\n",
      "    total number of kink turns: 3\n",
      "    total number of phosphate interactions: 1967\n",
      "[w]  number-of-residues=358 -- in shortened form\n",
      "[w]  number-of-residues=358 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:01:12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 84/1074 [10:13<6:10:16, 22.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t7b.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 802\n",
      "[w] arginine [1..A.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.635.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.668.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.635.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.668.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 85/1074 [10:14<4:23:43, 16.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rwn.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of amino acids: 349\n",
      "[w] arginine [1..A.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.125.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 86/1074 [10:14<3:06:45, 11.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5els.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 672\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 87/1074 [10:14<2:12:14,  8.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xs5.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 167\n",
      "    total number of base pairs: 2\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 88/1074 [10:15<1:33:32,  5.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v99.cif1'\n",
      "    total number of nucleotides: 2040\n",
      "    total number of amino acids: 70920\n",
      "    total number of base pairs: 660\n",
      "    total number of helices: 120\n",
      "    total number of stems: 120\n",
      "    total number of non-pairing interactions: 2091\n",
      "    total number of splayed-apart dinucleotides: 240\n",
      "                        consolidated into units: 120\n",
      "    total number of hairpin loops: 120\n",
      "    total number of non-loop single-stranded segments: 240\n",
      "    total number of phosphate interactions: 588\n",
      "\n",
      "Time used: 00:00:10:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 89/1074 [20:18<50:35:45, 184.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ok4.cif1'\n",
      "    total number of nucleotides: 60\n",
      "    total number of base pairs: 30\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 51\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 90/1074 [20:18<35:26:15, 129.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4zt0.cif1'\n",
      "    total number of nucleotides: 144\n",
      "    total number of amino acids: 2660\n",
      "[w] arginine [1..A.ARG.424.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.535.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.783.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.884.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.919.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.864.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.925.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.938.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.424.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.535.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.783.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.884.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.919.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.864.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.925.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.938.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1114.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 42\n",
      "    total number of helices: 8\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 136\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 16\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 68\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  8%|▊         | 91/1074 [20:23<25:08:41, 92.09s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/485d.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▊         | 92/1074 [20:23<17:36:39, 64.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ola.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 789\n",
      "[w] arginine [1..A.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▊         | 93/1074 [20:24<12:22:56, 45.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5i4a.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 1266\n",
      "[w] arginine [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 94/1074 [20:26<8:48:32, 32.36s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ob3.cif1'\n",
      "    total number of nucleotides: 69\n",
      "    total number of base pairs: 36\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 3\n",
      "    total number of G-tetrads: 2\n",
      "    total number of G4-helices: 1\n",
      "    total number of G4-stems: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 95/1074 [20:26<6:10:44, 22.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5m0i.cif1'\n",
      "    total number of nucleotides: 56\n",
      "    total number of amino acids: 960\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 96/1074 [20:27<4:22:57, 16.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1vqn.cif1'\n",
      "1..0.PSU.2621. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 2882\n",
      "    total number of amino acids: 3776\n",
      "    total number of base pairs: 1468\n",
      "    total number of multiplets: 246\n",
      "    total number of helices: 90\n",
      "    total number of stems: 179\n",
      "    total number of isolated WC/wobble pairs: 56\n",
      "    total number of non-pairing interactions: 4038\n",
      "    total number of atom-base capping interactions: 235\n",
      "    total number of splayed-apart dinucleotides: 409\n",
      "                        consolidated into units: 259\n",
      "    total number of hairpin loops: 68\n",
      "    total number of bulges: 36\n",
      "    total number of internal loops: 67\n",
      "    total number of junctions: 36\n",
      "    total number of non-loop single-stranded segments: 35\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/1vqn.cif1', hairpin loops (#1/#3) have 2 interactions [#8, #9 ...]\n",
      "    total number of kissing loops: 5\n",
      "    total number of A-minor (types I and II) motifs: 105\n",
      "    total number of eXtended A-minor (type X) motifs: 54\n",
      "    total number of U-turns: 32\n",
      "    total number of ribose zippers: 45\n",
      "    total number of kink turns: 8\n",
      "    total number of phosphate interactions: 1160\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:34\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 97/1074 [21:02<5:54:17, 21.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/419d.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 28\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 98/1074 [21:02<4:08:44, 15.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zq8.cif1'\n",
      "  1.C.MUM.747 0.172\n",
      "  1.D.MUM.747 0.133\n",
      "    total number of nucleotides: 26\n",
      "    total number of amino acids: 910\n",
      "[w] arginine [1..B.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.257.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.402.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.257.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.402.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.402.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 26\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 99/1074 [21:03<2:58:43, 11.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rge.cif1'\n",
      "    total number of nucleotides: 167\n",
      "    total number of base pairs: 90\n",
      "    total number of multiplets: 12\n",
      "    total number of helices: 6\n",
      "    total number of stems: 18\n",
      "    total number of non-pairing interactions: 211\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 27\n",
      "                        consolidated into units: 18\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 3\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of A-minor (types I and II) motifs: 5\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 39\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 100/1074 [21:04<2:10:58,  8.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3c3z.cif1'\n",
      "  1.B.UMS.3 0.122\n",
      "    total number of nucleotides: 46\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 47\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 101/1074 [21:04<1:33:19,  5.75s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6bm4.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 3500\n",
      "[w] arginine [1..A.ARG.839.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..H.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.839.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.25.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  9%|▉         | 102/1074 [21:08<1:23:52,  5.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5kvj.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 35\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|▉         | 103/1074 [21:09<1:00:30,  3.74s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5tpy.cif1'\n",
      "    total number of nucleotides: 71\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 87\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "    total number of hairpin loops: 4\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|▉         | 104/1074 [21:09<44:28,  2.75s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5bzu.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 364\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|▉         | 105/1074 [21:09<33:01,  2.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r1d.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 1\n",
      "    total number of stems: 8\n",
      "    total number of non-pairing interactions: 54\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of internal loops: 6\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|▉         | 106/1074 [21:10<24:17,  1.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/364d.cif1'\n",
      "    total number of nucleotides: 60\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 65\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|▉         | 107/1074 [21:10<18:13,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3czw.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|█         | 108/1074 [21:10<13:34,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jah.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 57\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 3\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|█         | 109/1074 [21:10<10:30,  1.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4prf.cif1'\n",
      "    total number of nucleotides: 74\n",
      "    total number of amino acids: 95\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|█         | 110/1074 [21:11<08:39,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1cvj.cif1'\n",
      "    total number of nucleotides: 67\n",
      "    total number of amino acids: 1245\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 16\n",
      "    total number of phosphate interactions: 25\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|█         | 111/1074 [21:12<11:09,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2y9h.cif1'\n",
      "    total number of nucleotides: 144\n",
      "    total number of amino acids: 1590\n",
      "    total number of base pairs: 54\n",
      "    total number of helices: 8\n",
      "    total number of stems: 8\n",
      "    total number of non-pairing interactions: 132\n",
      "    total number of splayed-apart dinucleotides: 22\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 8\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 67\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 10%|█         | 112/1074 [21:13<12:22,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qvc.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 364\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 113/1074 [21:13<09:58,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xzl.cif1'\n",
      "  1.A.ADP.1854 0.130\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 756\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 114/1074 [21:14<10:21,  1.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q0p.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 681\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 115/1074 [21:15<11:37,  1.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2i82.cif1'\n",
      "    total number of nucleotides: 84\n",
      "    total number of amino acids: 862\n",
      "    total number of base pairs: 32\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 84\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 4\n",
      "    total number of phosphate interactions: 34\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 116/1074 [21:16<13:05,  1.22it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2qkb.cif1'\n",
      "    total number of nucleotides: 39\n",
      "    total number of amino acids: 302\n",
      "[w] arginine [1..A.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.179.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 41\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 117/1074 [21:16<11:25,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rbq.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of non-pairing interactions: 31\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 118/1074 [21:16<09:31,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1di2.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of amino acids: 129\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 119/1074 [21:17<07:32,  2.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3fs0.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 21\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█         | 120/1074 [21:17<06:05,  2.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wtm.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 553\n",
      "[w] arginine [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█▏        | 121/1074 [21:17<06:43,  2.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5u3g.cif1'\n",
      "    total number of nucleotides: 85\n",
      "    total number of base pairs: 43\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 4\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 99\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█▏        | 122/1074 [21:18<06:17,  2.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ngd.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 280\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.986.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.993.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.986.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.993.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 11%|█▏        | 123/1074 [21:18<06:48,  2.33it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yzd.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 50\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 124/1074 [21:18<05:35,  2.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/409d.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 49\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 125/1074 [21:18<04:57,  3.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6i3p.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 2509\n",
      "[w] arginine [1..A.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1012.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1080.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1012.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1080.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.815.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.825.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1065.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1012.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.670.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1096.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 25\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 126/1074 [21:21<13:50,  1.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4o41.cif1'\n",
      "  1.D.U.412 0.120\n",
      "    total number of nucleotides: 45\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 45\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 127/1074 [21:21<11:06,  1.42it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wlh.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 1289\n",
      "[w] arginine [1..A.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.600.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.736.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.600.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.736.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.789.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1047.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1257.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1324.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 31\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 128/1074 [21:23<18:42,  1.19s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5c0y.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 774\n",
      "[w] arginine [1..A.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.363.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.399.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.407.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.363.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.399.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.407.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.363.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.443.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.510.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 129/1074 [21:24<15:53,  1.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3glp.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 130/1074 [21:24<11:58,  1.31it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ah5.cif1'\n",
      "    total number of nucleotides: 163\n",
      "    total number of amino acids: 1560\n",
      "    total number of base pairs: 74\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 169\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 27\n",
      "                        consolidated into units: 19\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 131/1074 [21:26<17:10,  1.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1nta.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 18\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 132/1074 [21:26<13:59,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2pjp.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of amino acids: 121\n",
      "[w] arginine [1..A.ARG.606.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.606.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 133/1074 [21:27<10:50,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lr5.cif1'\n",
      "  1.A.CBV.9 0.128\n",
      "  1.B.CBV.9 0.129\n",
      "  1.C.CBV.9 0.128\n",
      "  1.D.CBV.9 0.127\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 44\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 12%|█▏        | 134/1074 [21:27<08:31,  1.84it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6qw6.cif1'\n",
      "  1.5.U.58 0.128\n",
      "    total number of nucleotides: 272\n",
      "    total number of amino acids: 10350\n",
      "[w] arginine [1..X.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..41.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..42.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..42.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..4A.ARG.507.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..4A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..41.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..42.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..42.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4A.ARG.507.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4A.ARG.525.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4A.ARG.547.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4A.ARG.549.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4A.ARG.596.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.232.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.270.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.325.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.328.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.348.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.390.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.395.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.398.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.439.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4B.ARG.440.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4C.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4C.ARG.351.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4C.ARG.357.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4C.ARG.360.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4C.ARG.365.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4e.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4f.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4g.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..51.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..52.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.158.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.227.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.362.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.419.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.539.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.552.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.686.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.707.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.762.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.845.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.883.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.995.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1370.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1414.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1471.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1533.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.1865.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5A.ARG.2008.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.325.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.436.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.519.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.558.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.700.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.728.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.739.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.753.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.763.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.931.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.952.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.986.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1093.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1166.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1274.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1366.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1442.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1570.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1588.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1648.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1693.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1762.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.1993.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.2030.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5B.ARG.2081.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5C.ARG.569.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5C.ARG.730.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5C.ARG.814.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5C.ARG.940.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5D.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5D.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.367.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.374.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.378.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.391.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.405.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.420.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.628.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.669.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.676.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.703.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.729.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.770.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.855.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5J.ARG.907.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5O.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5O.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.339.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.461.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.703.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.751.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5X.ARG.754.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5e.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5f.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..5g.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..63.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..63.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..63.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..64.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..64.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..64.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..64.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..65.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..65.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..65.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..66.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..66.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..67.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..67.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..68.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..68.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..68.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..68.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.382.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.399.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.455.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.468.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.636.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.264.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.282.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.354.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.440.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.515.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.555.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 104\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 9\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 266\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 31\n",
      "                        consolidated into units: 20\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 7\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "    total number of U-turns: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 93\n",
      "\n",
      "Time used: 00:00:00:23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 135/1074 [21:50<1:54:05,  7.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ib7.cif1'\n",
      "  1.13.U.560 0.121\n",
      "  1.13.U.843 0.121\n",
      "  1.13.U.960 0.121\n",
      "  1.13.U.1056 0.121\n",
      "  1.1K.U8U.34 0.132\n",
      "  1.2K.7MG.47 0.136\n",
      "  1.1H.U.9 0.121\n",
      "  1.1H.U.270L 0.122\n",
      "  1.1H.U.270O 0.121\n",
      "1..1K.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1K.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2K.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2L.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9409\n",
      "    total number of amino acids: 11336\n",
      "[w] arginine [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..7E.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..71.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..7E.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..71.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..41.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B8.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q8.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..62.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..19.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..29.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..49.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..95.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E5.ARG.32.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4660\n",
      "    total number of multiplets: 676\n",
      "    total number of helices: 262\n",
      "    total number of stems: 585\n",
      "    total number of isolated WC/wobble pairs: 182\n",
      "    total number of non-pairing interactions: 13507\n",
      "    total number of atom-base capping interactions: 669\n",
      "    total number of splayed-apart dinucleotides: 1204\n",
      "                        consolidated into units: 767\n",
      "    total number of hairpin loops: 222\n",
      "    total number of bulges: 123\n",
      "    total number of internal loops: 237\n",
      "    total number of junctions: 111\n",
      "    total number of non-loop single-stranded segments: 102\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ib7.cif1', hairpin loops (#39/#41) have 2 interactions [#117, #118 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ib7.cif1', hairpin loops (#150/#152) have 2 interactions [#411, #412 ...]\n",
      "    total number of kissing loops: 16\n",
      "    total number of A-minor (types I and II) motifs: 245\n",
      "    total number of eXtended A-minor (type X) motifs: 200\n",
      "    total number of U-turns: 104\n",
      "    total number of ribose zippers: 128\n",
      "    total number of kink turns: 19\n",
      "    total number of phosphate interactions: 3439\n",
      "[w]  number-of-residues=484 -- in shortened form\n",
      "[w]  number-of-residues=484 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:03:07\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 136/1074 [24:57<15:59:27, 61.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5o7h.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 1370\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 29\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 137/1074 [24:58<11:15:28, 43.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1hc8.cif1'\n",
      "    total number of nucleotides: 116\n",
      "    total number of amino acids: 144\n",
      "    total number of base pairs: 57\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 158\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of U-turns: 6\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 47\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 138/1074 [24:59<7:56:21, 30.54s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2q1o.cif1'\n",
      "1..A.NF2.104. has unusual GLYCOSIDIC bond!\n",
      "1..B.NF2.104. has unusual GLYCOSIDIC bond!\n",
      "1..C.NF2.104. has unusual GLYCOSIDIC bond!\n",
      "1..D.NF2.104. has unusual GLYCOSIDIC bond!\n",
      "    total number of nucleotides: 48\n",
      "[w] Residue 1..A.NF2.104. missing planar atom [ N3 ]\n",
      "[w] Residue 1..A.NF2.104. missing planar atom [ N1 ]\n",
      "\t[i] 1..A.NF2.104.: replace missing base ring atom [ N3 ] with [C3] (d=0.08)\n",
      "\t[i] 1..A.NF2.104.: replace missing base ring atom [ N1 ] with [C1] (d=0.03)\n",
      "[w] Residue 1..B.NF2.104. missing planar atom [ N3 ]\n",
      "[w] Residue 1..B.NF2.104. missing planar atom [ N1 ]\n",
      "\t[i] 1..B.NF2.104.: replace missing base ring atom [ N3 ] with [C3] (d=0.09)\n",
      "\t[i] 1..B.NF2.104.: replace missing base ring atom [ N1 ] with [C1] (d=0.03)\n",
      "[w] Residue 1..C.NF2.104. missing planar atom [ N3 ]\n",
      "[w] Residue 1..C.NF2.104. missing planar atom [ N1 ]\n",
      "\t[i] 1..C.NF2.104.: replace missing base ring atom [ N3 ] with [C3] (d=0.13)\n",
      "\t[i] 1..C.NF2.104.: replace missing base ring atom [ N1 ] with [C1] (d=0.06)\n",
      "[w] Residue 1..D.NF2.104. missing planar atom [ N3 ]\n",
      "[w] Residue 1..D.NF2.104. missing planar atom [ N1 ]\n",
      "\t[i] 1..D.NF2.104.: replace missing base ring atom [ N3 ] with [C3] (d=0.09)\n",
      "\t[i] 1..D.NF2.104.: replace missing base ring atom [ N1 ] with [C1] (d=0.03)\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of internal loops: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 139/1074 [24:59<5:34:00, 21.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/354d.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 140/1074 [25:00<3:54:28, 15.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wzm.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 475\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 141/1074 [25:00<2:45:58, 10.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gd3.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 250\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 142/1074 [25:00<1:56:58,  7.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5x2g.cif1'\n",
      "    total number of nucleotides: 129\n",
      "    total number of amino acids: 749\n",
      "[w] arginine [1..A.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.645.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.866.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.645.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.866.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 62\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 5\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 134\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 57\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 143/1074 [25:01<1:25:46,  5.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5y58.cif1'\n",
      "    total number of nucleotides: 90\n",
      "    total number of amino acids: 3337\n",
      "[w] arginine [1..A.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.448.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.448.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.435.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.448.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.435.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.448.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.435.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.621.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 33\n",
      "    total number of helices: 6\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 96\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 3\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 37\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 13%|█▎        | 144/1074 [25:04<1:14:28,  4.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5awh.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of amino acids: 1474\n",
      "[w] arginine [1..B.ARG.421.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.421.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▎        | 145/1074 [25:06<1:01:06,  3.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5d0b.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 751\n",
      "[w] Residue 1..F.56B.34. missing planar atom [ N7 ]\n",
      "\t[i] 1..F.56B.34.: replace missing base ring atom [ N7 ] with [C7] (d=0.04)\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▎        | 146/1074 [25:07<45:07,  2.92s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2g91.cif1'\r\n",
      "    total number of nucleotides: 17\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 14\r\n",
      "    total number of splayed-apart dinucleotides: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▎        | 147/1074 [25:07<32:18,  2.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4u7u.cif1'\n",
      "    total number of nucleotides: 122\n",
      "    total number of amino acids: 6701\n",
      "[w] arginine [1..A.ARG.425.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..K.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.425.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.318.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 96\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 20\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 81\n",
      "\n",
      "Time used: 00:00:00:05\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 148/1074 [25:12<48:20,  3.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5b63.cif1'\n",
      "    total number of nucleotides: 127\n",
      "    total number of amino acids: 1116\n",
      "    total number of base pairs: 51\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 145\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of kissing loops: 2\n",
      "    total number of phosphate interactions: 27\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 149/1074 [25:14<41:29,  2.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k49.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of amino acids: 1059\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 150/1074 [25:15<33:53,  2.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1mms.cif1'\n",
      "    total number of nucleotides: 116\n",
      "    total number of amino acids: 203\n",
      "    total number of base pairs: 60\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 158\n",
      "    total number of atom-base capping interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of U-turns: 6\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 46\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 151/1074 [25:17<30:41,  2.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/413d.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 152/1074 [25:17<22:22,  1.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zl3.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 302\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 153/1074 [25:17<17:51,  1.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2vuq.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 154/1074 [25:18<13:42,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1bmv.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 559\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 14%|█▍        | 155/1074 [25:18<11:50,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5l2l.cif1'\n",
      "    total number of nucleotides: 41\n",
      "    total number of amino acids: 293\n",
      "[w] arginine [1..B.ARG.477.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.477.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.477.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.477.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of non-pairing interactions: 28\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▍        | 156/1074 [25:19<11:51,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1msw.cif1'\n",
      "    total number of nucleotides: 47\n",
      "    total number of amino acids: 863\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 10\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▍        | 157/1074 [25:20<12:38,  1.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5cnr.cif1'\n",
      "    total number of nucleotides: 54\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 51\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of bulges: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▍        | 158/1074 [25:20<09:47,  1.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ann.cif1'\r\n",
      "    total number of nucleotides: 23\r\n",
      "    total number of amino acids: 148\r\n",
      "    total number of base pairs: 6\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 20\r\n",
      "    total number of splayed-apart dinucleotides: 1\r\n",
      "    total number of hairpin loops: 1\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 4\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▍        | 159/1074 [25:20<07:35,  2.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zsd.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 1547\n",
      "[w] arginine [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▍        | 160/1074 [25:22<13:32,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3vyx.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 95\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 17\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▍        | 161/1074 [25:22<10:49,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3d2v.cif1'\n",
      "    total number of nucleotides: 156\n",
      "[w] Residue 1..A.PYI.84. missing planar atom [ N1 ]\n",
      "\t[i] 1..A.PYI.84.: replace missing base ring atom [ N1 ] with [C1] (d=0.03)\n",
      "[w] Residue 1..B.PYI.82. missing planar atom [ N1 ]\n",
      "\t[i] 1..B.PYI.82.: replace missing base ring atom [ N1 ] with [C1] (d=0.03)\n",
      "    total number of base pairs: 72\n",
      "    total number of multiplets: 13\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 202\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 27\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 8\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 34\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▌        | 162/1074 [25:23<10:11,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2gxb.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 125\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▌        | 163/1074 [25:23<08:32,  1.78it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3v74.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 394\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▌        | 164/1074 [25:24<07:52,  1.93it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1wne.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▌        | 165/1074 [25:24<07:31,  2.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4g0a.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 1248\n",
      "[w] arginine [1..A.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.307.] -- NH1/NH2 naming swapped\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 15%|█▌        | 166/1074 [25:25<10:07,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6b44.cif1'\n",
      "  1.M.U.42 0.121\n",
      "  1.M.U.47 0.120\n",
      "    total number of nucleotides: 114\n",
      "    total number of amino acids: 2879\n",
      "[w] arginine [1..A.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.152.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 38\n",
      "    total number of helices: 7\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 92\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 27\n",
      "                        consolidated into units: 18\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 5\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 43\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 167/1074 [25:27<16:23,  1.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3o7v.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 107\n",
      "[w] arginine [1..X.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.318.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 168/1074 [25:27<12:50,  1.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4u35.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of internal loops: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 169/1074 [25:28<09:51,  1.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2z75.cif1'\n",
      "    total number of nucleotides: 143\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 5\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 187\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 23\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 170/1074 [25:28<09:06,  1.65it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q0q.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 343\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 171/1074 [25:29<08:22,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2po1.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 503\n",
      "[w] arginine [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.134.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 172/1074 [25:29<07:36,  1.97it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5x6b.cif1'\n",
      "    total number of nucleotides: 77\n",
      "    total number of amino acids: 989\n",
      "[w] arginine [1..I.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.356.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.356.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..I.LLP.234. missing planar atom [ N3 ]\n",
      "\t[i] 1..I.LLP.234.: replace missing base ring atom [ N3 ] with [C3] (d=0.08)\n",
      "[w] Residue 1..J.LLP.234. missing planar atom [ N3 ]\n",
      "\t[i] 1..J.LLP.234.: replace missing base ring atom [ N3 ] with [C3] (d=0.08)\n",
      "[w] Residue [1..I.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.356.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.356.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 29\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 173/1074 [25:30<11:33,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3moj.cif1'\n",
      "    total number of nucleotides: 69\n",
      "    total number of amino acids: 75\n",
      "    total number of base pairs: 29\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 78\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▌        | 174/1074 [25:31<09:32,  1.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3snp.cif1'\n",
      "    total number of nucleotides: 58\n",
      "    total number of amino acids: 1700\n",
      "[w] arginine [1..A.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.832.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 58\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 4\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▋        | 175/1074 [25:32<14:51,  1.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3g9y.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 29\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▋        | 176/1074 [25:33<12:12,  1.23it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6db9.cif1'\n",
      "    total number of nucleotides: 60\n",
      "    total number of amino acids: 428\n",
      "    total number of base pairs: 26\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 68\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 16%|█▋        | 177/1074 [25:33<10:52,  1.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5new.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 128\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 178/1074 [25:34<09:27,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ngm.cif1'\n",
      "  1.AA.U.593 0.120\n",
      "    total number of nucleotides: 4559\n",
      "    total number of amino acids: 5610\n",
      "[w] arginine [1..Ab.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Ab.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..AA.A.2150. missing planar atom [ C4 ]\n",
      "[w] Residue 1..AA.A.2150. missing planar atom [ N3 ]\n",
      "[w] Residue 1..AA.A.2150. missing planar atom [ C2 ]\n",
      "[w] Residue 1..AA.A.2150. missing planar atom [ N1 ]\n",
      "[w] Residue [1..Ab.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ag.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ai.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ai.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aj.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aj.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ak.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Al.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Al.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Al.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Al.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Al.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Am.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Am.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..An.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..An.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..An.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..An.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ao.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ao.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ao.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ao.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ao.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ao.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ap.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ap.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ap.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aq.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ar.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..At.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Au.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Au.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Av.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Av.ARG.166.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Av.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AH.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AH.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AH.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AV.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AZ.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AZ.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A2.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A2.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A2.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A3.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A3.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A3.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A4.ARG.19.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 2268\n",
      "    total number of multiplets: 330\n",
      "    total number of helices: 144\n",
      "    total number of stems: 270\n",
      "    total number of isolated WC/wobble pairs: 107\n",
      "    total number of non-pairing interactions: 6654\n",
      "    total number of atom-base capping interactions: 295\n",
      "    total number of splayed-apart dinucleotides: 590\n",
      "                        consolidated into units: 381\n",
      "    total number of hairpin loops: 108\n",
      "    total number of bulges: 56\n",
      "    total number of internal loops: 123\n",
      "    total number of junctions: 60\n",
      "    total number of non-loop single-stranded segments: 37\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ngm.cif1', hairpin loops (#34/#36) have 2 interactions [#102, #103 ...]\n",
      "    total number of kissing loops: 6\n",
      "    total number of A-minor (types I and II) motifs: 110\n",
      "    total number of eXtended A-minor (type X) motifs: 104\n",
      "    total number of U-turns: 39\n",
      "    total number of ribose zippers: 50\n",
      "    total number of kink turns: 9\n",
      "    total number of phosphate interactions: 1490\n",
      "\n",
      "Time used: 00:00:00:47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 179/1074 [26:21<3:39:49, 14.74s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1wmq.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 288\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 180/1074 [26:22<2:37:39, 10.58s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5c45.cif1'\n",
      "    total number of nucleotides: 109\n",
      "    total number of base pairs: 50\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 4\n",
      "    total number of stems: 9\n",
      "    total number of non-pairing interactions: 143\n",
      "    total number of atom-base capping interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 19\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 20\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 181/1074 [26:23<1:51:16,  7.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1i6u.cif1'\n",
      "    total number of nucleotides: 74\n",
      "    total number of amino acids: 256\n",
      "    total number of base pairs: 40\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 76\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 182/1074 [26:23<1:19:07,  5.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hr7.cif1'\n",
      "    total number of nucleotides: 142\n",
      "    total number of amino acids: 722\n",
      "[w] arginine [1..B.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.103.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 64\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 162\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 25\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 43\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 183/1074 [26:24<59:26,  4.00s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xnr.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 75\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 184/1074 [26:24<43:15,  2.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6qic.cif1'\n",
      "    total number of nucleotides: 41\n",
      "    total number of amino acids: 2510\n",
      "[w] arginine [1..A.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.700.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1012.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.700.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1012.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.815.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.825.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.970.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1065.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.732.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1012.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.670.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1068.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1082.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1096.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 29\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 185/1074 [26:26<39:46,  2.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/217d.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 186/1074 [26:27<28:43,  1.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4c7o.cif1'\n",
      "    total number of nucleotides: 53\n",
      "    total number of amino acids: 1138\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 17%|█▋        | 187/1074 [26:27<24:24,  1.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1sds.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of amino acids: 336\n",
      "[w] Residue [1..B.GLU.55.] -- missing planar atom ' OE2'\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 188/1074 [26:28<19:23,  1.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5w1h.cif1'\n",
      "    total number of nucleotides: 41\n",
      "    total number of amino acids: 1301\n",
      "[w] arginine [1..A.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.736.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1047.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.736.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1047.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1257.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1324.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 30\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 189/1074 [26:31<24:52,  1.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qei.cif1'\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 562\n",
      "[w] arginine [1..A.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 77\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 190/1074 [26:31<20:35,  1.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2z2q.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 980\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 191/1074 [26:32<16:49,  1.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1urn.cif1'\n",
      "  1.P.U.20 0.135\n",
      "    total number of nucleotides: 56\n",
      "    total number of amino acids: 288\n",
      "[w] arginine [1..B.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.7.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 192/1074 [26:32<12:57,  1.13it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rtj.cif1'\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 520\n",
      "    total number of non-pairing interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 193/1074 [26:32<10:35,  1.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2y8w.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 215\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 194/1074 [26:33<08:43,  1.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jju.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 650\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.124.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 195/1074 [26:33<08:28,  1.73it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hjz.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 230\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 196/1074 [26:34<07:20,  1.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nvi.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 736\n",
      "[w] arginine [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.367.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.367.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.367.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 4\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 53\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 27\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 197/1074 [26:34<07:54,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3t5q.cif1'\n",
      "  1.L.U.2 0.121\n",
      "    total number of nucleotides: 43\n",
      "    total number of amino acids: 1807\n",
      "[w] arginine [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.323.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 17\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 40\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 18%|█▊        | 198/1074 [26:35<10:33,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/7msf.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▊        | 199/1074 [26:36<08:34,  1.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3s49.cif1'\r\n",
      "    total number of nucleotides: 56\r\n",
      "    total number of base pairs: 24\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 3\r\n",
      "    total number of non-pairing interactions: 59\r\n",
      "    total number of atom-base capping interactions: 5\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▊        | 200/1074 [26:36<06:38,  2.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3p4c.cif1'\r\n",
      "    total number of nucleotides: 16\r\n",
      "    total number of base pairs: 7\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 13\r\n",
      "    total number of non-loop single-stranded segments: 2\r\n",
      "    total number of phosphate interactions: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▊        | 201/1074 [26:36<05:11,  2.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xh7.cif1'\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 1283\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.790.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.794.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.843.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.863.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.891.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.790.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.794.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.843.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.863.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.891.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.909.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.951.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1226.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 39\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 88\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 47\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 202/1074 [26:38<12:14,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5l00.cif1'\n",
      "    total number of nucleotides: 64\n",
      "    total number of base pairs: 32\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 67\n",
      "    total number of atom-base capping interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 203/1074 [26:38<09:44,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bnn.cif1'\n",
      "    total number of nucleotides: 84\n",
      "    total number of base pairs: 34\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of internal loops: 8\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 204/1074 [26:39<08:07,  1.78it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1il2.cif1'\n",
      "  1.C.H2U.916 0.143\n",
      "  1.C.H2U.919 0.152\n",
      "  1.D.H2U.1916 0.152\n",
      "  1.D.H2U.1919 0.134\n",
      "1..C.PSU.913. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..C.PSU.932. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..C.PSU.955. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..D.PSU.1913. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..D.PSU.1932. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..D.PSU.1955. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 147\n",
      "    total number of amino acids: 1172\n",
      "[w] arginine [1..A.ARG.537.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.1537.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.537.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1537.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 174\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 205/1074 [26:40<11:35,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3hhn.cif1'\n",
      "    total number of nucleotides: 274\n",
      "    total number of amino acids: 185\n",
      "    total number of base pairs: 111\n",
      "    total number of multiplets: 14\n",
      "    total number of helices: 6\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 307\n",
      "    total number of atom-base capping interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 44\n",
      "                        consolidated into units: 30\n",
      "    total number of hairpin loops: 10\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of A-minor (types I and II) motifs: 8\n",
      "    total number of eXtended A-minor (type X) motifs: 7\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 49\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 206/1074 [26:41<11:31,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wwg.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 184\n",
      "[w] arginine [1..A.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 207/1074 [26:41<09:01,  1.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k50.cif1'\n",
      "    total number of nucleotides: 182\n",
      "    total number of amino acids: 1840\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..M.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.174.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 82\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of non-pairing interactions: 184\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of hairpin loops: 4\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 52\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 208/1074 [26:42<12:53,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3am1.cif1'\n",
      "    total number of nucleotides: 82\n",
      "    total number of amino acids: 236\n",
      "[w] arginine [1..A.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.240.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 94\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 19%|█▉        | 209/1074 [26:43<10:47,  1.33it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4h5p.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 976\n",
      "[w] arginine [1..B.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.114.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|█▉        | 210/1074 [26:44<10:56,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2o5i.cif1'\n",
      "    total number of nucleotides: 104\n",
      "    total number of amino acids: 5950\n",
      "    total number of base pairs: 44\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 100\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 31\n",
      "\n",
      "Time used: 00:00:00:09\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|█▉        | 211/1074 [26:53<48:47,  3.39s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4tz0.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 508\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|█▉        | 212/1074 [26:54<36:55,  2.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4p43.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|█▉        | 213/1074 [26:54<26:51,  1.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5mga.cif1'\n",
      "    total number of nucleotides: 78\n",
      "    total number of amino acids: 1182\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 77\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|█▉        | 214/1074 [26:56<26:10,  1.83s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/332d.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|██        | 215/1074 [26:56<19:05,  1.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2fz2.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 542\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|██        | 216/1074 [26:56<14:45,  1.03s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gv4.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 918\n",
      "[w] arginine [1..A.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.173.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.220.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.238.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.109.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|██        | 217/1074 [26:57<13:35,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3aev.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 345\n",
      "[w] arginine [1..A.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.203.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|██        | 218/1074 [26:58<12:13,  1.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4csf.cif1'\n",
      "    total number of nucleotides: 108\n",
      "    total number of amino acids: 5977\n",
      "    total number of base pairs: 11\n",
      "    total number of non-pairing interactions: 62\n",
      "    total number of splayed-apart dinucleotides: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 12\n",
      "    total number of phosphate interactions: 40\n",
      "\n",
      "Time used: 00:00:00:05\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|██        | 219/1074 [27:03<33:08,  2.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lys.cif1'\n",
      "    total number of nucleotides: 114\n",
      "    total number of base pairs: 60\n",
      "    total number of multiplets: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 10\n",
      "    total number of non-pairing interactions: 131\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 6\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 20%|██        | 220/1074 [27:04<26:59,  1.90s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kd5.cif1'\r\n",
      "    total number of nucleotides: 22\r\n",
      "    total number of base pairs: 10\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 2\r\n",
      "    total number of non-pairing interactions: 22\r\n",
      "    total number of internal loops: 1\r\n",
      "    total number of non-loop single-stranded segments: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 221/1074 [27:04<19:25,  1.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5d5l.cif1'\n",
      "    total number of nucleotides: 312\n",
      "[w] Residue 1..A.PRF.117. missing planar atom [ N7 ]\n",
      "\t[i] 1..A.PRF.117.: replace missing base ring atom [ N7 ] with [C7] (d=0.07)\n",
      "[w] Residue 1..B.PRF.109. missing planar atom [ N7 ]\n",
      "\t[i] 1..B.PRF.109.: replace missing base ring atom [ N7 ] with [C7] (d=0.08)\n",
      "[w] Residue 1..C.PRF.104. missing planar atom [ N7 ]\n",
      "\t[i] 1..C.PRF.104.: replace missing base ring atom [ N7 ] with [C7] (d=0.07)\n",
      "[w] Residue 1..D.PRF.106. missing planar atom [ N7 ]\n",
      "\t[i] 1..D.PRF.106.: replace missing base ring atom [ N7 ] with [C7] (d=0.07)\n",
      "    total number of base pairs: 145\n",
      "    total number of multiplets: 15\n",
      "    total number of helices: 8\n",
      "    total number of stems: 16\n",
      "    total number of non-pairing interactions: 361\n",
      "    total number of atom-base capping interactions: 20\n",
      "    total number of splayed-apart dinucleotides: 26\n",
      "                        consolidated into units: 16\n",
      "    total number of hairpin loops: 12\n",
      "    total number of non-loop single-stranded segments: 16\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 51\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 222/1074 [27:05<16:01,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r2c.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 434\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 223/1074 [27:05<12:35,  1.13it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pco.cif1'\n",
      "    total number of nucleotides: 50\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 224/1074 [27:06<09:44,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5vci.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of base pairs: 16\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 225/1074 [27:06<07:36,  1.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jch.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 1316\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 47\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 226/1074 [27:07<10:31,  1.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rdx.cif1'\n",
      "    total number of nucleotides: 78\n",
      "    total number of amino acids: 410\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 89\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 227/1074 [27:08<11:15,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3npq.cif1'\n",
      "    total number of nucleotides: 153\n",
      "    total number of amino acids: 3\n",
      "    total number of base pairs: 74\n",
      "    total number of multiplets: 14\n",
      "    total number of helices: 6\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 175\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 24\n",
      "    total number of eXtended A-minor (type X) motifs: 8\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██        | 228/1074 [27:09<13:12,  1.07it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1y26.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of base pairs: 41\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 85\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██▏       | 229/1074 [27:10<11:31,  1.22it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4bw0.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of amino acids: 117\n",
      "    total number of base pairs: 11\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 21%|██▏       | 230/1074 [27:10<09:59,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4tu0.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 643\n",
      "[w] arginine [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.28.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 231/1074 [27:11<09:45,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ohy.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 398\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 232/1074 [27:11<08:16,  1.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3egz.cif1'\n",
      "    total number of nucleotides: 65\n",
      "    total number of amino acids: 91\n",
      "    total number of base pairs: 29\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 67\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 233/1074 [27:12<10:34,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cgs.cif1'\n",
      "1..A.PSU.6. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 25\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 234/1074 [27:13<08:18,  1.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1utd.cif1'\n",
      "    total number of nucleotides: 35\n",
      "    total number of amino acids: 1570\n",
      "[w] arginine [1..N.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.66.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 11\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 235/1074 [27:14<11:06,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3a6p.cif1'\n",
      "    total number of nucleotides: 94\n",
      "    total number of amino acids: 2510\n",
      "    total number of base pairs: 44\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 96\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 236/1074 [27:16<19:08,  1.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ndh.cif1'\n",
      "  1.A.CBV.2 0.122\n",
      "    total number of nucleotides: 64\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 71\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of phosphate interactions: 13\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 237/1074 [27:17<15:47,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1xok.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 33\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 5\n",
      "    total number of bulges: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 238/1074 [27:17<12:09,  1.15it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jrt.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 239/1074 [27:18<09:30,  1.46it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6id0.cif1'\n",
      "    total number of nucleotides: 406\n",
      "    total number of amino acids: 10389\n",
      "[w] arginine [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.510.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.663.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.758.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.510.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.663.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.758.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.762.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.845.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.855.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.861.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.995.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1201.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1224.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1341.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1370.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1402.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1414.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1439.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1471.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1544.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1681.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1744.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1746.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1865.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.158.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.569.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.730.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.550.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.556.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.585.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.649.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.704.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.862.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 109\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 10\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 8\n",
      "    total number of non-pairing interactions: 354\n",
      "    total number of atom-base capping interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 61\n",
      "                        consolidated into units: 37\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 11\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 12\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 139\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 240/1074 [27:32<1:08:35,  4.93s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5eim.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 318\n",
      "[w] arginine [1..A.ARG.331.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.331.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 22%|██▏       | 241/1074 [27:33<50:01,  3.60s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5i9h.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 368\n",
      "[w] arginine [1..A.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.406.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.406.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 242/1074 [27:33<36:22,  2.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1l2x.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 15\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 27\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 243/1074 [27:34<27:22,  1.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3loa.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 46\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 244/1074 [27:34<20:24,  1.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5mwi.cif1'\r\n",
      "    total number of nucleotides: 26\r\n",
      "    total number of non-pairing interactions: 20\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 245/1074 [27:34<14:56,  1.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wwf.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 354\n",
      "[w] arginine [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.147.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 246/1074 [27:35<12:28,  1.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gc5.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 311\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 247/1074 [27:35<10:08,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1rna.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 32\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 248/1074 [27:36<09:31,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cfj.cif1'\n",
      "  1.1w.U.47 0.120\n",
      "  1.1y.U.33 0.122\n",
      "1..1A.PSU.1933. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.1939. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.2617. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1w.PSU.32. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1w.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1w.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1x.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1y.PSU.32. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1y.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1y.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2w.PSU.32. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2w.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2w.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2x.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2y.PSU.32. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2y.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2y.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9384\n",
      "    total number of amino acids: 11576\n",
      "[w] arginine [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1W.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..10.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..12.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..13.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..14.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1b.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1n.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1p.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Q.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..20.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..22.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..23.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2g.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2h.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2h.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2l.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2n.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2s.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2t.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4759\n",
      "    total number of multiplets: 722\n",
      "    total number of helices: 271\n",
      "    total number of stems: 593\n",
      "    total number of isolated WC/wobble pairs: 174\n",
      "    total number of non-pairing interactions: 13384\n",
      "    total number of atom-base capping interactions: 631\n",
      "    total number of splayed-apart dinucleotides: 1210\n",
      "                        consolidated into units: 754\n",
      "    total number of hairpin loops: 219\n",
      "    total number of bulges: 128\n",
      "    total number of internal loops: 217\n",
      "    total number of junctions: 117\n",
      "    total number of non-loop single-stranded segments: 85\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cfj.cif1', hairpin loops (#1/#3) have 2 interactions [#10, #11 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cfj.cif1', hairpin loops (#112/#114) have 2 interactions [#309, #310 ...]\n",
      "    total number of kissing loops: 16\n",
      "    total number of A-minor (types I and II) motifs: 274\n",
      "    total number of eXtended A-minor (type X) motifs: 202\n",
      "    total number of U-turns: 106\n",
      "    total number of ribose zippers: 124\n",
      "    total number of kink turns: 25\n",
      "    total number of phosphate interactions: 3402\n",
      "[w]  number-of-residues=399 -- in shortened form\n",
      "[w]  number-of-residues=399 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:02:44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 249/1074 [30:20<11:24:30, 49.78s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6ra4.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 257\n",
      "[w] arginine [1..A.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.247.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 250/1074 [30:20<7:59:51, 34.94s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nh5.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 271\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 251/1074 [30:21<5:36:49, 24.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wzq.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 23%|██▎       | 252/1074 [30:21<3:57:16, 17.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1f7u.cif1'\n",
      "  1.B.H2U.916 0.155\n",
      "  1.B.H2U.919 0.154\n",
      "  1.B.H2U.947 0.150\n",
      "1..B.PSU.901. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..B.PSU.927. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..B.PSU.955. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 76\n",
      "    total number of amino acids: 607\n",
      "    total number of base pairs: 30\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 82\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 23\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▎       | 253/1074 [30:22<2:51:29, 12.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1u1y.cif1'\n",
      "    total number of nucleotides: 33\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 32\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▎       | 254/1074 [30:23<2:01:57,  8.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dea.cif1'\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 28\n",
      "    total number of base pairs: 40\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 5\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of G-tetrads: 6\n",
      "    total number of G4-helices: 2\n",
      "    total number of G4-stems: 2\n",
      "    total number of phosphate interactions: 25\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▎       | 255/1074 [30:23<1:26:33,  6.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3p4d.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 16\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 256/1074 [30:24<1:01:54,  4.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pqu.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of amino acids: 1915\n",
      "[w] arginine [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 38\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 78\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 257/1074 [30:25<50:00,  3.67s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nmu.cif1'\n",
      "    total number of nucleotides: 95\n",
      "    total number of amino acids: 1430\n",
      "[w] arginine [1..A.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.367.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.367.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.225.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 39\n",
      "    total number of helices: 6\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 5\n",
      "    total number of non-pairing interactions: 100\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 8\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 26\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 258/1074 [30:26<39:35,  2.91s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4y1m.cif1'\n",
      "    total number of nucleotides: 185\n",
      "    total number of base pairs: 83\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 218\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 5\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 259/1074 [30:27<29:25,  2.17s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2zi0.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 115\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 260/1074 [30:27<22:18,  1.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1m8x.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 682\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 261/1074 [30:28<17:51,  1.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5z4j.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 349\n",
      "[w] arginine [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.456.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.456.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 262/1074 [30:28<14:01,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qu7.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 316\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 24%|██▍       | 263/1074 [30:28<10:49,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r1c.cif1'\n",
      "    total number of nucleotides: 288\n",
      "    total number of base pairs: 144\n",
      "    total number of helices: 10\n",
      "    total number of stems: 54\n",
      "    total number of non-pairing interactions: 305\n",
      "    total number of atom-base capping interactions: 16\n",
      "    total number of internal loops: 36\n",
      "    total number of phosphate interactions: 48\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▍       | 264/1074 [30:29<09:10,  1.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1s03.cif1'\n",
      "    total number of nucleotides: 94\n",
      "    total number of amino acids: 253\n",
      "    total number of base pairs: 45\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 4\n",
      "    total number of stems: 10\n",
      "    total number of non-pairing interactions: 106\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 8\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▍       | 265/1074 [30:30<10:22,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3oin.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 435\n",
      "[w]  with aas out of order: 1..B.SCY.254.[451] vs 1..B.ASP.71.[252]\n",
      "[w] arginine [1..A.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.211.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 18\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▍       | 266/1074 [30:30<08:48,  1.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dm6.cif1'\n",
      "  1.X.U.943 0.122\n",
      "  1.X.U.1231 0.121\n",
      "  1.X.U.1630 0.125\n",
      "  1.X.U.1951 0.128\n",
      "  1.X.U.1995 0.122\n",
      "  1.X.U.2348 0.121\n",
      "  1.X.U.2419 0.129\n",
      "  1.X.U.2500 0.123\n",
      "  1.X.U.2593 0.125\n",
      "    total number of nucleotides: 2902\n",
      "    total number of amino acids: 3487\n",
      "[w] arginine [1..A.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.222.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.199.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.222.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.199.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.7.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1412\n",
      "    total number of multiplets: 201\n",
      "    total number of helices: 82\n",
      "    total number of stems: 182\n",
      "    total number of isolated WC/wobble pairs: 64\n",
      "    total number of non-pairing interactions: 4093\n",
      "    total number of atom-base capping interactions: 183\n",
      "    total number of splayed-apart dinucleotides: 415\n",
      "                        consolidated into units: 261\n",
      "    total number of hairpin loops: 74\n",
      "[w]  sheared G-A pair [3]: symmetric internal loop: nts=12; [4,4]; linked by [#166,#167]\n",
      "    total number of bulges: 38\n",
      "    total number of internal loops: 78\n",
      "    total number of junctions: 37\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dm6.cif1', hairpin loops (#1/#3) have 2 interactions [#-2, #10 ...]\n",
      "    total number of kissing loops: 7\n",
      "    total number of A-minor (types I and II) motifs: 84\n",
      "    total number of eXtended A-minor (type X) motifs: 57\n",
      "    total number of U-turns: 27\n",
      "    total number of ribose zippers: 42\n",
      "    total number of kink turns: 5\n",
      "    total number of phosphate interactions: 1005\n",
      "\n",
      "Time used: 00:00:00:35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▍       | 267/1074 [31:05<2:28:51, 11.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4lgt.cif1'\n",
      "  1.E.FHU.2605 0.213\n",
      "  1.F.FHU.2605 0.229\n",
      "1..E.FHU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..F.FHU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 41\n",
      "    total number of amino acids: 498\n",
      "[w] arginine [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.18.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▍       | 268/1074 [31:07<1:49:14,  8.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xsl.cif1'\n",
      "  1.B.U.67 0.133\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▌       | 269/1074 [31:07<1:17:02,  5.74s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zse.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 1549\n",
      "[w] arginine [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.476.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.717.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.476.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.717.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▌       | 270/1074 [31:09<1:00:53,  4.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xd0.cif1'\n",
      "  *1.G.A23.32 0.246 (planarity=0.166)\n",
      "  1.G.A23.32 0.246\n",
      "[i] taken as a pyrimidine ... 1.G.A23.32\n",
      "  1.H.U.18 0.127\n",
      "  *1.H.A23.32 0.279 (planarity=0.120)\n",
      "  1.H.A23.32 0.279\n",
      "  1.I.A23.32 0.231\n",
      "[i] taken as a pyrimidine ... 1.I.A23.32\n",
      "  1.U.A23.32 0.217\n",
      "[i] taken as a pyrimidine ... 1.U.A23.32\n",
      "  1.V.A23.32 0.218\n",
      "[i] taken as a pyrimidine ... 1.V.A23.32\n",
      "  1.W.A23.32 0.234\n",
      "[i] taken as a pyrimidine ... 1.W.A23.32\n",
      "1..G.A23.32. has unusual GLYCOSIDIC bond!\n",
      "1..I.A23.32. has unusual GLYCOSIDIC bond!\n",
      "1..U.A23.32. has unusual GLYCOSIDIC bond!\n",
      "1..V.A23.32. has unusual GLYCOSIDIC bond!\n",
      "1..W.A23.32. has unusual GLYCOSIDIC bond!\n",
      "    total number of nucleotides: 216\n",
      "    total number of amino acids: 972\n",
      "[w] arginine [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Y.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.122.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 74\n",
      "    total number of multiplets: 20\n",
      "    total number of helices: 6\n",
      "    total number of stems: 11\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 201\n",
      "    total number of atom-base capping interactions: 22\n",
      "    total number of splayed-apart dinucleotides: 36\n",
      "                        consolidated into units: 21\n",
      "    total number of hairpin loops: 12\n",
      "    total number of non-loop single-stranded segments: 12\n",
      "    total number of A-minor (types I and II) motifs: 9\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 69\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▌       | 271/1074 [31:10<47:39,  3.56s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6e8s.cif1'\n",
      "    total number of nucleotides: 76\n",
      "    total number of base pairs: 49\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 5\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 83\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of G-tetrads: 4\n",
      "    total number of G4-helices: 2\n",
      "    total number of G4-stems: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▌       | 272/1074 [31:10<34:05,  2.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qil.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 591\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.233.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.343.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.233.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.343.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 25%|██▌       | 273/1074 [31:11<25:36,  1.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jji.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 2437\n",
      "[w] arginine [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.269.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.366.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.269.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.366.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.366.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.366.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.296.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.366.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.212.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.269.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.366.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.366.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 274/1074 [31:12<24:25,  1.83s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5w6v.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 813\n",
      "[w] arginine [1..A.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.633.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.633.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 275/1074 [31:13<20:16,  1.52s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3kna.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 276/1074 [31:13<16:03,  1.21s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4mcf.cif1'\n",
      "    total number of nucleotides: 78\n",
      "    total number of base pairs: 39\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 277/1074 [31:14<12:05,  1.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ev1.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 198\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 278/1074 [31:14<09:25,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gx6.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 159\n",
      "[w] arginine [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 279/1074 [31:14<07:37,  1.74it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xog.cif1'\n",
      "    total number of nucleotides: 65\n",
      "    total number of amino acids: 4067\n",
      "[w] arginine [1..B.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.173.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..H.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..K.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..W.ARG.777.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.173.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.777.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 32\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 68\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 18\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 280/1074 [31:19<22:39,  1.71s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rzd.cif1'\n",
      "    total number of nucleotides: 100\n",
      "[w] Residue 1..A.PRF.201. missing planar atom [ N7 ]\n",
      "\t[i] 1..A.PRF.201.: replace missing base ring atom [ N7 ] with [C7] (d=0.08)\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 113\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▌       | 281/1074 [31:19<16:51,  1.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4krf.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 817\n",
      "[w] arginine [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.633.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.633.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▋       | 282/1074 [31:20<16:26,  1.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6m7k.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 318\n",
      "[w] arginine [1..A.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.175.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▋       | 283/1074 [31:21<13:42,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5w0o.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 655\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 27\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 26%|██▋       | 284/1074 [31:21<11:45,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/353d.cif1'\n",
      "  1.A.C.12 0.134\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of internal loops: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 285/1074 [31:21<09:21,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3pey.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 391\n",
      "[w] arginine [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.380.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.426.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.432.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.380.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.426.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.432.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 286/1074 [31:22<10:03,  1.31it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ba2.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 721\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 287/1074 [31:23<10:37,  1.23it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4al5.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 189\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 288/1074 [31:24<09:45,  1.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zw4.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 217\n",
      "[w] arginine [1..A.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.179.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 289/1074 [31:24<07:49,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4peh.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 1745\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 290/1074 [31:26<11:24,  1.15it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6bgb.cif1'\n",
      "    total number of nucleotides: 48\n",
      "[w] Residue 1..A.A7C.5. missing planar atom [ N7 ]\n",
      "[w] Residue 1..A.A7C.5. missing planar atom [ C8 ]\n",
      "\t[i] 1..A.A7C.5.: replace missing base ring atom [ N7 ] with [C7] (d=0.19)\n",
      "\t[i] 1..A.A7C.5.: replace missing base ring atom [ C8 ] with [N8] (d=0.24)\n",
      "[w] Residue 1..B.A7C.5. missing planar atom [ N7 ]\n",
      "[w] Residue 1..B.A7C.5. missing planar atom [ C8 ]\n",
      "\t[i] 1..B.A7C.5.: replace missing base ring atom [ N7 ] with [C7] (d=0.19)\n",
      "\t[i] 1..B.A7C.5.: replace missing base ring atom [ C8 ] with [N8] (d=0.24)\n",
      "[w] Residue 1..C.A7C.5. missing planar atom [ N7 ]\n",
      "[w] Residue 1..C.A7C.5. missing planar atom [ C8 ]\n",
      "\t[i] 1..C.A7C.5.: replace missing base ring atom [ N7 ] with [C7] (d=0.19)\n",
      "\t[i] 1..C.A7C.5.: replace missing base ring atom [ C8 ] with [N8] (d=0.24)\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 52\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 291/1074 [31:26<09:23,  1.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4kyy.cif1'\r\n",
      "    total number of nucleotides: 34\r\n",
      "    total number of base pairs: 17\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 4\r\n",
      "    total number of non-pairing interactions: 36\r\n",
      "    total number of internal loops: 3\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 292/1074 [31:26<07:09,  1.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2r1s.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 293/1074 [31:26<05:42,  2.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3v71.cif1'\n",
      "  *1.B.EPE.13 0.241 (planarity=0.264)\n",
      "  1.B.EPE.13 0.241\n",
      "[i] taken as a pyrimidine ... 1.B.EPE.13\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 364\n",
      "[w] Residue 1..B.EPE.13. missing planar atom [ C4 ]\n",
      "[w] Residue 1..B.EPE.13. missing planar atom [ N3 ]\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 294/1074 [31:27<05:27,  2.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zsc.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 1549\n",
      "[w] arginine [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.636.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.636.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.636.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 27%|██▋       | 295/1074 [31:28<10:19,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f6c.cif1'\n",
      "  1.C.U.2 0.120\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 1022\n",
      "[w] arginine [1..A.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.368.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.371.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.456.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.470.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.368.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.371.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.456.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.470.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.371.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.406.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.456.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.470.] -- NH1/NH2 naming swapped\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 296/1074 [31:29<10:50,  1.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1rlg.cif1'\n",
      "    total number of nucleotides: 50\n",
      "    total number of amino acids: 230\n",
      "[w] arginine [1..B.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.4.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 67\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 297/1074 [31:30<09:13,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rmo.cif1'\n",
      "  1.L.A23.45 0.129\n",
      "    total number of nucleotides: 360\n",
      "    total number of amino acids: 1239\n",
      "    total number of base pairs: 137\n",
      "    total number of multiplets: 32\n",
      "    total number of helices: 8\n",
      "    total number of stems: 16\n",
      "    total number of non-pairing interactions: 473\n",
      "    total number of atom-base capping interactions: 17\n",
      "    total number of splayed-apart dinucleotides: 68\n",
      "                        consolidated into units: 42\n",
      "    total number of hairpin loops: 16\n",
      "    total number of non-loop single-stranded segments: 16\n",
      "    total number of eXtended A-minor (type X) motifs: 43\n",
      "    total number of ribose zippers: 8\n",
      "    total number of phosphate interactions: 103\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 298/1074 [31:31<13:01,  1.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4kq0.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 244\n",
      "[w] arginine [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.53.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of internal loops: 5\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 299/1074 [31:32<10:04,  1.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1xjr.cif1'\n",
      "    total number of nucleotides: 47\n",
      "    total number of base pairs: 22\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 60\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 300/1074 [31:32<07:51,  1.64it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mij.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of base pairs: 4\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 301/1074 [31:32<06:14,  2.06it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2izn.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 302/1074 [31:32<05:39,  2.27it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5cki.cif1'\n",
      "    total number of nucleotides: 59\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 60\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 303/1074 [31:34<08:50,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nma.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 304/1074 [31:34<08:30,  1.51it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v88.cif1'\n",
      "  1.A2.U.319 0.124\n",
      "  1.A2.U.380 0.123\n",
      "  1.A2.U.618 0.120\n",
      "  1.A2.U.719 0.122\n",
      "  1.A2.U.749 0.121\n",
      "  1.A2.U.831 0.126\n",
      "  1.A2.U.912 0.120\n",
      "  1.A2.U.1012 0.126\n",
      "  1.A2.U.1057 0.127\n",
      "    total number of nucleotides: 10433\n",
      "    total number of amino acids: 22904\n",
      "[w] arginine [1..AA.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AB.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AB.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AB.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AB.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AD.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AD.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AA.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AB.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AB.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AB.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AB.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AH.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AH.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AV.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AV.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AZ.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AZ.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aa.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aa.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ah.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.326.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BF.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BG.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BH.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BH.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BH.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BJ.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BJ.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BJ.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BM.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BM.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BQ.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BQ.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BQ.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BR.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BR.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BR.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BS.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BS.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BS.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BS.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BT.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BT.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BT.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BV.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BX.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BX.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BY.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BZ.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BZ.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bd.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bf.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bh.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bj.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bn.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bn.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bo.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Bp.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CA.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CB.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CB.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CB.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CB.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CH.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CH.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CL.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CU.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CX.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CX.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ca.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ca.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ca.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Cc.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ce.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Cf.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ch.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ch.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ch.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DA.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DA.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DA.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DB.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.326.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.158.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DH.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DH.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DH.ARG.166.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DH.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DI.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DI.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DJ.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DJ.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DJ.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DJ.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DV.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DW.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DX.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DX.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DZ.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DZ.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dd.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dd.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dh.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Di.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dj.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dn.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Do.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dp.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Dq.ARG.61.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 4865\n",
      "    total number of multiplets: 652\n",
      "    total number of helices: 302\n",
      "    total number of stems: 645\n",
      "    total number of isolated WC/wobble pairs: 249\n",
      "    total number of non-pairing interactions: 13600\n",
      "    total number of atom-base capping interactions: 732\n",
      "    total number of splayed-apart dinucleotides: 1637\n",
      "                        consolidated into units: 990\n",
      "    total number of hairpin loops: 232\n",
      "    total number of bulges: 150\n",
      "    total number of internal loops: 322\n",
      "    total number of junctions: 128\n",
      "    total number of non-loop single-stranded segments: 85\n",
      "    total number of kissing loops: 10\n",
      "    total number of A-minor (types I and II) motifs: 216\n",
      "    total number of eXtended A-minor (type X) motifs: 212\n",
      "    total number of U-turns: 102\n",
      "    total number of ribose zippers: 107\n",
      "    total number of kink turns: 19\n",
      "    total number of phosphate interactions: 5847\n",
      "[w]  number-of-residues=569 -- in shortened form\n",
      "[w]  number-of-residues=569 -- in shortened form\n",
      "\n",
      "Time used: 00:00:05:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 305/1074 [36:34<19:19:44, 90.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nd3.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of non-pairing interactions: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 28%|██▊       | 306/1074 [36:35<13:31:36, 63.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6jim.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 909\n",
      "[w] arginine [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.361.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.361.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▊       | 307/1074 [36:35<9:30:17, 44.61s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3e2e.cif1'\n",
      "    total number of nucleotides: 66\n",
      "    total number of amino acids: 832\n",
      "    total number of base pairs: 29\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 63\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▊       | 308/1074 [36:36<6:42:32, 31.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2bh2.cif1'\n",
      "  1.C.FMU.1939 0.122\n",
      "  1.D.FMU.1939 0.140\n",
      "    total number of nucleotides: 62\n",
      "    total number of amino acids: 836\n",
      "[w] arginine [1..A.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.68.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 52\n",
      "    total number of splayed-apart dinucleotides: 19\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 309/1074 [36:37<4:44:16, 22.30s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qcu.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 7\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 16\n",
      "    total number of non-pairing interactions: 109\n",
      "    total number of phosphate interactions: 6\n",
      "[w]  with 11 overlapped base pairs\n",
      "[w]  with 4 helix/stem mis-assignments\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 310/1074 [36:37<3:19:33, 15.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/205d.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 21\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 311/1074 [36:37<2:20:10, 11.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2uwm.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of amino acids: 396\n",
      "[w] arginine [1..A.ARG.472.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.476.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.530.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.439.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.472.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.476.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.530.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.472.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.476.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.530.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.439.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.472.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.476.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.530.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 50\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 312/1074 [36:38<1:39:24,  7.83s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4p3t.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 313/1074 [36:38<1:10:16,  5.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5btp.cif1'\n",
      "    total number of nucleotides: 124\n",
      "    total number of base pairs: 51\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 171\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 314/1074 [36:38<50:00,  3.95s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6f3h.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 1552\n",
      "[w] arginine [1..A.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.520.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.782.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.806.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.520.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.782.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.806.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.489.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.546.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.554.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.583.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 315/1074 [36:40<40:21,  3.19s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f8n.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 462\n",
      "[w] arginine [1..A.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.346.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.416.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.438.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.346.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.416.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.438.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 29%|██▉       | 316/1074 [36:40<30:01,  2.38s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rc8.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 610\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|██▉       | 317/1074 [36:41<23:18,  1.85s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5tsn.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 500\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|██▉       | 318/1074 [36:41<18:10,  1.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dcc.cif1'\n",
      "    total number of nucleotides: 37\n",
      "    total number of amino acids: 222\n",
      "[w] arginine [1..A.ARG.469.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.608.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.469.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.608.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|██▉       | 319/1074 [36:42<14:20,  1.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4gg4.cif1'\n",
      "  1.G.DT.-2 0.122\n",
      "    total number of nucleotides: 33\n",
      "    total number of amino acids: 488\n",
      "[w] arginine [1..A.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.584.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.584.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 32\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|██▉       | 320/1074 [36:42<12:18,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5gxh.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 672\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.684.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.684.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|██▉       | 321/1074 [36:43<10:59,  1.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zc9.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 378\n",
      "[w] arginine [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.283.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.353.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.362.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.365.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.283.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.353.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.362.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.365.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|██▉       | 322/1074 [36:43<09:10,  1.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bsu.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 290\n",
      "[w] arginine [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.72.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 51\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|███       | 323/1074 [36:44<07:25,  1.69it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rum.cif1'\n",
      "  1.A.U.70 0.121\n",
      "    total number of nucleotides: 92\n",
      "    total number of base pairs: 39\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 98\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|███       | 324/1074 [36:44<06:34,  1.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2bgg.cif1'\n",
      "    total number of nucleotides: 29\n",
      "    total number of amino acids: 789\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 26\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|███       | 325/1074 [36:45<06:53,  1.81it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1nuv.cif1'\n",
      "    total number of nucleotides: 96\n",
      "    total number of base pairs: 41\n",
      "    total number of helices: 5\n",
      "    total number of stems: 8\n",
      "    total number of non-pairing interactions: 94\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|███       | 326/1074 [36:45<05:42,  2.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wsb.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 2178\n",
      "[w] arginine [1..A.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.610.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.706.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.723.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.610.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.706.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.723.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.427.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.737.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 30%|███       | 327/1074 [36:47<11:52,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qqb.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 482\n",
      "[w] arginine [1..B.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.139.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 328/1074 [36:47<10:02,  1.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3hjw.cif1'\n",
      "  1.E.FHU.10 0.162\n",
      "1..E.FHU.10. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 71\n",
      "    total number of amino acids: 500\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 71\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 24\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 329/1074 [36:48<10:25,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pr6.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of amino acids: 95\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 330/1074 [36:49<09:52,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2asb.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 226\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 331/1074 [36:49<07:37,  1.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ed1.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of amino acids: 793\n",
      "[w] arginine [1..A.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.494.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..B.8AZ.12. missing planar atom [ C8 ]\n",
      "\t[i] 1..B.8AZ.12.: replace missing base ring atom [ C8 ] with [N8] (d=0.11)\n",
      "[w] Residue [1..A.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.494.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.674.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 45\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 332/1074 [36:50<07:23,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1saq.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 333/1074 [36:50<06:02,  2.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2jlt.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of phosphate interactions: 2\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 334/1074 [36:50<04:55,  2.51it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2x1f.cif1'\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 94\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███       | 335/1074 [36:50<04:06,  3.00it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mqk.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 455\n",
      "    total number of base pairs: 8\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███▏      | 336/1074 [36:51<04:07,  2.98it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ktj.cif1'\n",
      "    total number of nucleotides: 130\n",
      "    total number of base pairs: 73\n",
      "    total number of multiplets: 14\n",
      "    total number of helices: 3\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 154\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 12\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 31\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███▏      | 337/1074 [36:53<10:04,  1.22it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4xww.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 1088\n",
      "[w] arginine [1..A.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.375.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.501.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.511.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.421.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.375.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.501.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.511.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.421.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.511.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 31%|███▏      | 338/1074 [36:53<10:07,  1.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6joo.cif1'\n",
      "    total number of nucleotides: 131\n",
      "    total number of amino acids: 834\n",
      "[w] arginine [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.775.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.960.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1003.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.775.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.960.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1003.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1004.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1006.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1062.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 56\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 130\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 45\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 339/1074 [36:55<10:50,  1.13it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ebi.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 340/1074 [36:55<08:17,  1.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5js1.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 801\n",
      "[w] arginine [1..A.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 341/1074 [36:56<08:58,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3qjl.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 480\n",
      "[w] arginine [1..B.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.145.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 342/1074 [36:56<08:13,  1.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5z9x.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 288\n",
      "[w] arginine [1..A.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 343/1074 [36:56<06:45,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ghl.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 498\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 344/1074 [36:57<05:50,  2.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mja.cif1'\n",
      "    total number of nucleotides: 43\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 5\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 345/1074 [36:57<04:47,  2.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6aso.cif1'\n",
      "  1.I.U.89 0.121\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 906\n",
      "[w] arginine [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.37.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 23\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 59\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 346/1074 [36:58<05:47,  2.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1m5k.cif1'\n",
      "    total number of nucleotides: 226\n",
      "    total number of amino acids: 184\n",
      "    total number of base pairs: 106\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 4\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 5\n",
      "    total number of non-pairing interactions: 265\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 22\n",
      "                        consolidated into units: 16\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 7\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 39\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 347/1074 [36:59<08:28,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1dqf.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of bulges: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 348/1074 [36:59<06:51,  1.76it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zx2.cif1'\n",
      "    total number of nucleotides: 102\n",
      "    total number of amino acids: 3097\n",
      "[w] arginine [1..A.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.205.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.205.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.205.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.205.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.390.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.517.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.562.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.584.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.613.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.667.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.668.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.673.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.797.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.851.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.855.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.925.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1044.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1067.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.282.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.478.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.703.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.741.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.834.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.841.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.940.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1038.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1097.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.188.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 42\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 86\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 31\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 32%|███▏      | 349/1074 [37:03<18:11,  1.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5tdj.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of non-pairing interactions: 13\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 350/1074 [37:03<13:23,  1.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ddp.cif1'\n",
      "    total number of nucleotides: 122\n",
      "    total number of amino acids: 189\n",
      "[w] arginine [1..C.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.35.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 56\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 152\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 4\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 34\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 351/1074 [37:03<11:01,  1.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k5q.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 400\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 352/1074 [37:04<09:35,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qi2.cif1'\n",
      "    total number of nucleotides: 74\n",
      "    total number of amino acids: 612\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.297.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 27\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 73\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 4\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of phosphate interactions: 37\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 353/1074 [37:04<07:54,  1.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bx2.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 651\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 354/1074 [37:05<07:27,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1evv.cif1'\n",
      "  1.A.H2U.16 0.171\n",
      "  1.A.H2U.17 0.179\n",
      "1..A.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..A.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 76\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 93\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 355/1074 [37:05<06:00,  1.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3gm7.cif1'\n",
      "  1.G.U.5 0.134\n",
      "    total number of nucleotides: 36\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of internal loops: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 356/1074 [37:05<04:55,  2.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dn2.cif1'\n",
      "    total number of nucleotides: 109\n",
      "    total number of base pairs: 52\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 5\n",
      "    total number of stems: 9\n",
      "    total number of non-pairing interactions: 143\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 16\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 357/1074 [37:06<05:02,  2.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3hk2.cif1'\n",
      "    total number of nucleotides: 62\n",
      "    total number of amino acids: 1330\n",
      "[w] arginine [1..A.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.367.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.661.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 30\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 68\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 358/1074 [37:07<07:34,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4f3t.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 801\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.635.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.635.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 33%|███▎      | 359/1074 [37:08<08:13,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3eph.cif1'\n",
      "    total number of nucleotides: 138\n",
      "    total number of amino acids: 804\n",
      "[w] arginine [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.342.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.342.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 9\n",
      "    total number of helices: 4\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 163\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 6\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 50\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▎      | 360/1074 [37:09<10:40,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ski.cif1'\n",
      "  1.A.CCC.89 0.126\n",
      "  1.B.CCC.89 0.126\n",
      "    total number of nucleotides: 138\n",
      "    total number of base pairs: 66\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 170\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▎      | 361/1074 [37:09<08:32,  1.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5i9f.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 388\n",
      "[w] arginine [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▎      | 362/1074 [37:10<07:27,  1.59it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5e6m.cif1'\n",
      "    total number of nucleotides: 150\n",
      "    total number of amino acids: 1027\n",
      "[w] arginine [1..A.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.288.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.358.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 71\n",
      "    total number of multiplets: 9\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 160\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 35\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 363/1074 [37:11<10:46,  1.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wta.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 535\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 364/1074 [37:12<09:27,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wty.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 1091\n",
      "[w] arginine [1..B.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.318.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 365/1074 [37:13<10:32,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3p59.cif1'\n",
      "    total number of nucleotides: 98\n",
      "    total number of base pairs: 41\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 4\n",
      "    total number of stems: 11\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 102\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 6\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 366/1074 [37:13<08:20,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1hmh.cif1'\n",
      "    total number of nucleotides: 141\n",
      "    total number of base pairs: 68\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 6\n",
      "    total number of stems: 9\n",
      "    total number of non-pairing interactions: 190\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 3\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 6\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 367/1074 [37:14<07:19,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zbh.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 1028\n",
      "[w] arginine [1..B.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.308.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.308.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 368/1074 [37:14<07:43,  1.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/315d.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 369/1074 [37:15<06:01,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3gib.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 191\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 34%|███▍      | 370/1074 [37:15<05:00,  2.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5sup.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of amino acids: 1196\n",
      "[w] arginine [1..A.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.235.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.235.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.291.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▍      | 371/1074 [37:16<06:14,  1.88it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6icz.cif1'\n",
      "    total number of nucleotides: 420\n",
      "    total number of amino acids: 14175\n",
      "[w] arginine [1..A.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.663.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.758.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.762.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.845.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.474.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.663.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.758.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.762.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.845.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.995.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1370.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1414.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1427.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1471.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1865.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.160.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.569.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.730.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.785.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.457.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.545.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.602.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.495.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.139.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 124\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 11\n",
      "    total number of stems: 19\n",
      "    total number of isolated WC/wobble pairs: 9\n",
      "    total number of non-pairing interactions: 360\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 57\n",
      "                        consolidated into units: 32\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 3\n",
      "    total number of internal loops: 13\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 17\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 137\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▍      | 372/1074 [37:38<1:22:06,  7.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3jxq.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▍      | 373/1074 [37:38<58:47,  5.03s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dtd.cif1'\n",
      "    total number of nucleotides: 37\n",
      "    total number of amino acids: 1053\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.747.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.762.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.791.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.824.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.413.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.747.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.762.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.791.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.824.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▍      | 374/1074 [37:40<47:32,  4.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6c63.cif1'\n",
      "    total number of nucleotides: 104\n",
      "    total number of base pairs: 60\n",
      "    total number of multiplets: 9\n",
      "    total number of helices: 7\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 128\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 17\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 3\n",
      "    total number of G-tetrads: 9\n",
      "    total number of G4-helices: 2\n",
      "    total number of G4-stems: 3\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 41\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▍      | 375/1074 [37:40<34:30,  2.96s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6of1.cif1'\n",
      "  1.1a.U.841 0.123\n",
      "[i] taken as a pyrimidine ... 1.1w.CM0.34\n",
      "[i] taken as a pyrimidine ... 1.1y.CM0.34\n",
      "  1.2a.U.841 0.126\n",
      "[i] taken as a pyrimidine ... 1.2w.CM0.34\n",
      "[i] taken as a pyrimidine ... 1.2y.CM0.34\n",
      "  1.2y.7MG.46 0.120\n",
      "  1.2y.U.67 0.121\n",
      "1..1A.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1w.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1x.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1y.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2w.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2x.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2y.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9383\n",
      "    total number of amino acids: 11574\n",
      "[w] arginine [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1W.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..10.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..12.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..13.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..14.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1b.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1n.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2W.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..20.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..22.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..23.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..24.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2g.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2g.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2l.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2n.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2o.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2t.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4791\n",
      "    total number of multiplets: 747\n",
      "    total number of helices: 269\n",
      "    total number of stems: 594\n",
      "    total number of isolated WC/wobble pairs: 177\n",
      "    total number of non-pairing interactions: 13724\n",
      "    total number of atom-base capping interactions: 703\n",
      "    total number of splayed-apart dinucleotides: 1203\n",
      "                        consolidated into units: 757\n",
      "    total number of hairpin loops: 219\n",
      "    total number of bulges: 131\n",
      "    total number of internal loops: 220\n",
      "    total number of junctions: 117\n",
      "    total number of non-loop single-stranded segments: 81\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6of1.cif1', hairpin loops (#1/#3) have 2 interactions [#10, #11 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6of1.cif1', hairpin loops (#71/#73) have 2 interactions [#192, #193 ...]\n",
      "    total number of kissing loops: 16\n",
      "    total number of A-minor (types I and II) motifs: 277\n",
      "    total number of eXtended A-minor (type X) motifs: 213\n",
      "    total number of U-turns: 106\n",
      "    total number of ribose zippers: 129\n",
      "    total number of kink turns: 21\n",
      "    total number of phosphate interactions: 3562\n",
      "[w]  number-of-residues=379 -- in shortened form\n",
      "[w]  number-of-residues=379 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:02:47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▌      | 376/1074 [40:27<10:06:25, 52.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3pf4.cif1'\r\n",
      "    total number of nucleotides: 5\r\n",
      "    total number of amino acids: 132\r\n",
      "    total number of non-pairing interactions: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▌      | 377/1074 [40:27<7:04:27, 36.54s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qc0.cif1'\n",
      "  1.A.U.11 0.124\n",
      "    total number of nucleotides: 57\n",
      "    total number of base pairs: 28\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 56\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▌      | 378/1074 [40:28<4:57:40, 25.66s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5n94.cif1'\n",
      "  1.C.U.1 0.137\n",
      "  1.D.U.1 0.130\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 1698\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▌      | 379/1074 [40:29<3:34:15, 18.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6n4o.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 796\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▌      | 380/1074 [40:31<2:35:46, 13.47s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ccb.cif1'\n",
      "    total number of nucleotides: 78\n",
      "    total number of amino acids: 654\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 77\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 35%|███▌      | 381/1074 [40:33<1:55:54, 10.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2oeu.cif1'\n",
      "    total number of nucleotides: 63\n",
      "    total number of base pairs: 29\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 382/1074 [40:33<1:21:56,  7.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jxz.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of amino acids: 538\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 86\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 383/1074 [40:35<1:02:30,  5.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/359d.cif1'\n",
      "    total number of nucleotides: 41\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 384/1074 [40:35<44:23,  3.86s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2v7r.cif1'\r\n",
      "    total number of nucleotides: 14\r\n",
      "    total number of base pairs: 7\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 14\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 385/1074 [40:35<31:36,  2.75s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t5a.cif1'\n",
      "    total number of nucleotides: 62\n",
      "    total number of base pairs: 30\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 76\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 386/1074 [40:35<22:56,  2.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1c0a.cif1'\n",
      "  1.B.H2U.616 0.144\n",
      "  1.B.H2U.620 0.144\n",
      "  1.B.H2U.620A 0.162\n",
      "1..B.PSU.655. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..B.PSU.665. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 586\n",
      "[w] Residue 1..B.QUO.634. missing planar atom [ N7 ]\n",
      "\t[i] 1..B.QUO.634.: replace missing base ring atom [ N7 ] with [C7] (d=0.05)\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 94\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 387/1074 [40:37<20:43,  1.81s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5bz5.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 366\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 388/1074 [40:37<16:28,  1.44s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2qus.cif1'\n",
      "    total number of nucleotides: 138\n",
      "    total number of base pairs: 64\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 177\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▌      | 389/1074 [40:38<15:03,  1.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4q5s.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 3475\n",
      "\n",
      "Reverse linkage direction ...\n",
      "[w]  with nts out of order: 1..I.ATP.101.[3514] vs 1..I.C.2.[3506]\n",
      "[w] arginine [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.230.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.331.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.422.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.614.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.775.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.808.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1031.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.360.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.414.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.441.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.486.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.587.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.867.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.872.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1087.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1406.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.205.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.276.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 17\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:07\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▋      | 390/1074 [40:45<33:56,  2.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2gdi.cif1'\n",
      "    total number of nucleotides: 160\n",
      "    total number of base pairs: 78\n",
      "    total number of multiplets: 12\n",
      "    total number of helices: 5\n",
      "    total number of stems: 13\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 193\n",
      "    total number of atom-base capping interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 24\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 3\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 8\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 30\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▋      | 391/1074 [40:46<24:49,  2.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xwg.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 36%|███▋      | 392/1074 [40:46<18:30,  1.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/398d.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 393/1074 [40:46<13:37,  1.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5nwq.cif1'\n",
      "    total number of nucleotides: 82\n",
      "    total number of base pairs: 56\n",
      "    total number of multiplets: 16\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 92\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 394/1074 [40:49<19:21,  1.71s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cgp.cif1'\n",
      "1..A.PSU.6. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 25\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 395/1074 [40:49<14:12,  1.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2o3y.cif1'\n",
      "    total number of nucleotides: 43\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 396/1074 [40:50<10:44,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qvd.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 364\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 397/1074 [40:50<08:21,  1.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4u6k.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 40\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 398/1074 [40:50<06:33,  1.72it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4xw1.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 399/1074 [40:50<05:09,  2.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nl0.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 475\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 400/1074 [40:51<05:06,  2.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1r9f.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 121\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 38\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 401/1074 [40:51<04:59,  2.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e6b.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 11\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 37%|███▋      | 402/1074 [40:51<04:49,  2.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jvy.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 381\n",
      "    total number of non-pairing interactions: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 403/1074 [40:52<04:11,  2.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2von.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 373\n",
      "[w] arginine [1..B.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.57.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 404/1074 [40:52<03:59,  2.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2grb.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of base pairs: 42\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 3\n",
      "    total number of non-pairing interactions: 45\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of G-tetrads: 8\n",
      "    total number of G4-helices: 2\n",
      "    total number of G4-stems: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 405/1074 [40:52<03:23,  3.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zci.cif1'\n",
      "    total number of nucleotides: 92\n",
      "    total number of base pairs: 40\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 117\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of phosphate interactions: 9\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 406/1074 [40:52<03:22,  3.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5g4t.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of base pairs: 9\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 407/1074 [40:53<03:13,  3.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1q96.cif1'\n",
      "    total number of nucleotides: 81\n",
      "    total number of base pairs: 41\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 97\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 3\n",
      "    total number of U-turns: 3\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 408/1074 [40:53<03:04,  3.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rcj.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 188\n",
      "[w] arginine [1..A.ARG.390.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.390.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 409/1074 [40:53<02:45,  4.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1n32.cif1'\n",
      "1..Y.PSU.38. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 1532\n",
      "    total number of amino acids: 2389\n",
      "[w] arginine [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 785\n",
      "    total number of multiplets: 126\n",
      "    total number of helices: 44\n",
      "    total number of stems: 97\n",
      "    total number of isolated WC/wobble pairs: 23\n",
      "    total number of non-pairing interactions: 2243\n",
      "    total number of atom-base capping interactions: 105\n",
      "    total number of splayed-apart dinucleotides: 173\n",
      "                        consolidated into units: 108\n",
      "    total number of hairpin loops: 34\n",
      "    total number of bulges: 25\n",
      "    total number of internal loops: 36\n",
      "    total number of junctions: 19\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "    total number of A-minor (types I and II) motifs: 41\n",
      "    total number of eXtended A-minor (type X) motifs: 31\n",
      "    total number of U-turns: 24\n",
      "    total number of ribose zippers: 21\n",
      "    total number of kink turns: 4\n",
      "    total number of phosphate interactions: 596\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 410/1074 [41:07<48:54,  4.42s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4lx6.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of base pairs: 40\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 411/1074 [41:07<34:45,  3.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nd4.cif1'\r\n",
      "    total number of nucleotides: 16\r\n",
      "    total number of non-pairing interactions: 14\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 412/1074 [41:08<24:49,  2.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1jbr.cif1'\n",
      "    total number of nucleotides: 62\n",
      "    total number of amino acids: 298\n",
      "    total number of base pairs: 28\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 65\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 38%|███▊      | 413/1074 [41:08<18:59,  1.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5i9g.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 375\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▊      | 414/1074 [41:08<14:12,  1.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/464d.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▊      | 415/1074 [41:09<10:34,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6ck5.cif1'\n",
      "    total number of nucleotides: 208\n",
      "    total number of base pairs: 104\n",
      "    total number of multiplets: 13\n",
      "    total number of helices: 6\n",
      "    total number of stems: 17\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 267\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 15\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of ribose zippers: 4\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▊      | 416/1074 [41:09<09:37,  1.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nnh.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 344\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 417/1074 [41:10<08:42,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6msf.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 418/1074 [41:10<06:59,  1.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2h1m.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 32\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 419/1074 [41:10<05:26,  2.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hkc.cif1'\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 226\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 420/1074 [41:11<04:39,  2.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3p22.cif1'\n",
      "    total number of nucleotides: 194\n",
      "    total number of base pairs: 108\n",
      "    total number of multiplets: 30\n",
      "    total number of helices: 4\n",
      "    total number of stems: 12\n",
      "    total number of non-pairing interactions: 249\n",
      "    total number of atom-base capping interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 4\n",
      "    total number of non-loop single-stranded segments: 12\n",
      "    total number of A-minor (types I and II) motifs: 6\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 32\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 421/1074 [41:11<04:17,  2.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4lmz.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 176\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 422/1074 [41:11<03:51,  2.81it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1j6s.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of non-pairing interactions: 18\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 423/1074 [41:11<03:18,  3.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5det.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 185\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 39%|███▉      | 424/1074 [41:12<03:12,  3.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1pjg.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 17\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|███▉      | 425/1074 [41:12<02:51,  3.77it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6e0o.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 273\n",
      "[w] arginine [1..A.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.110.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|███▉      | 426/1074 [41:12<03:05,  3.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6muu.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 2083\n",
      "[w] arginine [1..A.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.687.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.660.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.687.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.724.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.735.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.755.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.776.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.777.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.173.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.173.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.383.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|███▉      | 427/1074 [41:14<07:47,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5d8t.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|███▉      | 428/1074 [41:14<05:58,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nku.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 630\n",
      "[w] arginine [1..A.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.351.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.351.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.349.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|███▉      | 429/1074 [41:15<05:40,  1.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gsj.cif1'\n",
      "  1.13.U.1 0.120\n",
      "  1.13.U.1056 0.120\n",
      "  1.1K.H2U.17 0.123\n",
      "  1.2K.7MG.47 0.130\n",
      "  1.1H.U.162 0.126\n",
      "  1.1H.U.871 0.124\n",
      "  1.1H.U.1313 0.123\n",
      "  1.1H.U.1394 0.129\n",
      "  1.1H.U.2262 0.121\n",
      "1..1K.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2K.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1L.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2L.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..3L.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9373\n",
      "    total number of amino acids: 11133\n",
      "[w] arginine [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..7E.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..4I.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..8I.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..7E.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4I.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..8I.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..41.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..51.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B8.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q8.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..22.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..62.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..19.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..29.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..49.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..95.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..95.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E5.ARG.32.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4658\n",
      "    total number of multiplets: 705\n",
      "    total number of helices: 270\n",
      "    total number of stems: 586\n",
      "    total number of isolated WC/wobble pairs: 172\n",
      "    total number of non-pairing interactions: 13401\n",
      "    total number of atom-base capping interactions: 652\n",
      "    total number of splayed-apart dinucleotides: 1212\n",
      "                        consolidated into units: 777\n",
      "    total number of hairpin loops: 222\n",
      "[w]  sheared G-A pair [5]: asymmetric internal loop: nts=9; [3,2]; linked by [#386,#-109]\n",
      "    total number of bulges: 125\n",
      "    total number of internal loops: 227\n",
      "    total number of junctions: 109\n",
      "    total number of non-loop single-stranded segments: 114\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gsj.cif1', hairpin loops (#39/#41) have 2 interactions [#118, #119 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gsj.cif1', hairpin loops (#151/#153) have 2 interactions [#412, #413 ...]\n",
      "    total number of kissing loops: 16\n",
      "    total number of A-minor (types I and II) motifs: 263\n",
      "    total number of eXtended A-minor (type X) motifs: 189\n",
      "    total number of U-turns: 102\n",
      "    total number of ribose zippers: 123\n",
      "    total number of kink turns: 20\n",
      "    total number of phosphate interactions: 3412\n",
      "[w]  number-of-residues=477 -- in shortened form\n",
      "[w]  number-of-residues=477 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:02:47\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|████      | 430/1074 [44:02<9:02:35, 50.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4c8z.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 469\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|████      | 431/1074 [44:02<6:21:01, 35.55s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2vod.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 375\n",
      "[w] arginine [1..B.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.57.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|████      | 432/1074 [44:03<4:28:10, 25.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wzj.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 534\n",
      "[w] arginine [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.329.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.468.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.629.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.329.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.468.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.629.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|████      | 433/1074 [44:04<3:09:48, 17.77s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1r3e.cif1'\n",
      "  1.C.FHU.410 0.224\n",
      "1..C.FHU.410. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 51\n",
      "    total number of amino acids: 305\n",
      "[w] arginine [1..A.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.240.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 57\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 40%|████      | 434/1074 [44:04<2:14:37, 12.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5y6z.cif1'\n",
      "    total number of nucleotides: 56\n",
      "    total number of amino acids: 917\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.455.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.455.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.455.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 23\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 47\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 435/1074 [44:05<1:36:57,  9.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5elh.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 282\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 436/1074 [44:05<1:08:36,  6.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5www.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 94\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 437/1074 [44:06<48:44,  4.59s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2e9t.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of amino acids: 948\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 16\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 438/1074 [44:06<36:22,  3.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/421d.cif1'\r\n",
      "    total number of nucleotides: 17\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 17\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 439/1074 [44:07<25:55,  2.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kxk.cif1'\n",
      "    total number of nucleotides: 70\n",
      "    total number of base pairs: 30\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 440/1074 [44:07<18:43,  1.77s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2awe.cif1'\n",
      "    total number of nucleotides: 47\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of non-pairing interactions: 62\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of G-tetrads: 6\n",
      "    total number of G4-helices: 2\n",
      "    total number of G4-stems: 2\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 441/1074 [44:07<13:50,  1.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yyw.cif1'\n",
      "    total number of nucleotides: 96\n",
      "    total number of amino acids: 875\n",
      "    total number of base pairs: 48\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 96\n",
      "    total number of atom-base capping interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 442/1074 [44:08<11:40,  1.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3g0h.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 408\n",
      "[w] arginine [1..A.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.372.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.372.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████      | 443/1074 [44:08<09:29,  1.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5axm.cif1'\n",
      "    total number of nucleotides: 73\n",
      "    total number of amino acids: 452\n",
      "[w] arginine [1..A.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.11.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 32\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 83\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████▏     | 444/1074 [44:09<08:08,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4m7d.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 1039\n",
      "[w] arginine [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.72.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 41%|████▏     | 445/1074 [44:09<06:59,  1.50it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6fpq.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 174\n",
      "[w] arginine [1..A.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.488.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 446/1074 [44:09<05:44,  1.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jrc.cif1'\n",
      "    total number of nucleotides: 114\n",
      "    total number of base pairs: 56\n",
      "    total number of multiplets: 9\n",
      "    total number of helices: 3\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 142\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 447/1074 [44:11<09:02,  1.15it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ohz.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 400\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 448/1074 [44:11<07:26,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jiy.cif1'\r\n",
      "    total number of nucleotides: 6\r\n",
      "    total number of non-pairing interactions: 2\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 449/1074 [44:11<05:40,  1.84it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4xwf.cif1'\n",
      "    total number of nucleotides: 64\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 69\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 450/1074 [44:12<04:42,  2.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4oog.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 480\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 451/1074 [44:12<04:23,  2.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1uvi.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 1992\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 452/1074 [44:14<08:30,  1.22it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q0r.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 343\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 453/1074 [44:14<07:31,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1jid.cif1'\n",
      "    total number of nucleotides: 29\n",
      "    total number of amino acids: 114\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 454/1074 [44:14<05:53,  1.75it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6c2p.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 653\n",
      "[w] arginine [1..B.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.134.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 455/1074 [44:15<05:34,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5yts.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 74\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 42%|████▏     | 456/1074 [44:15<04:46,  2.15it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cu1.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 84\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 457/1074 [44:16<04:54,  2.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ygh.cif1'\n",
      "  1.A.U.26 0.121\n",
      "    total number of nucleotides: 95\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 46\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 3\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 123\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 458/1074 [44:16<04:33,  2.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rc0.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 459/1074 [44:16<03:46,  2.71it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ex7.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 274\n",
      "[w] arginine [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.864.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.875.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.894.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1028.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.864.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.875.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.894.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1028.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 460/1074 [44:17<03:37,  2.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2oiu.cif1'\n",
      "    total number of nucleotides: 142\n",
      "    total number of base pairs: 65\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 4\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 173\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 461/1074 [44:17<04:10,  2.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k1v.cif1'\n",
      "    total number of nucleotides: 30\n",
      "[w] Residue 1..A.PRF.301. missing planar atom [ N7 ]\n",
      "\t[i] 1..A.PRF.301.: replace missing base ring atom [ N7 ] with [C7] (d=0.08)\n",
      "    total number of base pairs: 24\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 8\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 462/1074 [44:17<03:38,  2.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5elk.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 121\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 463/1074 [44:18<03:05,  3.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k4z.cif1'\n",
      "    total number of nucleotides: 134\n",
      "    total number of amino acids: 1848\n",
      "    total number of base pairs: 60\n",
      "    total number of helices: 6\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 123\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of phosphate interactions: 38\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 464/1074 [44:19<07:08,  1.42it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1u9s.cif1'\n",
      "    total number of nucleotides: 155\n",
      "    total number of base pairs: 80\n",
      "    total number of multiplets: 14\n",
      "    total number of helices: 4\n",
      "    total number of stems: 13\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 200\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 6\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of U-turns: 5\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 37\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 465/1074 [44:21<09:23,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fj4.cif1'\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 624\n",
      "    total number of base pairs: 25\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 69\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 466/1074 [44:21<08:25,  1.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wti.cif1'\n",
      "    total number of nucleotides: 154\n",
      "    total number of amino acids: 982\n",
      "[w] arginine [1..Z.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Z.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Z.ARG.324.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Z.ARG.852.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Z.ARG.944.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Z.ARG.966.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.291.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.324.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.852.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.944.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.966.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 5\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 170\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 9\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of phosphate interactions: 69\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 43%|████▎     | 467/1074 [44:23<10:09,  1.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e58.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of base pairs: 8\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▎     | 468/1074 [44:23<07:44,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1iha.cif1'\r\n",
      "    total number of nucleotides: 18\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 2\r\n",
      "    total number of non-pairing interactions: 18\r\n",
      "    total number of splayed-apart dinucleotides: 1\r\n",
      "    total number of internal loops: 1\r\n",
      "    total number of non-loop single-stranded segments: 2\r\n",
      "    total number of phosphate interactions: 3\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▎     | 469/1074 [44:23<05:54,  1.71it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4am3.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 1864\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 470/1074 [44:24<07:59,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4n48.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 808\n",
      "[w] arginine [1..A.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.471.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.471.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.220.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.396.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.546.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 471/1074 [44:25<07:16,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ew4.cif1'\n",
      "    total number of nucleotides: 39\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 472/1074 [44:25<05:38,  1.78it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4j7l.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 358\n",
      "[w] arginine [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 473/1074 [44:25<04:59,  2.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4c4w.cif1'\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 616\n",
      "    total number of base pairs: 27\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 474/1074 [44:26<05:28,  1.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3sqx.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 508\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 475/1074 [44:26<05:12,  1.92it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6h0r.cif1'\n",
      "    total number of nucleotides: 62\n",
      "    total number of base pairs: 29\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 64\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of phosphate interactions: 2\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 476/1074 [44:27<04:32,  2.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dh6.cif1'\n",
      "    total number of nucleotides: 68\n",
      "    total number of base pairs: 25\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 44%|████▍     | 477/1074 [44:27<03:47,  2.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1f8v.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 1049\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▍     | 478/1074 [44:28<04:30,  2.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4xw0.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▍     | 479/1074 [44:28<03:42,  2.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xtm.cif1'\n",
      "    total number of nucleotides: 94\n",
      "    total number of amino acids: 244\n",
      "[w] arginine [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.34.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 46\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 5\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 126\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 5\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of kink turns: 3\n",
      "    total number of phosphate interactions: 27\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▍     | 480/1074 [44:29<06:04,  1.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4yoe.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 181\n",
      "[w] arginine [1..A.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.146.] -- NH1/NH2 naming swapped\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▍     | 481/1074 [44:29<05:03,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4j39.cif1'\n",
      "  1.B.U.12 0.130\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 122\n",
      "[w] arginine [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 16\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▍     | 482/1074 [44:29<04:09,  2.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5msf.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▍     | 483/1074 [44:30<03:39,  2.69it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4mdx.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 230\n",
      "[w] arginine [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▌     | 484/1074 [44:30<03:07,  3.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3x1l.cif1'\n",
      "    total number of nucleotides: 54\n",
      "    total number of amino acids: 2470\n",
      "[w] arginine [1..A.ARG.769.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.769.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.4.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 42\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▌     | 485/1074 [44:32<07:02,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2dr2.cif1'\n",
      "    total number of nucleotides: 75\n",
      "    total number of amino acids: 374\n",
      "    total number of base pairs: 29\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 75\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▌     | 486/1074 [44:32<06:19,  1.55it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1fuf.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 28\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▌     | 487/1074 [44:32<05:19,  1.84it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3t3o.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 553\n",
      "[w] arginine [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.286.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.431.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.286.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.431.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 45%|████▌     | 488/1074 [44:33<05:04,  1.92it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nj7.cif1'\n",
      "    total number of nucleotides: 60\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 3\n",
      "    total number of stems: 9\n",
      "    total number of non-pairing interactions: 67\n",
      "    total number of internal loops: 6\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 489/1074 [44:33<04:08,  2.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3hga.cif1'\r\n",
      "    total number of nucleotides: 6\r\n",
      "    total number of non-pairing interactions: 4\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 490/1074 [44:33<03:17,  2.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pmw.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 1382\n",
      "    total number of non-pairing interactions: 18\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 491/1074 [44:35<07:36,  1.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1wpu.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 296\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 492/1074 [44:35<06:03,  1.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5kla.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 337\n",
      "[w] arginine [1..A.ARG.1199.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1275.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1329.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1199.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1275.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1329.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 493/1074 [44:36<05:30,  1.76it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3d2s.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 270\n",
      "[w] arginine [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.201.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.201.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.201.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.231.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.201.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.231.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 494/1074 [44:36<04:34,  2.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6bsg.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 949\n",
      "[w] arginine [1..A.ARG.356.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.356.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.356.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.356.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 495/1074 [44:37<05:24,  1.78it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6nod.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 1032\n",
      "[w] arginine [1..B.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.263.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.310.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.408.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.382.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.408.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.263.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.310.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.408.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.382.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.408.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▌     | 496/1074 [44:38<06:15,  1.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2w89.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▋     | 497/1074 [44:38<04:52,  1.97it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6r9p.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 607\n",
      "[w] arginine [1..A.ARG.524.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.624.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.804.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.995.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1000.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1030.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.524.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.624.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.804.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.995.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1000.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1030.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▋     | 498/1074 [44:38<04:56,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3amt.cif1'\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 417\n",
      "[w] arginine [1..A.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.369.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 94\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 46%|████▋     | 499/1074 [44:39<05:43,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kh6.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 23\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 500/1074 [44:39<05:00,  1.91it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ued.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of atom-base capping interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 501/1074 [44:40<04:04,  2.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5k78.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 1745\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 502/1074 [44:41<05:54,  1.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qik.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 599\n",
      "[w] arginine [1..A.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.343.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.297.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.352.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.354.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.159.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.343.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.297.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.352.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.354.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 503/1074 [44:41<06:01,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5z98.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 364\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ASP.100.] -- missing planar atom ' OD2'\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 504/1074 [44:42<05:10,  1.84it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qoz.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 630\n",
      "[w] arginine [1..E.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.247.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 505/1074 [44:42<05:13,  1.81it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1g2e.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 167\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 506/1074 [44:42<04:14,  2.23it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4x2b.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 477\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 507/1074 [44:43<04:18,  2.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2r7t.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 1073\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 508/1074 [44:44<06:34,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4xnr.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 509/1074 [44:45<05:51,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4zdo.cif1'\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 1802\n",
      "[w] Residue 1..A.PLR.1001. missing planar atom [ N3 ]\n",
      "\t[i] 1..A.PLR.1001.: replace missing base ring atom [ N3 ] with [C3] (d=0.07)\n",
      "[w] Residue 1..B.PLR.1001. missing planar atom [ N3 ]\n",
      "\t[i] 1..B.PLR.1001.: replace missing base ring atom [ N3 ] with [C3] (d=0.06)\n",
      "[w] Residue 1..C.PLR.1001. missing planar atom [ N3 ]\n",
      "\t[i] 1..C.PLR.1001.: replace missing base ring atom [ N3 ] with [C3] (d=0.06)\n",
      "[w] Residue 1..D.PLR.1001. missing planar atom [ N3 ]\n",
      "\t[i] 1..D.PLR.1001.: replace missing base ring atom [ N3 ] with [C3] (d=0.06)\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 4\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 47%|████▋     | 510/1074 [44:46<08:04,  1.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4cs1.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 511/1074 [44:46<06:52,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4gcw.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of amino acids: 307\n",
      "[w] arginine [1..A.ARG.306.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.306.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 23\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 66\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 512/1074 [44:47<06:52,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2zko.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 140\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 513/1074 [44:47<05:28,  1.71it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/280d.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 48\n",
      "    total number of internal loops: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 514/1074 [44:48<04:24,  2.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qlm.cif1'\n",
      "    total number of nucleotides: 110\n",
      "    total number of base pairs: 55\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 142\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 3\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 515/1074 [44:49<05:37,  1.66it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3u4m.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of amino acids: 228\n",
      "[w] arginine [1..A.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.160.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.160.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 41\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 104\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 3\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 516/1074 [44:49<05:59,  1.55it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e5c.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 517/1074 [44:49<04:41,  1.98it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4tyw.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 508\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 518/1074 [44:50<05:38,  1.64it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4enc.cif1'\n",
      "    total number of nucleotides: 52\n",
      "    total number of base pairs: 21\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 65\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 519/1074 [44:51<04:36,  2.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ivn.cif1'\n",
      "    total number of nucleotides: 138\n",
      "    total number of base pairs: 72\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 7\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 167\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 6\n",
      "    total number of kissing loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 8\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 48%|████▊     | 520/1074 [44:52<06:05,  1.51it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/333d.cif1'\r\n",
      "    total number of nucleotides: 8\r\n",
      "    total number of non-pairing interactions: 5\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▊     | 521/1074 [44:52<04:34,  2.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dhb.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of atom-base capping interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▊     | 522/1074 [44:52<03:55,  2.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5z4a.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 351\n",
      "[w] arginine [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▊     | 523/1074 [44:52<03:51,  2.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6noh.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 402\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 524/1074 [44:53<04:32,  2.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ftm.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of base pairs: 23\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 52\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 525/1074 [44:53<03:40,  2.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5amr.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 1657\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 526/1074 [44:56<10:21,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d9l.cif1'\n",
      "    total number of nucleotides: 75\n",
      "    total number of amino acids: 1516\n",
      "[w] arginine [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.285.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.541.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.285.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.541.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.678.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.285.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.541.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.678.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 34\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 78\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 35\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 527/1074 [44:58<11:52,  1.30s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1br3.cif1'\n",
      "    total number of nucleotides: 41\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 528/1074 [44:58<08:53,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5o3j.cif1'\r\n",
      "    total number of nucleotides: 2\r\n",
      "    total number of amino acids: 80\r\n",
      "[w] Residue [1..A.ARG.25.] -- missing planar atom ' NH2'\r\n",
      "    total number of non-pairing interactions: 1\r\n",
      "    total number of splayed-apart dinucleotides: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 529/1074 [44:58<06:36,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3sn2.cif1'\n",
      "    total number of nucleotides: 29\n",
      "    total number of amino acids: 848\n",
      "[w] arginine [1..A.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.536.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.699.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.832.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 530/1074 [44:59<07:48,  1.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2q1r.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 49%|████▉     | 531/1074 [45:00<06:19,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1j1u.cif1'\n",
      "    total number of nucleotides: 74\n",
      "    total number of amino acids: 300\n",
      "[w] arginine [1..A.ARG.303.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.303.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 96\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|████▉     | 532/1074 [45:00<05:43,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wtd.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 534\n",
      "[w] arginine [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|████▉     | 533/1074 [45:01<05:13,  1.73it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ki6.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 798\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|████▉     | 534/1074 [45:01<05:43,  1.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5yki.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 383\n",
      "[w] arginine [1..A.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.310.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.310.] -- NH1/NH2 naming swapped\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|████▉     | 535/1074 [45:02<04:50,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2yjy.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 678\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|████▉     | 536/1074 [45:02<04:56,  1.81it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/435d.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|█████     | 537/1074 [45:02<03:54,  2.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3avy.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of amino acids: 1203\n",
      "[w] arginine [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.547.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.573.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.612.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.618.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.658.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.710.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.711.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.914.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.937.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.997.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1088.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1197.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1210.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|█████     | 538/1074 [45:04<06:46,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5j7l.cif1'\n",
      "  1.AA.G7M.527 0.167\n",
      "  1.BA.G7M.527 0.164\n",
      "  1.CA.G7M.2069 0.167\n",
      "  1.DA.G7M.2069 0.141\n",
      "  1.DA.H2U.2449 0.154\n",
      "1..AA.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..BA.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.746. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.955. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.3TD.1915. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.2457. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.2504. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.2580. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.2604. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..CA.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.746. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.955. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.3TD.1915. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.2457. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.2504. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.2580. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.2604. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..DA.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9101\n",
      "    total number of amino acids: 11607\n",
      "[w] arginine [1..AB.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AF.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AI.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AI.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AL.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AN.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AP.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C1.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AB.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C1.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C1.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C1.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C2.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C3.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C4.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C4.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C4.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C4.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C5.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C0.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C0.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C0.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.225.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D2.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D4.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D4.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D4.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D4.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D5.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CC.ARG.212.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DC.ARG.212.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CH.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CQ.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CU.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CY.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DH.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DK.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DK.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DK.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DK.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DM.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DN.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DS.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DT.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DU.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DV.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DV.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DV.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DV.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DV.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DY.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DZ.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DZ.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DZ.ARG.52.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 4655\n",
      "    total number of multiplets: 735\n",
      "    total number of helices: 262\n",
      "    total number of stems: 571\n",
      "    total number of isolated WC/wobble pairs: 175\n",
      "    total number of non-pairing interactions: 12839\n",
      "    total number of atom-base capping interactions: 598\n",
      "    total number of splayed-apart dinucleotides: 1197\n",
      "                        consolidated into units: 736\n",
      "    total number of hairpin loops: 214\n",
      "    total number of bulges: 132\n",
      "    total number of internal loops: 239\n",
      "    total number of junctions: 114\n",
      "    total number of non-loop single-stranded segments: 44\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5j7l.cif1', hairpin loops (#69/#71) have 2 interactions [#-50, #217 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5j7l.cif1', hairpin loops (#143/#145) have 2 interactions [#-114, #402 ...]\n",
      "    total number of kissing loops: 14\n",
      "    total number of A-minor (types I and II) motifs: 287\n",
      "    total number of eXtended A-minor (type X) motifs: 174\n",
      "    total number of U-turns: 96\n",
      "    total number of ribose zippers: 135\n",
      "    total number of kink turns: 22\n",
      "    total number of phosphate interactions: 3210\n",
      "[w]  number-of-residues=368 -- in shortened form\n",
      "[w]  number-of-residues=368 -- in shortened form\n",
      "\n",
      "Time used: 00:00:03:14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|█████     | 539/1074 [48:18<8:45:13, 58.90s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/439d.cif1'\n",
      "  1.A.U.80 0.133\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|█████     | 540/1074 [48:19<6:07:26, 41.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5aox.cif1'\n",
      "    total number of nucleotides: 170\n",
      "    total number of amino acids: 319\n",
      "    total number of base pairs: 89\n",
      "    total number of multiplets: 12\n",
      "    total number of helices: 4\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 195\n",
      "    total number of atom-base capping interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 3\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5aox.cif1', hairpin loops (#1/#3) have 2 interactions [#-2, #3 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5aox.cif1', hairpin loops (#4/#6) have 2 interactions [#-5, #9 ...]\n",
      "    total number of kissing loops: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 40\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|█████     | 541/1074 [48:20<4:20:12, 29.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ts2.cif1'\n",
      "    total number of nucleotides: 91\n",
      "    total number of base pairs: 46\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 95\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of G-tetrads: 2\n",
      "    total number of G4-helices: 1\n",
      "    total number of G4-stems: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 50%|█████     | 542/1074 [48:20<3:02:40, 20.60s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kq2.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 365\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 543/1074 [48:21<2:08:15, 14.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5k7d.cif1'\n",
      "    total number of nucleotides: 58\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 64\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 544/1074 [48:21<1:31:22, 10.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4yye.cif1'\n",
      "    total number of nucleotides: 150\n",
      "    total number of amino acids: 837\n",
      "[w] arginine [1..A.ARG.320.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.320.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 71\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 154\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 30\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 545/1074 [48:22<1:06:48,  7.58s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4w5n.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 806\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 546/1074 [48:23<48:44,  5.54s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1xmq.cif1'\n",
      "    total number of nucleotides: 1522\n",
      "    total number of amino acids: 2396\n",
      "[w] arginine [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 766\n",
      "    total number of multiplets: 123\n",
      "    total number of helices: 44\n",
      "    total number of stems: 95\n",
      "    total number of isolated WC/wobble pairs: 24\n",
      "    total number of non-pairing interactions: 2185\n",
      "    total number of atom-base capping interactions: 97\n",
      "    total number of splayed-apart dinucleotides: 186\n",
      "                        consolidated into units: 115\n",
      "    total number of hairpin loops: 34\n",
      "    total number of bulges: 22\n",
      "    total number of internal loops: 38\n",
      "    total number of junctions: 19\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "    total number of A-minor (types I and II) motifs: 42\n",
      "    total number of eXtended A-minor (type X) motifs: 31\n",
      "    total number of U-turns: 22\n",
      "    total number of ribose zippers: 20\n",
      "    total number of kink turns: 3\n",
      "    total number of phosphate interactions: 583\n",
      "\n",
      "Time used: 00:00:00:12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 547/1074 [48:35<1:05:44,  7.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3hax.cif1'\n",
      "  1.F.FHU.7 0.164\n",
      "1..F.FHU.7. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 72\n",
      "    total number of amino acids: 503\n",
      "[w] arginine [1..C.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.46.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 77\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 548/1074 [48:36<47:17,  5.39s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1d4r.cif1'\n",
      "    total number of nucleotides: 85\n",
      "    total number of base pairs: 28\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 85\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 549/1074 [48:36<33:49,  3.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4fvu.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 208\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████     | 550/1074 [48:36<24:08,  2.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4gv6.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 207\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████▏    | 551/1074 [48:36<17:26,  2.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3iev.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 302\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████▏    | 552/1074 [48:37<13:10,  1.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5vw1.cif1'\n",
      "    total number of nucleotides: 73\n",
      "    total number of amino acids: 1448\n",
      "[w] arginine [1..A.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.324.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.655.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.324.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.655.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.820.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.864.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.884.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.919.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1114.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 40\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 51%|█████▏    | 553/1074 [48:39<14:46,  1.70s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1duq.cif1'\n",
      "    total number of nucleotides: 104\n",
      "    total number of base pairs: 47\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 12\n",
      "    total number of non-pairing interactions: 104\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 8\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 554/1074 [48:39<10:48,  1.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2csx.cif1'\n",
      "    total number of nucleotides: 150\n",
      "    total number of amino acids: 928\n",
      "    total number of base pairs: 63\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 178\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 6\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 28\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 555/1074 [48:40<10:31,  1.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2azx.cif1'\n",
      "    total number of nucleotides: 144\n",
      "    total number of amino acids: 778\n",
      "    total number of base pairs: 64\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 162\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 6\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 556/1074 [48:41<09:52,  1.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2gun.cif1'\r\n",
      "    total number of nucleotides: 17\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 2\r\n",
      "    total number of non-pairing interactions: 16\r\n",
      "    total number of bulges: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 557/1074 [48:41<07:15,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6az3.cif1'\n",
      "  1.1.U.973 0.121\n",
      "  1.1.U.1257 0.122\n",
      "  1.7.U.99 0.122\n",
      "    total number of nucleotides: 3507\n",
      "    total number of amino acids: 6390\n",
      "[w] arginine [1..A.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.326.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.234.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.310.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.328.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.206.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.158.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..h.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..k.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..n.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..o.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..o.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..p.ARG.87.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1575\n",
      "    total number of multiplets: 218\n",
      "    total number of helices: 98\n",
      "    total number of stems: 214\n",
      "    total number of isolated WC/wobble pairs: 94\n",
      "    total number of non-pairing interactions: 4444\n",
      "    total number of atom-base capping interactions: 211\n",
      "    total number of splayed-apart dinucleotides: 580\n",
      "                        consolidated into units: 341\n",
      "    total number of hairpin loops: 68\n",
      "    total number of bulges: 54\n",
      "    total number of internal loops: 89\n",
      "    total number of junctions: 36\n",
      "    total number of non-loop single-stranded segments: 87\n",
      "    total number of kissing loops: 6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of A-minor (types I and II) motifs: 65\n",
      "    total number of eXtended A-minor (type X) motifs: 59\n",
      "    total number of U-turns: 22\n",
      "    total number of ribose zippers: 36\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 1363\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 558/1074 [49:16<1:34:39, 11.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5v7c.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 144\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 559/1074 [49:16<1:07:08,  7.82s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zz5.cif1'\n",
      "    total number of nucleotides: 58\n",
      "    total number of base pairs: 25\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 52\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 560/1074 [49:17<47:59,  5.60s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4yaz.cif1'\n",
      "    total number of nucleotides: 170\n",
      "    total number of base pairs: 91\n",
      "    total number of multiplets: 18\n",
      "    total number of helices: 6\n",
      "    total number of stems: 14\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 206\n",
      "    total number of atom-base capping interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 25\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 8\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 561/1074 [49:19<37:47,  4.42s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2zzn.cif1'\n",
      "    total number of nucleotides: 144\n",
      "    total number of amino acids: 673\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.315.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 63\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 159\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 22\n",
      "                        consolidated into units: 15\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 42\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 562/1074 [49:19<28:33,  3.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jng.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 887\n",
      "    total number of non-pairing interactions: 26\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 52%|█████▏    | 563/1074 [49:20<22:27,  2.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1fxl.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 167\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 564/1074 [49:21<16:31,  1.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5voe.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 282\n",
      "    total number of base pairs: 15\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 565/1074 [49:21<13:18,  1.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nh6.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 272\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.986.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1038.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.986.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1038.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 566/1074 [49:22<10:14,  1.21s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kuq.cif1'\n",
      "    total number of nucleotides: 57\n",
      "    total number of amino acids: 84\n",
      "    total number of base pairs: 27\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 60\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 567/1074 [49:22<08:07,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xbm.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 1041\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 568/1074 [49:23<07:18,  1.15it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6fz0.cif1'\n",
      "  1.A.CBV.4 0.130\n",
      "    total number of nucleotides: 49\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 25\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 52\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 569/1074 [49:23<05:37,  1.50it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mj0.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 118\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 570/1074 [49:23<04:25,  1.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4r3i.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 164\n",
      "[w] arginine [1..A.ARG.395.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.404.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.475.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.395.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.404.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.475.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 571/1074 [49:23<03:37,  2.31it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4b3r.cif1'\n",
      "    total number of nucleotides: 1531\n",
      "    total number of amino acids: 2404\n",
      "    total number of base pairs: 791\n",
      "    total number of multiplets: 128\n",
      "    total number of helices: 44\n",
      "    total number of stems: 98\n",
      "    total number of isolated WC/wobble pairs: 22\n",
      "    total number of non-pairing interactions: 2196\n",
      "    total number of atom-base capping interactions: 103\n",
      "    total number of splayed-apart dinucleotides: 180\n",
      "                        consolidated into units: 110\n",
      "    total number of hairpin loops: 34\n",
      "    total number of bulges: 25\n",
      "    total number of internal loops: 36\n",
      "    total number of junctions: 19\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "    total number of A-minor (types I and II) motifs: 41\n",
      "    total number of eXtended A-minor (type X) motifs: 37\n",
      "    total number of U-turns: 24\n",
      "    total number of ribose zippers: 19\n",
      "    total number of kink turns: 4\n",
      "    total number of phosphate interactions: 596\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 572/1074 [49:35<32:31,  3.89s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3qgb.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 400\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 573/1074 [49:36<23:42,  2.84s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f8j.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of amino acids: 462\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.346.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.416.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.438.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.455.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.346.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.416.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.438.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.455.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 53%|█████▎    | 574/1074 [49:36<17:37,  2.12s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/255d.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▎    | 575/1074 [49:36<12:45,  1.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3og8.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 254\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 28\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▎    | 576/1074 [49:37<09:38,  1.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f0q.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 372\n",
      "[w] arginine [1..B.ARG.453.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.453.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 27\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▎    | 577/1074 [49:37<07:30,  1.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3vjr.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of amino acids: 386\n",
      "[w] arginine [1..A.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.174.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 578/1074 [49:37<06:06,  1.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ssf.cif1'\n",
      "  1.B.DT.16 0.127\n",
      "  1.B.DT.17 0.120\n",
      "  1.B.DT.23 0.129\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 579/1074 [49:37<04:41,  1.76it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5elr.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 225\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 580/1074 [49:38<03:52,  2.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/397d.cif1'\n",
      "    total number of nucleotides: 27\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 581/1074 [49:38<03:08,  2.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zkj.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 600\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 23\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 46\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 582/1074 [49:38<03:42,  2.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wtc.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 534\n",
      "[w] arginine [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 583/1074 [49:39<03:45,  2.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4u3l.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 18\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 584/1074 [49:39<03:01,  2.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zsl.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 1550\n",
      "[w] arginine [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.627.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.473.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.627.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 54%|█████▍    | 585/1074 [49:41<05:53,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bnt.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of non-pairing interactions: 20\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▍    | 586/1074 [49:41<04:32,  1.79it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3la5.cif1'\n",
      "    total number of nucleotides: 71\n",
      "    total number of base pairs: 36\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 76\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▍    | 587/1074 [49:41<04:05,  1.98it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4f02.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 389\n",
      "[w] arginine [1..A.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.83.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▍    | 588/1074 [49:42<03:32,  2.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3gvn.cif1'\n",
      "  1.B.C.67 0.145\n",
      "  1.B.C.69 0.120\n",
      "    total number of nucleotides: 14\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▍    | 589/1074 [49:42<02:55,  2.77it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jzu.cif1'\n",
      "  *1.B.EPE.201 0.243 (planarity=0.268)\n",
      "  1.B.EPE.201 0.243\n",
      "[i] taken as a pyrimidine ... 1.B.EPE.201\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 308\n",
      "[w] Residue 1..B.EPE.201. missing planar atom [ C4 ]\n",
      "[w] Residue 1..B.EPE.201. missing planar atom [ N3 ]\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▍    | 590/1074 [49:42<02:41,  2.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4b3g.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 1222\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▌    | 591/1074 [49:43<04:07,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hby.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▌    | 592/1074 [49:43<03:18,  2.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3gs5.cif1'\n",
      "    total number of nucleotides: 61\n",
      "    total number of base pairs: 30\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 73\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 9\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▌    | 593/1074 [49:44<03:23,  2.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ndi.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of base pairs: 40\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 84\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of phosphate interactions: 13\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▌    | 594/1074 [49:44<02:57,  2.71it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5the.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 2049\n",
      "[w] arginine [1..E.ARG.1025.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.1025.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▌    | 595/1074 [49:46<06:16,  1.27it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mdi.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 406\n",
      "[w] arginine [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.221.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 55%|█████▌    | 596/1074 [49:46<05:21,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1b7f.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 334\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 597/1074 [49:46<04:46,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6h4n.cif1'\n",
      "  1.A.U.2312 0.124\n",
      "    total number of nucleotides: 4636\n",
      "    total number of amino acids: 6188\n",
      "[w] arginine [1..b.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..b.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..c.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..c.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..e.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..i.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..l.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..p.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..l.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..p.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..y.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..y.ARG.283.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..y.ARG.340.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.17.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2309\n",
      "    total number of multiplets: 347\n",
      "    total number of helices: 124\n",
      "    total number of stems: 288\n",
      "    total number of isolated WC/wobble pairs: 84\n",
      "    total number of non-pairing interactions: 6217\n",
      "    total number of atom-base capping interactions: 279\n",
      "    total number of splayed-apart dinucleotides: 616\n",
      "                        consolidated into units: 386\n",
      "    total number of hairpin loops: 108\n",
      "    total number of bulges: 66\n",
      "    total number of internal loops: 119\n",
      "    total number of junctions: 57\n",
      "    total number of non-loop single-stranded segments: 25\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6h4n.cif1', hairpin loops (#33/#35) have 2 interactions [#-21, #110 ...]\n",
      "    total number of kissing loops: 7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of A-minor (types I and II) motifs: 134\n",
      "    total number of eXtended A-minor (type X) motifs: 92\n",
      "    total number of U-turns: 48\n",
      "    total number of ribose zippers: 59\n",
      "    total number of kink turns: 10\n",
      "    total number of phosphate interactions: 1344\n",
      "\n",
      "Time used: 00:00:01:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 598/1074 [50:50<2:34:41, 19.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k4v.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of amino acids: 924\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.128.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 599/1074 [50:51<1:49:38, 13.85s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jc7.cif1'\n",
      "    total number of nucleotides: 50\n",
      "    total number of amino acids: 1288\n",
      "    total number of base pairs: 23\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 53\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 600/1074 [50:52<1:19:23, 10.05s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2r7v.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 1073\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 601/1074 [50:53<58:27,  7.42s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4x4n.cif1'\n",
      "    total number of nucleotides: 85\n",
      "    total number of amino acids: 1747\n",
      "[w] arginine [1..A.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.373.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 39\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 90\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 9\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 27\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 602/1074 [50:55<45:17,  5.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1si2.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 120\n",
      "[w] arginine [1..A.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.253.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 603/1074 [50:55<32:39,  4.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cjz.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 28\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▌    | 604/1074 [50:56<23:59,  3.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1ec6.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of amino acids: 171\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▋    | 605/1074 [50:56<17:41,  2.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3qsu.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 851\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 56%|█████▋    | 606/1074 [50:57<13:43,  1.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k27.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of base pairs: 26\n",
      "    total number of helices: 1\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 62\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 4\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 607/1074 [50:57<10:26,  1.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5tbw.cif1'\n",
      "    total number of nucleotides: 10420\n",
      "    total number of amino acids: 22625\n",
      "[w] arginine [1..j.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..j.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..j.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..j.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..j.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..k.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..l.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..l.ARG.326.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..m.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..j.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..k.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..l.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..l.ARG.326.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..o.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..p.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..q.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..q.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..q.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..r.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..r.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..u.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..u.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..v.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..y.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..y.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..y.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..z.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..z.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..z.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..0.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..8.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..8.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..9.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AA.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AA.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CD.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CE.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CF.ARG.326.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CI.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CJ.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CK.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CL.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CL.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CM.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CN.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CO.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CP.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CR.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CS.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CU.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CU.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CU.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CU.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CV.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CX.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CZ.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DA.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DB.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DB.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DF.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DH.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DJ.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DL.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DQ.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DR.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..p0.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..sM.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..sM.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..sM.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s0.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s1.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s1.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s1.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s1.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s2.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s3.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s3.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s3.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s3.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s3.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s4.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s4.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s4.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s4.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s4.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s4.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s5.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s5.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s5.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s5.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s6.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s6.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s6.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s6.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s6.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s7.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s7.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s8.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s8.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s8.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s8.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s8.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s8.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..s9.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c1.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c3.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c4.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c4.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c6.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c7.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c7.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c8.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c8.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c8.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c8.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c8.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d0.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d1.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d3.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d3.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d4.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d4.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d5.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d5.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d5.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d5.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d6.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d6.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d6.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d8.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e0.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e1.ARG.119.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 4824\n",
      "    total number of multiplets: 640\n",
      "    total number of helices: 301\n",
      "    total number of stems: 651\n",
      "    total number of isolated WC/wobble pairs: 243\n",
      "    total number of non-pairing interactions: 13564\n",
      "    total number of atom-base capping interactions: 686\n",
      "    total number of splayed-apart dinucleotides: 1635\n",
      "                        consolidated into units: 984\n",
      "    total number of hairpin loops: 232\n",
      "    total number of bulges: 149\n",
      "    total number of internal loops: 321\n",
      "    total number of junctions: 128\n",
      "    total number of non-loop single-stranded segments: 87\n",
      "    total number of kissing loops: 10\n",
      "    total number of A-minor (types I and II) motifs: 218\n",
      "    total number of eXtended A-minor (type X) motifs: 209\n",
      "    total number of U-turns: 102\n",
      "    total number of ribose zippers: 109\n",
      "    total number of kink turns: 17\n",
      "    total number of phosphate interactions: 5655\n",
      "[w]  number-of-residues=606 -- in shortened form\n",
      "[w]  number-of-residues=606 -- in shortened form\n",
      "\n",
      "Time used: 00:00:04:48\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 608/1074 [55:46<11:19:54, 87.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3kmq.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 609/1074 [55:47<7:56:46, 61.52s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jf2.cif1'\n",
      "    total number of nucleotides: 78\n",
      "[w] Residue 1..A.PRF.101. missing planar atom [ N7 ]\n",
      "\t[i] 1..A.PRF.101.: replace missing base ring atom [ N7 ] with [C7] (d=0.09)\n",
      "    total number of base pairs: 37\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 86\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 610/1074 [55:47<5:34:48, 43.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2r8s.cif1'\n",
      "    total number of nucleotides: 159\n",
      "    total number of amino acids: 433\n",
      "    total number of base pairs: 72\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 4\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 193\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 26\n",
      "                        consolidated into units: 16\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 6\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 23\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 611/1074 [55:48<3:54:59, 30.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1u0b.cif1'\n",
      "    total number of nucleotides: 74\n",
      "    total number of amino acids: 461\n",
      "[w] arginine [1..B.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.368.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.427.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.459.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.368.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.427.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.459.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.460.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 35\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 612/1074 [55:49<2:47:12, 21.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5m0j.cif1'\n",
      "    total number of nucleotides: 56\n",
      "    total number of amino acids: 1008\n",
      "[w] arginine [1..D.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.26.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 53\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 613/1074 [55:51<1:59:45, 15.59s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5elt.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 298\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 614/1074 [55:51<1:24:23, 11.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kfo.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of non-pairing interactions: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 615/1074 [55:51<59:31,  7.78s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rbz.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 616/1074 [55:51<41:58,  5.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t83.cif1'\n",
      "    total number of nucleotides: 89\n",
      "    total number of base pairs: 47\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 112\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 28\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 57%|█████▋    | 617/1074 [55:52<30:27,  4.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wtl.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 553\n",
      "[w] arginine [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.505.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 618/1074 [55:53<22:48,  3.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5udi.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 460\n",
      "    total number of non-pairing interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 619/1074 [55:53<17:09,  2.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6do8.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 135\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 620/1074 [55:53<12:32,  1.66s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ao5.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 621/1074 [55:54<09:13,  1.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xj2.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 1813\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 622/1074 [55:55<09:02,  1.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xdb.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 162\n",
      "[w] arginine [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 623/1074 [55:55<06:52,  1.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dlr.cif1'\n",
      "    total number of nucleotides: 107\n",
      "    total number of base pairs: 57\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 3\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 132\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 44\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 624/1074 [55:55<05:47,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mdg.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 410\n",
      "[w] arginine [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.221.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 625/1074 [55:56<05:00,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6f4g.cif1'\n",
      "    total number of nucleotides: 50\n",
      "    total number of amino acids: 536\n",
      "[w] arginine [1..A.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.67.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 48\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 626/1074 [55:56<04:37,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3avx.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of amino acids: 1203\n",
      "[w] arginine [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.564.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.573.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.612.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.618.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.710.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.711.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.752.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.914.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.937.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.997.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1088.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1197.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1249.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 627/1074 [55:58<06:48,  1.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1s76.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 829\n",
      "    total number of base pairs: 13\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 10\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 58%|█████▊    | 628/1074 [55:59<06:50,  1.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6is0.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 468\n",
      "[w] arginine [1..A.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.378.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.378.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▊    | 629/1074 [55:59<05:41,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1z7f.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 48\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▊    | 630/1074 [56:00<04:38,  1.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4hot.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 476\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 631/1074 [56:00<04:40,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jgn.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 125\n",
      "[w] arginine [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 17\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 632/1074 [56:00<03:50,  1.92it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6fpx.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 528\n",
      "[w] arginine [1..A.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.431.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.431.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.431.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.431.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.488.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 633/1074 [56:01<03:31,  2.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4i67.cif1'\r\n",
      "    total number of nucleotides: 4\r\n",
      "    total number of amino acids: 76\r\n",
      "[w] arginine [1..A.ARG.444.] -- NH1/NH2 naming swapped\r\n",
      "[w] arginine [1..A.ARG.480.] -- NH1/NH2 naming swapped\r\n",
      "[w] arginine [1..A.ARG.487.] -- NH1/NH2 naming swapped\r\n",
      "[w] Residue [1..A.ARG.444.] -- NH1/NH2 naming swapped\r\n",
      "[w] Residue [1..A.ARG.480.] -- NH1/NH2 naming swapped\r\n",
      "[w] Residue [1..A.ARG.487.] -- NH1/NH2 naming swapped\r\n",
      "    total number of non-pairing interactions: 1\r\n",
      "    total number of atom-base capping interactions: 1\r\n",
      "    total number of splayed-apart dinucleotides: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 634/1074 [56:01<02:49,  2.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3f2q.cif1'\n",
      "  1.X.CCC.112 0.128\n",
      "    total number of nucleotides: 110\n",
      "    total number of base pairs: 52\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 5\n",
      "    total number of stems: 9\n",
      "    total number of non-pairing interactions: 145\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 635/1074 [56:01<02:33,  2.87it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1ddl.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 551\n",
      "[w] arginine [1..B.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.67.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 636/1074 [56:02<02:31,  2.89it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2bte.cif1'\n",
      "    total number of nucleotides: 158\n",
      "    total number of amino acids: 1756\n",
      "    total number of base pairs: 76\n",
      "    total number of multiplets: 9\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 192\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 17\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of kissing loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 33\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 637/1074 [56:04<06:10,  1.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zx7.cif1'\n",
      "    total number of nucleotides: 116\n",
      "    total number of base pairs: 50\n",
      "    total number of helices: 2\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 132\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "    total number of internal loops: 8\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 7\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 638/1074 [56:04<05:08,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3v7e.cif1'\n",
      "    total number of nucleotides: 254\n",
      "    total number of amino acids: 160\n",
      "    total number of base pairs: 126\n",
      "    total number of multiplets: 13\n",
      "    total number of helices: 8\n",
      "    total number of stems: 14\n",
      "    total number of isolated WC/wobble pairs: 7\n",
      "    total number of non-pairing interactions: 321\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 7\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 6\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 59%|█████▉    | 639/1074 [56:05<05:46,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4n2s.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 197\n",
      "[w] arginine [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|█████▉    | 640/1074 [56:05<04:33,  1.59it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4kr6.cif1'\n",
      "    total number of nucleotides: 78\n",
      "    total number of amino acids: 776\n",
      "[w] arginine [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.382.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.382.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.382.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 24\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 78\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|█████▉    | 641/1074 [56:06<04:41,  1.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4tyy.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 505\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|█████▉    | 642/1074 [56:07<05:12,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5kpy.cif1'\n",
      "    total number of nucleotides: 71\n",
      "    total number of amino acids: 1\n",
      "[w]  chainId 'A' with nts=71 and aas=1\n",
      "    total number of base pairs: 36\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 70\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|█████▉    | 643/1074 [56:07<04:19,  1.66it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qu2.cif1'\n",
      "    total number of nucleotides: 75\n",
      "    total number of amino acids: 917\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 83\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 27\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|█████▉    | 644/1074 [56:09<06:12,  1.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3siu.cif1'\n",
      "    total number of nucleotides: 56\n",
      "    total number of amino acids: 721\n",
      "[w] arginine [1..A.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.137.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.203.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 67\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|██████    | 645/1074 [56:09<06:03,  1.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ibb.cif1'\n",
      "  1.13.U.125 0.127\n",
      "  1.13.U.560 0.126\n",
      "  1.13.U.723 0.123\n",
      "  1.1K.CM0.34 0.140\n",
      "[i] taken as a pyrimidine ... 1.1K.CM0.34\n",
      "  1.2K.7MG.47 0.133\n",
      "  1.1H.U.114 0.122\n",
      "  1.1H.U.810 0.121\n",
      "  1.1H.A.945 0.132\n",
      "  1.1H.U.1255 0.125\n",
      "1..1K.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2K.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1L.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2L.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9372\n",
      "    total number of amino acids: 11387\n",
      "[w] arginine [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..8I.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..71.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..8I.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..71.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..41.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B8.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q8.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..62.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..79.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..19.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..29.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..49.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..95.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E5.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I5.ARG.58.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4652\n",
      "    total number of multiplets: 701\n",
      "    total number of helices: 259\n",
      "    total number of stems: 583\n",
      "    total number of isolated WC/wobble pairs: 178\n",
      "    total number of non-pairing interactions: 13488\n",
      "    total number of atom-base capping interactions: 684\n",
      "    total number of splayed-apart dinucleotides: 1188\n",
      "                        consolidated into units: 755\n",
      "    total number of hairpin loops: 222\n",
      "    total number of bulges: 124\n",
      "    total number of internal loops: 229\n",
      "    total number of junctions: 111\n",
      "    total number of non-loop single-stranded segments: 94\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ibb.cif1', hairpin loops (#39/#41) have 2 interactions [#117, #118 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ibb.cif1', hairpin loops (#151/#153) have 2 interactions [#411, #412 ...]\n",
      "    total number of kissing loops: 16\n",
      "    total number of A-minor (types I and II) motifs: 261\n",
      "    total number of eXtended A-minor (type X) motifs: 188\n",
      "    total number of U-turns: 103\n",
      "    total number of ribose zippers: 129\n",
      "    total number of kink turns: 20\n",
      "    total number of phosphate interactions: 3398\n",
      "[w]  number-of-residues=487 -- in shortened form\n",
      "[w]  number-of-residues=487 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:03:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|██████    | 646/1074 [59:14<6:38:27, 55.86s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5vj9.cif1'\n",
      "    total number of nucleotides: 64\n",
      "    total number of base pairs: 36\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 68\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of phosphate interactions: 18\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|██████    | 647/1074 [59:15<4:41:35, 39.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3zc0.cif1'\n",
      "    total number of nucleotides: 39\n",
      "    total number of amino acids: 2269\n",
      "    total number of non-pairing interactions: 36\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|██████    | 648/1074 [59:17<3:20:12, 28.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ktg.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 121\n",
      "[w] arginine [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of internal loops: 5\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 60%|██████    | 649/1074 [59:17<2:20:28, 19.83s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4x4v.cif1'\n",
      "    total number of nucleotides: 64\n",
      "    total number of amino acids: 874\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 650/1074 [59:19<1:41:00, 14.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wwe.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 174\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.153.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 651/1074 [59:19<1:11:14, 10.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wrt.cif1'\n",
      "    total number of nucleotides: 29\n",
      "    total number of amino acids: 1701\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 652/1074 [59:22<55:30,  7.89s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jab.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 653/1074 [59:22<39:08,  5.58s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2qux.cif1'\n",
      "  1.I.U.18 0.123\n",
      "  1.I.U.19 0.124\n",
      "  1.L.U.22 0.125\n",
      "    total number of nucleotides: 150\n",
      "    total number of amino acids: 1452\n",
      "[w] arginine [1..A.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.54.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 54\n",
      "    total number of helices: 6\n",
      "    total number of stems: 12\n",
      "    total number of non-pairing interactions: 146\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 48\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 6\n",
      "    total number of phosphate interactions: 43\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 654/1074 [59:24<32:08,  4.59s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3fte.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 233\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 655/1074 [59:24<22:58,  3.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/283d.cif1'\r\n",
      "    total number of nucleotides: 12\r\n",
      "    total number of non-pairing interactions: 9\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 656/1074 [59:24<16:19,  2.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2zzm.cif1'\n",
      "    total number of nucleotides: 85\n",
      "    total number of amino acids: 334\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.261.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.315.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 36\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 102\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 27\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████    | 657/1074 [59:25<12:10,  1.75s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r2d.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 428\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████▏   | 658/1074 [59:25<09:12,  1.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yvp.cif1'\n",
      "    total number of nucleotides: 52\n",
      "    total number of amino acids: 1061\n",
      "[w] arginine [1..A.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.531.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.531.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.405.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.531.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████▏   | 659/1074 [59:26<08:46,  1.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rzo.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 3561\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:05\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 61%|██████▏   | 660/1074 [59:31<16:40,  2.42s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2nvq.cif1'\n",
      "    total number of nucleotides: 53\n",
      "    total number of amino acids: 3561\n",
      "    total number of base pairs: 21\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 15\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:05\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 661/1074 [59:37<22:34,  3.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4d25.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 427\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 662/1074 [59:37<16:35,  2.42s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/404d.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 663/1074 [59:37<12:03,  1.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4zlr.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 556\n",
      "[w] arginine [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.894.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1028.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.812.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.837.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.875.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.894.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1028.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.812.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.837.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.875.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 664/1074 [59:38<09:34,  1.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4rkv.cif1'\r\n",
      "    total number of nucleotides: 12\r\n",
      "    total number of base pairs: 2\r\n",
      "    total number of non-pairing interactions: 13\r\n",
      "    total number of atom-base capping interactions: 2\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 2\r\n",
      "    total number of phosphate interactions: 5\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 665/1074 [59:38<06:59,  1.03s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fjc.cif1'\n",
      "    total number of nucleotides: 95\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 46\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 3\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 117\n",
      "    total number of atom-base capping interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 666/1074 [59:38<05:41,  1.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ngg.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 267\n",
      "[w] arginine [1..A.ARG.779.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.779.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 667/1074 [59:39<04:31,  1.50it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1mhk.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of base pairs: 9\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 32\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 668/1074 [59:39<03:33,  1.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/361d.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 669/1074 [59:39<02:54,  2.33it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jbg.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 661\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 670/1074 [59:40<03:33,  1.89it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d3p.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 62%|██████▏   | 671/1074 [59:40<02:51,  2.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4g9z.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 205\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 672/1074 [59:40<02:28,  2.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2j0s.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 667\n",
      "[w] arginine [1..C.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..T.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.176.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 673/1074 [59:41<02:36,  2.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xuz.cif1'\n",
      "    total number of nucleotides: 156\n",
      "    total number of amino acids: 2424\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.385.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.645.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.385.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.645.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.737.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.836.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.883.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.385.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.543.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.737.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.836.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.883.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.1033.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 76\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 6\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 175\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 87\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 674/1074 [59:45<09:23,  1.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5udz.cif1'\n",
      "    total number of nucleotides: 47\n",
      "    total number of amino acids: 278\n",
      "[w] arginine [1..A.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.138.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 675/1074 [59:45<07:30,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1gax.cif1'\n",
      "    total number of nucleotides: 152\n",
      "    total number of amino acids: 1726\n",
      "    total number of base pairs: 64\n",
      "    total number of multiplets: 9\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 163\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 34\n",
      "                        consolidated into units: 16\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 37\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 676/1074 [59:47<09:51,  1.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d1v.cif1'\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 434\n",
      "[w] arginine [1..A.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 677/1074 [59:48<07:32,  1.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2e9z.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.417.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.417.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 678/1074 [59:48<06:07,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3b0u.cif1'\n",
      "  1.A.H2U.20 0.191\n",
      "  1.B.H2U.20 0.194\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 624\n",
      "[w] arginine [1..X.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..X.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.241.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.248.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.286.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.294.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.307.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.241.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.248.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.286.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.300.] -- NH1/NH2 naming swapped\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 679/1074 [59:49<05:11,  1.27it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ngc.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 276\n",
      "[w] arginine [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.986.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.934.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.986.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 680/1074 [59:49<04:12,  1.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5vaj.cif1'\n",
      "    total number of nucleotides: 23\n",
      "    total number of amino acids: 271\n",
      "[w] arginine [1..B.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.123.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 63%|██████▎   | 681/1074 [59:49<03:31,  1.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1efo.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of bulges: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▎   | 682/1074 [59:49<02:50,  2.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1vq6.cif1'\n",
      "1..0.PSU.2621. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 2881\n",
      "    total number of amino acids: 3776\n",
      "    total number of base pairs: 1473\n",
      "    total number of multiplets: 249\n",
      "    total number of helices: 88\n",
      "    total number of stems: 180\n",
      "    total number of isolated WC/wobble pairs: 56\n",
      "    total number of non-pairing interactions: 4113\n",
      "    total number of atom-base capping interactions: 242\n",
      "    total number of splayed-apart dinucleotides: 414\n",
      "                        consolidated into units: 259\n",
      "    total number of hairpin loops: 68\n",
      "    total number of bulges: 37\n",
      "    total number of internal loops: 67\n",
      "    total number of junctions: 36\n",
      "    total number of non-loop single-stranded segments: 33\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/1vq6.cif1', hairpin loops (#1/#3) have 2 interactions [#8, #9 ...]\n",
      "    total number of kissing loops: 5\n",
      "    total number of A-minor (types I and II) motifs: 104\n",
      "    total number of eXtended A-minor (type X) motifs: 54\n",
      "    total number of U-turns: 34\n",
      "    total number of ribose zippers: 46\n",
      "    total number of kink turns: 8\n",
      "    total number of phosphate interactions: 1150\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:33\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▎   | 683/1074 [1:00:23<1:07:02, 10.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1n38.cif1'\n",
      "  1.A.U3H.1291 0.146\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 1264\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▎   | 684/1074 [1:00:24<50:09,  7.72s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2vop.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 184\n",
      "[w] arginine [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 685/1074 [1:00:25<35:55,  5.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yyk.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 442\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 48\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 686/1074 [1:00:25<25:45,  3.98s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d30.cif1'\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 188\n",
      "[w] arginine [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.245.] -- NH1/NH2 naming swapped\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 687/1074 [1:00:25<18:32,  2.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lqo.cif1'\n",
      "  1.A.CBV.2 0.122\n",
      "  1.B.CBV.2 0.122\n",
      "    total number of nucleotides: 20\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 688/1074 [1:00:26<13:21,  2.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5bzv.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 364\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 689/1074 [1:00:26<10:21,  1.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3wbm.cif1'\n",
      "    total number of nucleotides: 50\n",
      "    total number of amino acids: 345\n",
      "    total number of base pairs: 23\n",
      "    total number of helices: 1\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 53\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 690/1074 [1:00:27<07:56,  1.24s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1a9n.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 523\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 48\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 691/1074 [1:00:27<06:37,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1duh.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 64%|██████▍   | 692/1074 [1:00:27<04:56,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dto.cif1'\n",
      "  1.B.M7G.0 0.131\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 853\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▍   | 693/1074 [1:00:28<05:15,  1.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wfl.cif1'\n",
      "    total number of nucleotides: 107\n",
      "    total number of base pairs: 52\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 3\n",
      "    total number of stems: 7\n",
      "    total number of non-pairing interactions: 131\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 28\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▍   | 694/1074 [1:00:29<04:26,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1dfu.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 94\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▍   | 695/1074 [1:00:29<03:31,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3td0.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▍   | 696/1074 [1:00:29<02:48,  2.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2v6w.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▍   | 697/1074 [1:00:29<02:19,  2.71it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pcj.cif1'\n",
      "1..A.PSU.5. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 35\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▍   | 698/1074 [1:00:29<01:59,  3.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3akz.cif1'\n",
      "    total number of nucleotides: 300\n",
      "    total number of amino acids: 1856\n",
      "    total number of base pairs: 139\n",
      "    total number of multiplets: 21\n",
      "    total number of helices: 8\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 354\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 49\n",
      "                        consolidated into units: 29\n",
      "    total number of hairpin loops: 12\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 3\n",
      "    total number of ribose zippers: 4\n",
      "    total number of phosphate interactions: 73\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▌   | 699/1074 [1:00:32<07:02,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5c5w.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of non-pairing interactions: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▌   | 700/1074 [1:00:33<05:36,  1.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4m59.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 1370\n",
      "[w] arginine [1..A.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.400.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.565.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.590.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.400.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.565.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.590.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.336.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.400.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.420.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.565.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.590.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.742.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.752.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▌   | 701/1074 [1:00:34<05:58,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5a0v.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 905\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▌   | 702/1074 [1:00:35<05:32,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nh3.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 268\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 65%|██████▌   | 703/1074 [1:00:35<04:21,  1.42it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/472d.cif1'\n",
      "  1.A.U.5 0.120\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 17\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 704/1074 [1:00:35<03:39,  1.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dme.cif1'\n",
      "    total number of nucleotides: 103\n",
      "    total number of base pairs: 54\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 123\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 705/1074 [1:00:36<03:25,  1.79it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2g8i.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 134\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 706/1074 [1:00:36<02:51,  2.15it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4hor.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 482\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 707/1074 [1:00:36<02:51,  2.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k64.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 400\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 708/1074 [1:00:37<02:38,  2.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3qrp.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 214\n",
      "[w] arginine [1..A.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 7\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 709/1074 [1:00:37<02:24,  2.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2p7e.cif1'\n",
      "    total number of nucleotides: 61\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 9\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 710/1074 [1:00:37<02:15,  2.69it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4frg.cif1'\n",
      "    total number of nucleotides: 168\n",
      "    total number of base pairs: 75\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 8\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 200\n",
      "    total number of atom-base capping interactions: 16\n",
      "    total number of splayed-apart dinucleotides: 31\n",
      "                        consolidated into units: 25\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 6\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 24\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▌   | 711/1074 [1:00:38<02:06,  2.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1si3.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 120\n",
      "[w] arginine [1..A.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.253.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▋   | 712/1074 [1:00:38<01:51,  3.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2qk9.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 153\n",
      "[w] arginine [1..A.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.157.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 38\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▋   | 713/1074 [1:00:38<01:43,  3.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1h3e.cif1'\n",
      "1..B.PSU.35. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..B.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 84\n",
      "    total number of amino acids: 428\n",
      "    total number of base pairs: 39\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 97\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 66%|██████▋   | 714/1074 [1:00:39<02:40,  2.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hnj.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 54\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 715/1074 [1:00:39<02:13,  2.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wzg.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 534\n",
      "[w] arginine [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.619.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 716/1074 [1:00:40<02:34,  2.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6f4h.cif1'\n",
      "    total number of nucleotides: 66\n",
      "    total number of amino acids: 277\n",
      "[w] arginine [1..A.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.67.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 51\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 717/1074 [1:00:40<02:15,  2.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3avu.cif1'\n",
      "  1.T.U.2105 0.122\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 1203\n",
      "[w] arginine [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.547.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.564.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.573.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.612.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.618.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.710.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.711.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.752.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.858.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.937.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.997.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1088.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1197.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1249.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 7\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 718/1074 [1:00:41<04:12,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1lnt.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 13\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 719/1074 [1:00:42<03:22,  1.75it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qln.cif1'\n",
      "    total number of nucleotides: 39\n",
      "    total number of amino acids: 862\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 33\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 720/1074 [1:00:43<04:39,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d12.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 203\n",
      "[w] arginine [1..B.ARG.472.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.540.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.472.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.540.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 721/1074 [1:00:43<03:40,  1.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zdj.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 722/1074 [1:00:43<02:57,  1.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4oav.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 1347\n",
      "[w] arginine [1..B.ARG.257.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.257.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.257.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.257.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 723/1074 [1:00:45<04:21,  1.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3i62.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 509\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 67%|██████▋   | 724/1074 [1:00:45<03:50,  1.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2py9.cif1'\n",
      "  1.E.U.6 0.127\n",
      "  1.F.U.6 0.145\n",
      "  1.F.U.12 0.132\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 279\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 725/1074 [1:00:45<03:02,  1.91it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3skl.cif1'\n",
      "  1.A.CCC.87 0.127\n",
      "  1.B.CCC.87 0.127\n",
      "    total number of nucleotides: 134\n",
      "    total number of base pairs: 64\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 157\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 16\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of phosphate interactions: 29\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 726/1074 [1:00:46<02:32,  2.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1uvm.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 1992\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 727/1074 [1:00:48<05:04,  1.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2dlc.cif1'\n",
      "1..Y.PSU.535. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..Y.PSU.555. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 70\n",
      "    total number of amino acids: 340\n",
      "[w] arginine [1..X.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.338.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 27\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 69\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 728/1074 [1:00:48<04:10,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1m8v.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of amino acids: 995\n",
      "    total number of non-pairing interactions: 47\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 16\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 20\n",
      "    total number of phosphate interactions: 33\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 729/1074 [1:00:48<03:45,  1.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ed2.cif1'\n",
      "    total number of nucleotides: 92\n",
      "    total number of amino acids: 770\n",
      "[w] arginine [1..A.ARG.494.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.494.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..B.8AZ.13. missing planar atom [ C8 ]\n",
      "\t[i] 1..B.8AZ.13.: replace missing base ring atom [ C8 ] with [N8] (d=0.07)\n",
      "[w] Residue 1..E.8AZ.13. missing planar atom [ C8 ]\n",
      "\t[i] 1..E.8AZ.13.: replace missing base ring atom [ C8 ] with [N8] (d=0.06)\n",
      "[w] Residue [1..A.ARG.494.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.494.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.674.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 44\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 88\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of phosphate interactions: 28\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 730/1074 [1:00:49<03:51,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2rfk.cif1'\n",
      "    total number of nucleotides: 61\n",
      "    total number of amino acids: 461\n",
      "    total number of base pairs: 28\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 58\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 5\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 12\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 731/1074 [1:00:50<03:34,  1.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2az0.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 141\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 38\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 732/1074 [1:00:50<02:59,  1.91it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4kxt.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 806\n",
      "[w] arginine [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 733/1074 [1:00:51<03:30,  1.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1g1x.cif1'\n",
      "    total number of nucleotides: 169\n",
      "    total number of amino acids: 466\n",
      "[w] arginine [1..A.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.71.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 72\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 6\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 204\n",
      "    total number of atom-base capping interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 52\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 734/1074 [1:00:51<03:33,  1.59it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3lrr.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 242\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 68%|██████▊   | 735/1074 [1:00:52<03:03,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1fjg.cif1'\n",
      "    total number of nucleotides: 1513\n",
      "    total number of amino acids: 2404\n",
      "[w] arginine [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 778\n",
      "    total number of multiplets: 126\n",
      "    total number of helices: 44\n",
      "    total number of stems: 94\n",
      "    total number of isolated WC/wobble pairs: 23\n",
      "    total number of non-pairing interactions: 2226\n",
      "    total number of atom-base capping interactions: 98\n",
      "    total number of splayed-apart dinucleotides: 179\n",
      "                        consolidated into units: 111\n",
      "    total number of hairpin loops: 33\n",
      "    total number of bulges: 24\n",
      "    total number of internal loops: 36\n",
      "    total number of junctions: 19\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "    total number of A-minor (types I and II) motifs: 39\n",
      "    total number of eXtended A-minor (type X) motifs: 34\n",
      "    total number of U-turns: 22\n",
      "    total number of ribose zippers: 19\n",
      "    total number of kink turns: 5\n",
      "    total number of phosphate interactions: 600\n",
      "\n",
      "Time used: 00:00:00:12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▊   | 736/1074 [1:01:04<21:54,  3.89s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ytt.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 75\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▊   | 737/1074 [1:01:04<15:35,  2.78s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4n0t.cif1'\n",
      "    total number of nucleotides: 65\n",
      "    total number of amino acids: 363\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 58\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 6\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▊   | 738/1074 [1:01:04<11:49,  2.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rkf.cif1'\n",
      "    total number of nucleotides: 272\n",
      "    total number of base pairs: 148\n",
      "    total number of multiplets: 23\n",
      "    total number of helices: 16\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 12\n",
      "    total number of non-pairing interactions: 336\n",
      "    total number of atom-base capping interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 28\n",
      "                        consolidated into units: 20\n",
      "    total number of hairpin loops: 12\n",
      "    total number of junctions: 8\n",
      "    total number of kissing loops: 4\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 12\n",
      "    total number of phosphate interactions: 60\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 739/1074 [1:01:05<09:40,  1.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ed5.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 337\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ASP.34.] -- missing planar atom ' OD2'\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.153.] -- missing planar atom ' NH1'\n",
      "[w] Residue [1..B.GLU.35.] -- missing planar atom ' OE2'\n",
      "[w] Residue [1..B.ARG.37.] -- missing planar atom ' NH2'\n",
      "[w] Residue [1..B.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.GLU.162.] -- missing planar atom ' OE2'\n",
      "[w] Residue [1..B.ASN.186.] -- missing planar atom ' OD1'\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 740/1074 [1:01:05<07:15,  1.30s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2c4q.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 387\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 27\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 741/1074 [1:01:06<05:30,  1.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ibk.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 13\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of G-tetrads: 3\n",
      "    total number of G4-helices: 1\n",
      "    total number of G4-stems: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 742/1074 [1:01:06<04:07,  1.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/418d.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 743/1074 [1:01:06<03:11,  1.73it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2jlw.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 899\n",
      "[w] arginine [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.241.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.387.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.427.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.539.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.617.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.241.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.387.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.427.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.539.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.617.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.617.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 744/1074 [1:01:07<03:19,  1.65it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ahu.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 198\n",
      "[w] arginine [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.32.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 745/1074 [1:01:07<02:45,  1.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/246d.cif1'\r\n",
      "    total number of nucleotides: 16\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 18\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 69%|██████▉   | 746/1074 [1:01:07<02:11,  2.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1m8w.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 681\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|██████▉   | 747/1074 [1:01:08<02:29,  2.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nia.cif1'\n",
      "    total number of nucleotides: 159\n",
      "    total number of amino acids: 2160\n",
      "    total number of base pairs: 54\n",
      "    total number of helices: 6\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 145\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 24\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|██████▉   | 748/1074 [1:01:09<04:33,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1feu.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of amino acids: 374\n",
      "    total number of base pairs: 35\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 92\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|██████▉   | 749/1074 [1:01:10<03:44,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2ez6.cif1'\n",
      "    total number of nucleotides: 56\n",
      "    total number of amino acids: 436\n",
      "    total number of base pairs: 26\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 65\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|██████▉   | 750/1074 [1:01:10<03:16,  1.65it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5bz1.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 371\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|██████▉   | 751/1074 [1:01:11<03:05,  1.74it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6e8t.cif1'\n",
      "    total number of nucleotides: 143\n",
      "    total number of base pairs: 88\n",
      "    total number of multiplets: 13\n",
      "    total number of helices: 8\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 149\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of G-tetrads: 8\n",
      "    total number of G4-helices: 4\n",
      "    total number of G4-stems: 4\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|███████   | 752/1074 [1:01:11<03:18,  1.62it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2qwy.cif1'\n",
      "    total number of nucleotides: 159\n",
      "    total number of amino acids: 3\n",
      "    total number of base pairs: 92\n",
      "    total number of multiplets: 31\n",
      "    total number of helices: 2\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 194\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 12\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 6\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 9\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 17\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|███████   | 753/1074 [1:01:14<05:48,  1.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2oe8.cif1'\n",
      "    total number of nucleotides: 33\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|███████   | 754/1074 [1:01:14<04:49,  1.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4g7o.cif1'\n",
      "    total number of nucleotides: 84\n",
      "    total number of amino acids: 6973\n",
      "[w] arginine [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.331.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.353.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.614.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.626.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.775.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.808.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1031.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.360.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.414.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.434.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.441.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.486.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.586.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.587.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.867.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.872.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1406.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.230.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.331.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.409.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.614.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.619.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.626.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.808.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.1031.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.220.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.224.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.312.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.315.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.414.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.441.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.486.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.586.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.587.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.832.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.867.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.872.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.1087.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.1108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.1130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.1133.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.1310.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.1406.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.222.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.232.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.372.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 69\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 8\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 26\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|███████   | 755/1074 [1:01:24<19:13,  3.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2y8y.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 201\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|███████   | 756/1074 [1:01:24<13:54,  2.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4p97.cif1'\n",
      "    total number of nucleotides: 54\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 53\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of bulges: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 70%|███████   | 757/1074 [1:01:25<10:15,  1.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nkb.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 83\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 11\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 758/1074 [1:01:25<07:31,  1.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q0l.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 681\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 759/1074 [1:01:25<06:09,  1.17s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e59.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 760/1074 [1:01:26<04:40,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5eeu.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 1543\n",
      "[w] arginine [1..F.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..R.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.58.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 26\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 761/1074 [1:01:27<05:10,  1.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6ifn.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 2281\n",
      "[w] arginine [1..A.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.657.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.657.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.171.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.302.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 21\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 28\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 762/1074 [1:01:29<06:29,  1.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6fql.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 390\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 763/1074 [1:01:29<05:08,  1.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2vpl.cif1'\n",
      "    total number of nucleotides: 94\n",
      "    total number of amino acids: 274\n",
      "    total number of base pairs: 40\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 108\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 8\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 3\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 30\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 764/1074 [1:01:30<04:46,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6h9h.cif1'\n",
      "  1.D.23G.22 0.134\n",
      "  1.E.23G.22 0.128\n",
      "    total number of nucleotides: 47\n",
      "    total number of amino acids: 500\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 30\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████   | 765/1074 [1:01:31<04:35,  1.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1e7k.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 248\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████▏  | 766/1074 [1:01:31<04:00,  1.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3v6y.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 393\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 71%|███████▏  | 767/1074 [1:01:32<03:15,  1.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4l81.cif1'\n",
      "    total number of nucleotides: 97\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 48\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 3\n",
      "    total number of stems: 8\n",
      "    total number of non-pairing interactions: 116\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 4\n",
      "    total number of U-turns: 3\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 768/1074 [1:01:32<02:38,  1.93it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ifd.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of amino acids: 3373\n",
      "[w] arginine [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.233.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.233.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.262.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.227.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.206.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.232.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.294.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.370.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.397.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.448.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.493.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.535.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.649.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.675.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.753.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.773.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.806.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.830.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.848.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.886.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.908.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.548.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.600.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 769/1074 [1:01:34<05:31,  1.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pqv.cif1'\n",
      "    total number of nucleotides: 68\n",
      "    total number of base pairs: 26\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 76\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 770/1074 [1:01:35<04:36,  1.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2bx2.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 500\n",
      "[w] arginine [1..L.ARG.-1.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.368.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.371.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.456.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.-1.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.249.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.364.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.368.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.371.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.456.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.470.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 771/1074 [1:01:35<03:52,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xpa.cif1'\n",
      "  1.D.U.12 0.122\n",
      "    total number of nucleotides: 39\n",
      "    total number of amino acids: 679\n",
      "[w] arginine [1..A.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.324.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.324.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 772/1074 [1:01:36<04:01,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wtj.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 555\n",
      "[w] arginine [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 773/1074 [1:01:37<04:07,  1.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1uvl.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 1992\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 774/1074 [1:01:39<06:25,  1.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3fht.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 779\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 775/1074 [1:01:40<05:38,  1.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3h5x.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 470\n",
      "[w] arginine [1..A.ARG.393.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.393.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 776/1074 [1:01:41<04:49,  1.03it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f5f.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of amino acids: 604\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.280.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 30\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 85\n",
      "    total number of atom-base capping interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 32\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 777/1074 [1:01:41<04:14,  1.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2atw.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 448\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 72%|███████▏  | 778/1074 [1:01:42<03:25,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/402d.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of internal loops: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 779/1074 [1:01:42<02:42,  1.81it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5d8h.cif1'\n",
      "    total number of nucleotides: 74\n",
      "    total number of amino acids: 378\n",
      "[w] arginine [1..C.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.61.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 99\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 3\n",
      "    total number of ribose zippers: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 780/1074 [1:01:43<03:08,  1.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4m6d.cif1'\n",
      "    total number of nucleotides: 250\n",
      "    total number of amino acids: 774\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.128.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 115\n",
      "    total number of multiplets: 18\n",
      "    total number of helices: 10\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 21\n",
      "    total number of non-pairing interactions: 275\n",
      "    total number of atom-base capping interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 39\n",
      "                        consolidated into units: 33\n",
      "    total number of hairpin loops: 12\n",
      "    total number of bulges: 6\n",
      "    total number of internal loops: 13\n",
      "    total number of non-loop single-stranded segments: 22\n",
      "    total number of eXtended A-minor (type X) motifs: 6\n",
      "    total number of U-turns: 5\n",
      "    total number of phosphate interactions: 47\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 781/1074 [1:01:44<04:35,  1.07it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mxh.cif1'\n",
      "    total number of nucleotides: 93\n",
      "    total number of amino acids: 87\n",
      "[w] arginine [1..P.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.36.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 47\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 2\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 114\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 3\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 782/1074 [1:01:45<04:46,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d8a.cif1'\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 1516\n",
      "[w] arginine [1..A.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.201.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.285.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.201.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.285.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.421.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.541.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.606.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.678.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.285.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.322.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.460.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.541.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.678.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 34\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "    total number of bulges: 2\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 783/1074 [1:01:47<06:12,  1.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5gjb.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 440\n",
      "[w] arginine [1..A.ARG.559.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.559.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 784/1074 [1:01:48<04:53,  1.01s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ovb.cif1'\n",
      "    total number of nucleotides: 71\n",
      "    total number of amino acids: 878\n",
      "[w] arginine [1..B.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.32.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 26\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 84\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 785/1074 [1:01:48<04:28,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ks8.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 492\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 42\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 786/1074 [1:01:49<03:33,  1.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5b2t.cif1'\n",
      "    total number of nucleotides: 117\n",
      "    total number of amino acids: 1326\n",
      "[w] arginine [1..B.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.403.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.461.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.403.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.437.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.461.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.467.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.635.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.661.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.859.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.967.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.976.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1060.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1084.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1279.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.1337.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 49\n",
      "    total number of helices: 5\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 120\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 50\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 787/1074 [1:01:51<05:37,  1.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yfg.cif1'\n",
      "  1.A.H2U.16 0.190\n",
      "  1.A.H2U.47 0.187\n",
      "1..A.RIA.64. has unusual GLYCOSIDIC bond!\n",
      "    total number of nucleotides: 75\n",
      "    total number of base pairs: 36\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 82\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 788/1074 [1:01:51<04:18,  1.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v4f.cif1'\n",
      "    total number of nucleotides: 49\n",
      "    total number of amino acids: 3132\n",
      "[w] arginine [1..AC.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AN.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BC.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BN.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.266.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.66.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 22\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 73%|███████▎  | 789/1074 [1:01:54<07:28,  1.57s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qf6.cif1'\n",
      "  1.B.H2U.16 0.158\n",
      "  1.B.H2U.17 0.206\n",
      "  1.B.H2U.20 0.134\n",
      "1..B.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 77\n",
      "    total number of amino acids: 641\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▎  | 790/1074 [1:01:55<06:22,  1.35s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1f27.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of base pairs: 14\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▎  | 791/1074 [1:01:55<04:47,  1.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4tuw.cif1'\n",
      "    total number of nucleotides: 54\n",
      "    total number of amino acids: 141\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 54\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▎  | 792/1074 [1:01:56<04:03,  1.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3adl.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 76\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 793/1074 [1:01:56<03:20,  1.40it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2db3.cif1'\n",
      "    total number of nucleotides: 29\n",
      "    total number of amino acids: 1680\n",
      "    total number of non-pairing interactions: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 794/1074 [1:01:58<04:29,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1s77.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of amino acids: 828\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 12\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 795/1074 [1:01:59<04:28,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5nfv.cif1'\n",
      "  1.B.DU.-19 0.135\n",
      "    total number of nucleotides: 101\n",
      "    total number of amino acids: 1258\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.583.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.740.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.746.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.921.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1166.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.1244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.583.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.740.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.746.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.921.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1166.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1244.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 46\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 105\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 48\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 796/1074 [1:02:01<06:27,  1.39s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5uz6.cif1'\n",
      "    total number of nucleotides: 96\n",
      "    total number of base pairs: 42\n",
      "    total number of helices: 3\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 99\n",
      "    total number of atom-base capping interactions: 15\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 3\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 797/1074 [1:02:01<04:45,  1.03s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qm6.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 342\n",
      "[w] arginine [1..A.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.68.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 798/1074 [1:02:02<03:42,  1.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4u34.cif1'\r\n",
      "    total number of nucleotides: 14\r\n",
      "    total number of base pairs: 7\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 14\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 799/1074 [1:02:02<02:49,  1.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2uxc.cif1'\n",
      "    total number of nucleotides: 1525\n",
      "    total number of amino acids: 2404\n",
      "[w] arginine [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.118.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 778\n",
      "    total number of multiplets: 122\n",
      "    total number of helices: 44\n",
      "    total number of stems: 95\n",
      "    total number of isolated WC/wobble pairs: 25\n",
      "    total number of non-pairing interactions: 2224\n",
      "    total number of atom-base capping interactions: 112\n",
      "    total number of splayed-apart dinucleotides: 183\n",
      "                        consolidated into units: 113\n",
      "    total number of hairpin loops: 33\n",
      "    total number of bulges: 26\n",
      "    total number of internal loops: 36\n",
      "    total number of junctions: 19\n",
      "    total number of non-loop single-stranded segments: 14\n",
      "    total number of A-minor (types I and II) motifs: 41\n",
      "    total number of eXtended A-minor (type X) motifs: 33\n",
      "    total number of U-turns: 22\n",
      "    total number of ribose zippers: 22\n",
      "    total number of kink turns: 3\n",
      "    total number of phosphate interactions: 595\n",
      "\n",
      "Time used: 00:00:00:12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 74%|███████▍  | 800/1074 [1:02:14<19:08,  4.19s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zq0.cif1'\n",
      "  1.B.MUM.747 0.158\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 454\n",
      "[w] arginine [1..A.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.402.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.402.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▍  | 801/1074 [1:02:15<13:59,  3.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4h8k.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 276\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.55.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 28\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▍  | 802/1074 [1:02:15<10:20,  2.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1l9a.cif1'\n",
      "    total number of nucleotides: 126\n",
      "    total number of amino acids: 87\n",
      "    total number of base pairs: 63\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 133\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 7\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▍  | 803/1074 [1:02:16<08:05,  1.79s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3g9c.cif1'\n",
      "    total number of nucleotides: 606\n",
      "    total number of amino acids: 360\n",
      "    total number of base pairs: 272\n",
      "    total number of multiplets: 33\n",
      "    total number of helices: 20\n",
      "    total number of stems: 36\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 722\n",
      "    total number of atom-base capping interactions: 24\n",
      "    total number of splayed-apart dinucleotides: 76\n",
      "                        consolidated into units: 54\n",
      "    total number of hairpin loops: 20\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 15\n",
      "    total number of A-minor (types I and II) motifs: 21\n",
      "    total number of eXtended A-minor (type X) motifs: 9\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 16\n",
      "    total number of phosphate interactions: 67\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▍  | 804/1074 [1:02:18<08:28,  1.88s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4jrd.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of non-pairing interactions: 21\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▍  | 805/1074 [1:02:18<06:24,  1.43s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5btm.cif1'\n",
      "    total number of nucleotides: 86\n",
      "    total number of base pairs: 38\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 88\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▌  | 806/1074 [1:02:19<04:56,  1.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1zbi.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 271\n",
      "[w] arginine [1..B.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.123.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▌  | 807/1074 [1:02:19<03:54,  1.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6dcl.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of amino acids: 353\n",
      "[w] arginine [1..B.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.178.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▌  | 808/1074 [1:02:20<04:02,  1.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1uvj.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 1992\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▌  | 809/1074 [1:02:22<05:17,  1.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3t5n.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 282\n",
      "[w] arginine [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.323.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 75%|███████▌  | 810/1074 [1:02:22<04:06,  1.07it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k62.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 400\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 811/1074 [1:02:23<03:19,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k61.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 393\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 812/1074 [1:02:23<02:49,  1.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dno.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 165\n",
      "[w] arginine [1..A.ARG.381.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.483.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.381.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ASN.439.] -- missing planar atom ' ND2'\n",
      "[w] Residue [1..A.ARG.483.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 813/1074 [1:02:23<02:12,  1.97it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1et4.cif1'\n",
      "    total number of nucleotides: 175\n",
      "    total number of base pairs: 95\n",
      "    total number of multiplets: 21\n",
      "    total number of helices: 3\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 15\n",
      "    total number of non-pairing interactions: 204\n",
      "    total number of atom-base capping interactions: 16\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 10\n",
      "    total number of bulges: 5\n",
      "    total number of non-loop single-stranded segments: 20\n",
      "    total number of A-minor (types I and II) motifs: 5\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 814/1074 [1:02:25<03:17,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k31.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 815/1074 [1:02:25<02:34,  1.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ice.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 2432\n",
      "[w] arginine [1..A.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.305.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.160.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.353.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.144.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 816/1074 [1:02:27<04:29,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1b23.cif1'\n",
      "  1.R.H2U.20 0.222\n",
      "  1.R.H2U.21 0.217\n",
      "1..R.PSU.32. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..R.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..R.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 75\n",
      "    total number of amino acids: 406\n",
      "    total number of base pairs: 34\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 84\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 817/1074 [1:02:28<04:21,  1.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/377d.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▌  | 818/1074 [1:02:28<03:15,  1.31it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wte.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 535\n",
      "[w] arginine [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.278.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▋  | 819/1074 [1:02:28<02:52,  1.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wti.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 553\n",
      "[w] arginine [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.380.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▋  | 820/1074 [1:02:29<02:37,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5k8h.cif1'\r\n",
      "    total number of nucleotides: 2\r\n",
      "    total number of non-pairing interactions: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 76%|███████▋  | 821/1074 [1:02:29<01:58,  2.13it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2zue.cif1'\n",
      "    total number of nucleotides: 76\n",
      "    total number of amino acids: 628\n",
      "[w] arginine [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 95\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 822/1074 [1:02:30<02:22,  1.76it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e48.cif1'\n",
      "    total number of nucleotides: 60\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 7\n",
      "    total number of non-pairing interactions: 56\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of internal loops: 6\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 823/1074 [1:02:30<01:53,  2.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4k4u.cif1'\n",
      "    total number of nucleotides: 72\n",
      "    total number of amino acids: 924\n",
      "    total number of base pairs: 32\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 66\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 824/1074 [1:02:31<02:14,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5d0a.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 1509\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 825/1074 [1:02:32<02:53,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6qzp.cif1'\n",
      "  1.S2.HYG.2039 0.196\n",
      "[i] taken as a pyrimidine ... 1.S2.HYG.2039\n",
      "1..L5.PSU.1582. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.1677. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.1683. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.B8H.1860. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.2508. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.3715. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.3729. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.B8H.3762. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.3764. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4293. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.B8H.4296. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4403. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4442. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4450. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4500. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4531. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4628. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L5.PSU.4636. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.PSU.119. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.PSU.612. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.PSU.822. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.PSU.823. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.PSU.1081. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.PSU.1243. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..S2.B8N.1248. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 5865\n",
      "    total number of amino acids: 11713\n",
      "[w] arginine [1..LA.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LA.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LA.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LA.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LA.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LB.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LB.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LB.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..LB.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..S2.HYG.2039. missing planar atom [ N3 ]\n",
      "[w] Residue 1..S2.HYG.2039. missing planar atom [ N1 ]\n",
      "[w] Residue [1..LA.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LA.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LA.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LA.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LA.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LB.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LB.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LB.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LB.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LB.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LB.ARG.378.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.205.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.311.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LC.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LD.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LD.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LD.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LD.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LD.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LD.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LE.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LE.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LE.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LF.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LF.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LF.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LF.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LF.ARG.157.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LF.ARG.245.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LG.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LG.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LG.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LG.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LG.ARG.234.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LH.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LH.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LH.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LI.ARG.210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LJ.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LJ.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LJ.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LL.ARG.158.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LM.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LM.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LM.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LM.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LM.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LN.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LO.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LO.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LO.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LO.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LO.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LO.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LP.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LP.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LP.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LP.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LQ.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LQ.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LQ.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LQ.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LQ.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LQ.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LR.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LS.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LS.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LS.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LS.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LS.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LS.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LT.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LT.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LT.ARG.130.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LT.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LV.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LV.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LV.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LV.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LV.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LW.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LW.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LX.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LX.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LY.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LY.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LZ.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..LZ.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..La.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..La.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..La.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..La.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..La.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lb.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lb.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lb.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lb.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lc.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ld.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Le.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Le.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Le.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Le.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Le.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lf.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lf.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lg.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lh.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lh.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lh.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Li.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Li.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Li.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Li.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lj.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lj.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lj.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lj.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lj.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lk.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lk.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ll.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ll.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ll.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lm.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lm.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lm.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ln.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ln.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lo.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lo.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lo.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lp.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lp.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lp.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lp.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lr.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lr.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lr.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lr.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lr.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lr.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Lz.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SA.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SA.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SA.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SA.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SB.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SD.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SD.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SD.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SE.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SF.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SF.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SF.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SF.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SF.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SH.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SH.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SH.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SH.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SH.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SH.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SI.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SI.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SI.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SI.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SK.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SP.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SP.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SQ.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SQ.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SQ.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SQ.ARG.126.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SQ.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SQ.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SR.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SR.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SS.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SS.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SS.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SS.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SS.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SS.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..ST.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..ST.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..ST.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SU.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SU.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SV.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SX.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SX.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SX.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SX.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SX.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sa.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sa.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sa.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sa.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sa.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sa.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sc.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sc.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sc.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sg.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sg.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Sg.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.123.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.166.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SC.ARG.227.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SG.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SG.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SG.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SG.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SG.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SG.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SJ.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SN.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SN.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SN.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SN.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SN.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SN.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SO.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SO.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SO.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SO.ARG.150.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SW.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SW.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SY.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SY.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SY.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..SZ.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Se.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Se.ARG.31.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 2568\n",
      "    total number of multiplets: 333\n",
      "    total number of helices: 159\n",
      "    total number of stems: 344\n",
      "    total number of isolated WC/wobble pairs: 147\n",
      "    total number of non-pairing interactions: 7465\n",
      "    total number of atom-base capping interactions: 329\n",
      "    total number of splayed-apart dinucleotides: 940\n",
      "                        consolidated into units: 556\n",
      "    total number of hairpin loops: 117\n",
      "    total number of bulges: 87\n",
      "    total number of internal loops: 182\n",
      "    total number of junctions: 63\n",
      "    total number of non-loop single-stranded segments: 65\n",
      "    total number of kissing loops: 6\n",
      "    total number of A-minor (types I and II) motifs: 94\n",
      "    total number of eXtended A-minor (type X) motifs: 96\n",
      "    total number of U-turns: 48\n",
      "    total number of ribose zippers: 56\n",
      "    total number of kink turns: 5\n",
      "    total number of phosphate interactions: 2296\n",
      "[w]  number-of-residues=450 -- in shortened form\n",
      "[w]  number-of-residues=450 -- in shortened form\n",
      "\n",
      "Time used: 00:00:01:26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 826/1074 [1:03:58<1:48:26, 26.24s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6az1.cif1'\n",
      "  1.1.U.1192 0.125\n",
      "  1.1.U.1419 0.123\n",
      "  1.4.U.60 0.123\n",
      "1..1.M1Y.1539. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1.C4J.1543. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 2001\n",
      "    total number of amino acids: 4810\n",
      "[w] arginine [1..A.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.210.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.131.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.167.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.218.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.253.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.213.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.186.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.191.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.160.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.29.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.5.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.173.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..b.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..d.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..e.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..g.ARG.304.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 865\n",
      "    total number of multiplets: 116\n",
      "    total number of helices: 60\n",
      "    total number of stems: 120\n",
      "    total number of isolated WC/wobble pairs: 49\n",
      "    total number of non-pairing interactions: 2577\n",
      "    total number of atom-base capping interactions: 112\n",
      "    total number of splayed-apart dinucleotides: 306\n",
      "                        consolidated into units: 169\n",
      "    total number of hairpin loops: 34\n",
      "    total number of bulges: 19\n",
      "    total number of internal loops: 59\n",
      "    total number of junctions: 21\n",
      "    total number of non-loop single-stranded segments: 61\n",
      "    total number of A-minor (types I and II) motifs: 32\n",
      "    total number of eXtended A-minor (type X) motifs: 43\n",
      "    total number of U-turns: 21\n",
      "    total number of ribose zippers: 18\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 676\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 827/1074 [1:04:15<1:37:37, 23.72s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rg5.cif1'\n",
      "    total number of nucleotides: 172\n",
      "    total number of base pairs: 77\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 203\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 8\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 828/1074 [1:04:16<1:08:34, 16.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e7a.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 536\n",
      "[w] arginine [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 829/1074 [1:04:17<48:54, 11.98s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6fq3.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 390\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 830/1074 [1:04:17<34:40,  8.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gd2.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 244\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 831/1074 [1:04:18<24:38,  6.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2hw8.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 228\n",
      "    total number of base pairs: 16\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 77%|███████▋  | 832/1074 [1:04:18<17:36,  4.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4z7k.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 413\n",
      "    total number of base pairs: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 9\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 833/1074 [1:04:18<12:48,  3.19s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cf2.cif1'\n",
      "    total number of nucleotides: 35\n",
      "    total number of amino acids: 283\n",
      "    total number of base pairs: 16\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 834/1074 [1:04:19<09:29,  2.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2oe5.cif1'\n",
      "    total number of nucleotides: 33\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 835/1074 [1:04:19<06:55,  1.74s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4n2q.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 198\n",
      "[w] arginine [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.196.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 836/1074 [1:04:19<05:11,  1.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2f8s.cif1'\n",
      "  1.D.U.18 0.144\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 1408\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 48\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 837/1074 [1:04:21<05:06,  1.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1jzv.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of base pairs: 18\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 838/1074 [1:04:21<03:46,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2g3s.cif1'\n",
      "    total number of nucleotides: 80\n",
      "    total number of base pairs: 40\n",
      "    total number of helices: 4\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 80\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 839/1074 [1:04:21<02:52,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4fen.cif1'\n",
      "    total number of nucleotides: 68\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 840/1074 [1:04:22<02:39,  1.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bx3.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 625\n",
      "[w] arginine [1..B.ARG.651.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.651.] -- NH1/NH2 naming swapped\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 841/1074 [1:04:22<02:22,  1.64it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ay2.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of amino acids: 628\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 842/1074 [1:04:23<02:20,  1.66it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2g92.cif1'\r\n",
      "1..A.NF2.4. has unusual GLYCOSIDIC bond!\r\n",
      "1..B.NF2.16. has unusual GLYCOSIDIC bond!\r\n",
      "    total number of nucleotides: 24\r\n",
      "[w] Residue 1..A.NF2.4. missing planar atom [ N3 ]\r\n",
      "[w] Residue 1..A.NF2.4. missing planar atom [ N1 ]\r\n",
      "\t[i] 1..A.NF2.4.: replace missing base ring atom [ N3 ] with [C3] (d=0.02)\r\n",
      "\t[i] 1..A.NF2.4.: replace missing base ring atom [ N1 ] with [C1] (d=0.11)\r\n",
      "[w] Residue 1..B.NF2.16. missing planar atom [ N3 ]\r\n",
      "[w] Residue 1..B.NF2.16. missing planar atom [ N1 ]\r\n",
      "\t[i] 1..B.NF2.16.: replace missing base ring atom [ N3 ] with [C3] (d=0.03)\r\n",
      "\t[i] 1..B.NF2.16.: replace missing base ring atom [ N1 ] with [C1] (d=0.14)\r\n",
      "    total number of base pairs: 10\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 3\r\n",
      "    total number of non-pairing interactions: 23\r\n",
      "    total number of internal loops: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 78%|███████▊  | 843/1074 [1:04:23<01:48,  2.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5nxt.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▊  | 844/1074 [1:04:23<01:48,  2.12it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1t0e.cif1'\n",
      "    total number of nucleotides: 35\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 35\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▊  | 845/1074 [1:04:24<01:29,  2.55it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1q2r.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 1496\n",
      "[w] arginine [1..A.ARG.289.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..A.9DG.387. missing planar atom [ N9 ]\n",
      "\t[i] 1..A.9DG.387.: replace missing base ring atom [ N9 ] with [C9] (d=0.07)\n",
      "[w] Residue 1..B.9DG.387. missing planar atom [ N9 ]\n",
      "\t[i] 1..B.9DG.387.: replace missing base ring atom [ N9 ] with [C9] (d=0.08)\n",
      "[w] Residue 1..C.9DG.387. missing planar atom [ N9 ]\n",
      "\t[i] 1..C.9DG.387.: replace missing base ring atom [ N9 ] with [C9] (d=0.08)\n",
      "[w] Residue 1..D.9DG.387. missing planar atom [ N9 ]\n",
      "\t[i] 1..D.9DG.387.: replace missing base ring atom [ N9 ] with [C9] (d=0.08)\n",
      "[w] Residue [1..A.ARG.289.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 846/1074 [1:04:25<02:22,  1.60it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cd6.cif1'\n",
      "1..0.PSU.2621. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 2879\n",
      "    total number of amino acids: 3776\n",
      "    total number of base pairs: 1474\n",
      "    total number of multiplets: 247\n",
      "    total number of helices: 85\n",
      "    total number of stems: 179\n",
      "    total number of isolated WC/wobble pairs: 57\n",
      "    total number of non-pairing interactions: 4097\n",
      "    total number of atom-base capping interactions: 225\n",
      "    total number of splayed-apart dinucleotides: 411\n",
      "                        consolidated into units: 259\n",
      "    total number of hairpin loops: 68\n",
      "    total number of bulges: 37\n",
      "    total number of internal loops: 67\n",
      "    total number of junctions: 36\n",
      "    total number of non-loop single-stranded segments: 34\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cd6.cif1', hairpin loops (#1/#3) have 2 interactions [#8, #9 ...]\n",
      "    total number of kissing loops: 5\n",
      "    total number of A-minor (types I and II) motifs: 97\n",
      "    total number of eXtended A-minor (type X) motifs: 57\n",
      "    total number of U-turns: 33\n",
      "    total number of ribose zippers: 44\n",
      "    total number of kink turns: 7\n",
      "    total number of phosphate interactions: 1157\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 847/1074 [1:04:56<36:47,  9.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2bbv.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 985\n",
      "[w] arginine [1..B.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.58.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 848/1074 [1:04:56<26:19,  6.99s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5b2p.cif1'\n",
      "    total number of nucleotides: 132\n",
      "    total number of amino acids: 1455\n",
      "[w] arginine [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.455.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.528.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.646.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.704.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.856.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.455.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.528.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.646.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.704.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.856.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.939.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1047.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1061.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1404.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1438.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1560.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1585.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 59\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 4\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 140\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "    total number of hairpin loops: 3\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 71\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 849/1074 [1:04:59<21:53,  5.84s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4j50.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 850/1074 [1:05:00<15:44,  4.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ho1.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 673\n",
      "[w] arginine [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.377.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.392.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 851/1074 [1:05:00<11:38,  3.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3cul.cif1'\n",
      "    total number of nucleotides: 184\n",
      "    total number of amino acids: 183\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 5\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 182\n",
      "    total number of atom-base capping interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 15\n",
      "    total number of hairpin loops: 6\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 5\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 852/1074 [1:05:01<08:51,  2.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5c9h.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 470\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 32\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 79%|███████▉  | 853/1074 [1:05:02<06:38,  1.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1sa9.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 41\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|███████▉  | 854/1074 [1:05:02<04:48,  1.31s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4zld.cif1'\n",
      "    total number of nucleotides: 17\n",
      "    total number of amino acids: 151\n",
      "[w] arginine [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.226.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|███████▉  | 855/1074 [1:05:02<03:34,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5m0h.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|███████▉  | 856/1074 [1:05:02<02:40,  1.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5zuu.cif1'\n",
      "  *1.B.EPE.501 0.258 (planarity=0.287)\n",
      "  1.B.EPE.501 0.258\n",
      "[i] taken as a pyrimidine ... 1.B.EPE.501\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 632\n",
      "[w] arginine [1..C.ARG.287.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.289.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..B.EPE.501. missing planar atom [ C4 ]\n",
      "[w] Residue 1..B.EPE.501. missing planar atom [ N3 ]\n",
      "\t[i] 1..B.EPE.501.: replace missing base ring atom [ C4 ] with [N4] (d=0.14)\n",
      "[w] Residue [1..C.ARG.287.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.289.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.287.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.300.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.387.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|███████▉  | 857/1074 [1:05:03<02:18,  1.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/422d.cif1'\r\n",
      "    total number of nucleotides: 24\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 2\r\n",
      "    total number of non-pairing interactions: 22\r\n",
      "    total number of internal loops: 1\r\n",
      "    total number of non-loop single-stranded segments: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|███████▉  | 858/1074 [1:05:03<01:45,  2.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3hsb.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 404\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.15.] -- NH1/NH2 naming swapped\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|███████▉  | 859/1074 [1:05:03<01:28,  2.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3s14.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 3561\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 19\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|████████  | 860/1074 [1:05:07<05:09,  1.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5dhc.cif1'\n",
      "    total number of nucleotides: 64\n",
      "    total number of base pairs: 32\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 68\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|████████  | 861/1074 [1:05:07<03:50,  1.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1dqh.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 18\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of bulges: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|████████  | 862/1074 [1:05:07<02:56,  1.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2zy6.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of base pairs: 12\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|████████  | 863/1074 [1:05:08<02:22,  1.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2quw.cif1'\n",
      "    total number of nucleotides: 138\n",
      "    total number of base pairs: 60\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 5\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 169\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "    total number of hairpin loops: 6\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of kissing loops: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 80%|████████  | 864/1074 [1:05:08<01:55,  1.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hr6.cif1'\n",
      "    total number of nucleotides: 136\n",
      "    total number of amino acids: 710\n",
      "[w] arginine [1..A.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.103.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 63\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 155\n",
      "    total number of atom-base capping interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 25\n",
      "                        consolidated into units: 17\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 38\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 865/1074 [1:05:09<02:05,  1.66it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1uvn.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of amino acids: 1992\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 3\n",
      "    total number of isolated WC/wobble pairs: 6\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 866/1074 [1:05:10<03:11,  1.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rer.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 368\n",
      "[w] arginine [1..C.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.16.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 867/1074 [1:05:10<02:29,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ws2.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 926\n",
      "[w] arginine [1..A.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.298.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.271.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.298.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 868/1074 [1:05:11<02:25,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nmr.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 175\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 869/1074 [1:05:11<01:57,  1.75it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1h4s.cif1'\n",
      "1..T.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 69\n",
      "    total number of amino acids: 948\n",
      "[w] arginine [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.284.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 78\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 870/1074 [1:05:12<02:18,  1.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3lqx.cif1'\n",
      "    total number of nucleotides: 49\n",
      "    total number of amino acids: 69\n",
      "    total number of base pairs: 22\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 52\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 871/1074 [1:05:13<01:50,  1.83it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2b3j.cif1'\n",
      "    total number of nucleotides: 54\n",
      "    total number of amino acids: 604\n",
      "[w] arginine [1..A.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.125.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 60\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████  | 872/1074 [1:05:13<01:59,  1.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6hu6.cif1'\n",
      "  1.F.U.205 0.130\n",
      "  1.F.U.208 0.122\n",
      "    total number of nucleotides: 57\n",
      "    total number of base pairs: 28\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 63\n",
      "    total number of atom-base capping interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████▏ | 873/1074 [1:05:13<01:35,  2.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3iab.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of amino acids: 255\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████▏ | 874/1074 [1:05:14<01:18,  2.54it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4l8h.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 244\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 18\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 81%|████████▏ | 875/1074 [1:05:14<01:09,  2.88it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r9w.cif1'\n",
      "    total number of nucleotides: 35\n",
      "    total number of amino acids: 302\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 876/1074 [1:05:14<01:09,  2.86it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3p6y.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 2195\n",
      "[w] arginine [1..A.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..F.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..J.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..M.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.66.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of non-pairing interactions: 9\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 7\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 877/1074 [1:05:16<02:08,  1.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1icg.cif1'\n",
      "    total number of nucleotides: 18\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 18\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 878/1074 [1:05:17<02:37,  1.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6du4.cif1'\n",
      "    total number of nucleotides: 29\n",
      "    total number of amino acids: 287\n",
      "[w] arginine [1..A.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.230.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.241.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.204.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.230.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.241.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.243.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 879/1074 [1:05:17<02:10,  1.50it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bnr.cif1'\n",
      "    total number of nucleotides: 89\n",
      "    total number of base pairs: 37\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 98\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 5\n",
      "    total number of internal loops: 8\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 880/1074 [1:05:18<02:02,  1.58it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3eqt.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 269\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 881/1074 [1:05:18<01:40,  1.92it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1j9h.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of base pairs: 22\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 34\n",
      "    total number of bulges: 4\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 882/1074 [1:05:18<01:22,  2.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1ser.cif1'\n",
      "  1.T.H2U.20A 0.231\n",
      "1..T.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 64\n",
      "    total number of amino acids: 793\n",
      "    total number of base pairs: 29\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 79\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 883/1074 [1:05:19<01:46,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3kfu.cif1'\n",
      "1..K.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..L.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..M.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..N.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 286\n",
      "    total number of amino acids: 3494\n",
      "[w] arginine [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.254.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 138\n",
      "    total number of multiplets: 14\n",
      "    total number of helices: 8\n",
      "    total number of stems: 16\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 339\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 34\n",
      "                        consolidated into units: 20\n",
      "    total number of hairpin loops: 8\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 12\n",
      "    total number of U-turns: 4\n",
      "    total number of phosphate interactions: 107\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 884/1074 [1:05:23<04:38,  1.47s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2o3x.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 885/1074 [1:05:23<03:25,  1.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6aay.cif1'\n",
      "    total number of nucleotides: 52\n",
      "    total number of amino acids: 1199\n",
      "[w] arginine [1..A.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.330.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.425.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.450.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.491.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.526.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.720.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.734.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.330.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.425.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.450.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.491.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.526.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.720.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.722.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.734.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.890.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.991.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1000.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1002.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1080.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1090.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 82%|████████▏ | 886/1074 [1:05:25<04:10,  1.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ii9.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 342\n",
      "[w] arginine [1..A.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.101.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 887/1074 [1:05:25<03:23,  1.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1i5l.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 1010\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 888/1074 [1:05:26<03:00,  1.03it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5en1.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 184\n",
      "[w] arginine [1..A.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 889/1074 [1:05:26<02:23,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ho4.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 179\n",
      "[w] arginine [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.89.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 890/1074 [1:05:27<01:57,  1.56it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wwx.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 84\n",
      "[w] arginine [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 891/1074 [1:05:27<01:46,  1.72it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2jlv.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 902\n",
      "[w] arginine [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.274.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.274.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.617.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.526.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.617.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 892/1074 [1:05:28<02:11,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pwd.cif1'\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 1934\n",
      "    total number of base pairs: 38\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 20\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 893/1074 [1:05:30<02:57,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wan.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of amino acids: 487\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 894/1074 [1:05:30<02:23,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1fix.cif1'\r\n",
      "    total number of nucleotides: 20\r\n",
      "    total number of base pairs: 10\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 20\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 895/1074 [1:05:30<01:47,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4u8t.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 929\n",
      "[w] arginine [1..A.ARG.267.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.296.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.296.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.267.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.296.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.296.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.296.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.272.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.277.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of splayed-apart dinucleotides: 24\n",
      "                        consolidated into units: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 83%|████████▎ | 896/1074 [1:05:31<02:06,  1.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xpg.cif1'\n",
      "    total number of nucleotides: 37\n",
      "    total number of amino acids: 684\n",
      "[w] arginine [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.95.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.350.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.532.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.661.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 15\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▎ | 897/1074 [1:05:32<02:09,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5i9d.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 360\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▎ | 898/1074 [1:05:32<01:56,  1.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2q66.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 519\n",
      "[w] arginine [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.510.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.510.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▎ | 899/1074 [1:05:33<01:47,  1.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4c8y.cif1'\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 468\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 900/1074 [1:05:34<01:57,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ytx.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 74\n",
      "    total number of non-pairing interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 901/1074 [1:05:34<01:31,  1.89it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6eri.cif1'\n",
      "    total number of nucleotides: 4401\n",
      "    total number of amino acids: 6348\n",
      "[w] arginine [1..AC.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AC.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.256.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AC.ARG.269.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.214.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AD.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.105.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.158.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AE.ARG.221.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AF.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.93.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.189.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.203.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.242.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AK.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.139.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.238.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.247.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AL.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AM.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AN.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AO.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.116.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.144.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.161.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AP.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AQ.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AR.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.176.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AS.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.147.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AU.ARG.160.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AX.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AY.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AZ.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AZ.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aa.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ab.ARG.39.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.117.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ac.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ad.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Ae.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Af.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aw.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aw.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Aw.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.178.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BB.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BC.ARG.183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.199.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BE.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BE.ARG.299.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BG.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BG.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BH.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BH.ARG.25.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BH.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BJ.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BK.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BK.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BK.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BK.ARG.136.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.12.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BL.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BM.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BM.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BM.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BN.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BP.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BR.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BR.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BU.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BV.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BV.ARG.140.] -- NH1/NH2 naming swapped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    total number of base pairs: 2173\n",
      "    total number of multiplets: 326\n",
      "    total number of helices: 126\n",
      "    total number of stems: 286\n",
      "    total number of isolated WC/wobble pairs: 91\n",
      "    total number of non-pairing interactions: 6031\n",
      "    total number of atom-base capping interactions: 252\n",
      "    total number of splayed-apart dinucleotides: 582\n",
      "                        consolidated into units: 364\n",
      "    total number of hairpin loops: 106\n",
      "    total number of bulges: 56\n",
      "    total number of internal loops: 129\n",
      "    total number of junctions: 59\n",
      "    total number of non-loop single-stranded segments: 24\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6eri.cif1', hairpin loops (#1/#3) have 2 interactions [#-2, #9 ...]\n",
      "    total number of kissing loops: 7\n",
      "    total number of A-minor (types I and II) motifs: 133\n",
      "    total number of eXtended A-minor (type X) motifs: 92\n",
      "    total number of U-turns: 46\n",
      "    total number of ribose zippers: 53\n",
      "    total number of kink turns: 9\n",
      "    total number of phosphate interactions: 1536\n",
      "\n",
      "Time used: 00:00:00:59\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 902/1074 [1:06:33<51:34, 17.99s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6hau.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 546\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 903/1074 [1:06:34<36:40, 12.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bt7.cif1'\n",
      "  1.C.5MU.54 0.154\n",
      "  1.D.5MU.54 0.165\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 738\n",
      "[w] arginine [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.234.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.234.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 39\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 904/1074 [1:06:34<26:11,  9.24s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3sj2.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 905/1074 [1:06:34<18:21,  6.52s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2a0p.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 18\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 906/1074 [1:06:35<12:57,  4.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ud5.cif1'\n",
      "    total number of nucleotides: 142\n",
      "    total number of amino acids: 172\n",
      "[w] arginine [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.71.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.71.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 62\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 148\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 11\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 37\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 84%|████████▍ | 907/1074 [1:06:35<09:22,  3.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1ooa.cif1'\n",
      "    total number of nucleotides: 58\n",
      "    total number of amino acids: 626\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.228.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.228.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 62\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 15\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▍ | 908/1074 [1:06:36<07:16,  2.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4x4u.cif1'\n",
      "    total number of nucleotides: 79\n",
      "    total number of amino acids: 1745\n",
      "[w] arginine [1..C.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.373.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.373.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 31\n",
      "    total number of helices: 4\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 11\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 27\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▍ | 909/1074 [1:06:38<06:38,  2.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5guh.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 759\n",
      "[w] arginine [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.424.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.540.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.606.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.628.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.725.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.785.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.185.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.223.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.424.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.540.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.606.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.628.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.725.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.785.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.790.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.792.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▍ | 910/1074 [1:06:39<05:18,  1.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1hys.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of amino acids: 1412\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▍ | 911/1074 [1:06:40<04:32,  1.67s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6ia2.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of internal loops: 5\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▍ | 912/1074 [1:06:40<03:23,  1.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6cmn.cif1'\n",
      "    total number of nucleotides: 27\n",
      "    total number of amino acids: 91\n",
      "[w] arginine [1..A.ARG.36.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.36.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 11\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▌ | 913/1074 [1:06:40<02:35,  1.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6ar5.cif1'\n",
      "    total number of nucleotides: 27\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 26\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▌ | 914/1074 [1:06:41<01:56,  1.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v9f.cif1'\n",
      "1..0.PSU.2621. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 2930\n",
      "    total number of amino acids: 4311\n",
      "[w] arginine [1..A.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.206.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.206.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.129.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.234.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.248.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.312.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.145.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.74.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.94.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.99.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.73.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.122.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.154.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Y.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.38.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1485\n",
      "    total number of multiplets: 247\n",
      "    total number of helices: 87\n",
      "    total number of stems: 182\n",
      "    total number of isolated WC/wobble pairs: 54\n",
      "    total number of non-pairing interactions: 4114\n",
      "    total number of atom-base capping interactions: 247\n",
      "    total number of splayed-apart dinucleotides: 412\n",
      "                        consolidated into units: 256\n",
      "    total number of hairpin loops: 72\n",
      "    total number of bulges: 37\n",
      "    total number of internal loops: 71\n",
      "    total number of junctions: 37\n",
      "    total number of non-loop single-stranded segments: 24\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v9f.cif1', hairpin loops (#1/#3) have 2 interactions [#8, #9 ...]\n",
      "    total number of kissing loops: 6\n",
      "    total number of A-minor (types I and II) motifs: 107\n",
      "    total number of eXtended A-minor (type X) motifs: 48\n",
      "    total number of U-turns: 33\n",
      "    total number of ribose zippers: 46\n",
      "    total number of kink turns: 9\n",
      "    total number of phosphate interactions: 1184\n",
      "\n",
      "Time used: 00:00:00:39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▌ | 915/1074 [1:07:20<32:30, 12.26s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3rw6.cif1'\n",
      "  1.H.CCC.62 0.130\n",
      "  1.F.CCC.62 0.128\n",
      "    total number of nucleotides: 124\n",
      "    total number of amino acids: 489\n",
      "[w] arginine [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.276.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.332.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.276.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 56\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 10\n",
      "    total number of non-pairing interactions: 143\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 6\n",
      "    total number of U-turns: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 24\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▌ | 916/1074 [1:07:21<23:24,  8.89s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3wzi.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 212\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▌ | 917/1074 [1:07:21<16:31,  6.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3tzr.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of base pairs: 16\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 85%|████████▌ | 918/1074 [1:07:21<11:41,  4.50s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3bsx.cif1'\n",
      "    total number of nucleotides: 19\n",
      "    total number of amino acids: 681\n",
      "    total number of non-pairing interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 919/1074 [1:07:22<08:41,  3.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1pvo.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 2397\n",
      "[w] arginine [1..E.ARG.160.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.160.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 11\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 920/1074 [1:07:24<07:21,  2.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2g4b.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 172\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 921/1074 [1:07:24<05:17,  2.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2f8k.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 84\n",
      "    total number of base pairs: 7\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 15\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of hairpin loops: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 922/1074 [1:07:24<03:49,  1.51s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4c9d.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 522\n",
      "[w] Residue [1..A.ARG.32.] -- missing planar atom ' NH2'\n",
      "[w] Residue [1..A.ARG.189.] -- missing planar atom ' NH2'\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 923/1074 [1:07:25<03:05,  1.23s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d06.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of amino acids: 736\n",
      "[w] arginine [1..A.ARG.510.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..B.8AZ.13. missing planar atom [ C8 ]\n",
      "\t[i] 1..B.8AZ.13.: replace missing base ring atom [ C8 ] with [N8] (d=0.07)\n",
      "[w] Residue [1..A.ARG.510.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.674.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.674.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 22\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 45\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 14\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 924/1074 [1:07:25<02:36,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6e1s.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 925/1074 [1:07:26<01:56,  1.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4p95.cif1'\n",
      "    total number of nucleotides: 189\n",
      "    total number of base pairs: 91\n",
      "    total number of multiplets: 16\n",
      "    total number of helices: 6\n",
      "    total number of stems: 14\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 246\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 5\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 5\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 3\n",
      "    total number of phosphate interactions: 24\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▌ | 926/1074 [1:07:26<01:42,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3m7n.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of amino acids: 2044\n",
      "[w] arginine [1..A.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.121.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.164.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.212.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.212.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.78.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.97.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.194.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.212.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.240.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.27.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.250.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 9\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▋ | 927/1074 [1:07:27<02:01,  1.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ns3.cif1'\n",
      "    total number of nucleotides: 63\n",
      "    total number of amino acids: 273\n",
      "[w] arginine [1..A.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.115.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▋ | 928/1074 [1:07:28<01:41,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4y4o.cif1'\n",
      "  1.1A.U.1418 0.121\n",
      "  1.2A.U.2130 0.123\n",
      "  1.2B.U.8 0.122\n",
      "  1.2a.U.841 0.121\n",
      "1..1A.PSU.1933. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.1939. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1A.PSU.2617. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1911. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.1917. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2A.PSU.2605. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2a.PSU.516. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 8981\n",
      "    total number of amino acids: 11833\n",
      "[w] arginine [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1G.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue 1..1B.A.57. missing planar atom [ C8 ]\n",
      "[w] Residue [1..1D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1G.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1G.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1I.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Q.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1W.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1X.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1Z.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..10.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..12.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..13.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..14.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1b.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1b.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1c.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1e.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1g.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1h.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1l.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1p.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1y.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2D.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2O.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2P.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2R.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2T.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2U.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2X.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2Z.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..20.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..22.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..23.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..24.ARG.68.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..28.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2d.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2f.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2h.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2i.ARG.66.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2n.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2o.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2q.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2r.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2s.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2t.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2y.ARG.95.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4621\n",
      "    total number of multiplets: 747\n",
      "    total number of helices: 249\n",
      "    total number of stems: 567\n",
      "    total number of isolated WC/wobble pairs: 170\n",
      "    total number of non-pairing interactions: 12980\n",
      "    total number of atom-base capping interactions: 662\n",
      "    total number of splayed-apart dinucleotides: 1172\n",
      "                        consolidated into units: 729\n",
      "    total number of hairpin loops: 207\n",
      "    total number of bulges: 134\n",
      "    total number of internal loops: 214\n",
      "    total number of junctions: 112\n",
      "    total number of non-loop single-stranded segments: 65\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4y4o.cif1', hairpin loops (#1/#3) have 2 interactions [#10, #11 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4y4o.cif1', hairpin loops (#105/#107) have 2 interactions [#293, #294 ...]\n",
      "    total number of kissing loops: 14\n",
      "    total number of A-minor (types I and II) motifs: 287\n",
      "    total number of eXtended A-minor (type X) motifs: 211\n",
      "    total number of U-turns: 100\n",
      "    total number of ribose zippers: 133\n",
      "    total number of kink turns: 25\n",
      "    total number of phosphate interactions: 3617\n",
      "[w]  number-of-residues=390 -- in shortened form\n",
      "[w]  number-of-residues=390 -- in shortened form\n",
      "\n",
      "Time used: 00:00:03:14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 86%|████████▋ | 929/1074 [1:10:42<2:22:25, 58.93s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6htu.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 217\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 39\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 930/1074 [1:10:43<1:39:10, 41.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3boy.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of amino acids: 444\n",
      "[w] arginine [1..B.ARG.70.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.70.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1\n",
      "    total number of non-pairing interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 931/1074 [1:10:43<1:09:09, 29.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ndb.cif1'\n",
      "    total number of nucleotides: 136\n",
      "    total number of amino acids: 507\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 159\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 13\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 5\n",
      "    total number of junctions: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 34\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 932/1074 [1:10:44<48:43, 20.59s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1i9x.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of base pairs: 13\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 26\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 933/1074 [1:10:44<34:00, 14.47s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1l3z.cif1'\n",
      "    total number of nucleotides: 14\n",
      "    total number of non-pairing interactions: 14\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 934/1074 [1:10:44<23:45, 10.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3zgz.cif1'\n",
      "    total number of nucleotides: 159\n",
      "    total number of amino acids: 1729\n",
      "    total number of base pairs: 74\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 184\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 26\n",
      "                        consolidated into units: 18\n",
      "    total number of hairpin loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 57\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 935/1074 [1:10:46<17:58,  7.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2i91.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of amino acids: 1046\n",
      "[w] arginine [1..A.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.85.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.149.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.384.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.531.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.34.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.146.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.174.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.184.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.342.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.487.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.514.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.531.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 45\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 4\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 26\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 936/1074 [1:10:47<13:07,  5.70s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1h38.cif1'\n",
      "    total number of nucleotides: 137\n",
      "    total number of amino acids: 3428\n",
      "[w] arginine [1..A.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.52.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 65\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 8\n",
      "    total number of stems: 8\n",
      "    total number of non-pairing interactions: 125\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "    total number of non-loop single-stranded segments: 9\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:09\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 937/1074 [1:10:56<15:11,  6.65s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3qgc.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 400\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 938/1074 [1:10:57<11:00,  4.86s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1mji.cif1'\n",
      "    total number of nucleotides: 68\n",
      "    total number of amino acids: 359\n",
      "    total number of base pairs: 31\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 77\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 10\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 87%|████████▋ | 939/1074 [1:10:58<08:31,  3.79s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xs7.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 86\n",
      "    total number of base pairs: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 940/1074 [1:10:58<06:07,  2.74s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5swd.cif1'\n",
      "    total number of nucleotides: 133\n",
      "    total number of base pairs: 64\n",
      "    total number of multiplets: 5\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 164\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 13\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 941/1074 [1:10:59<04:40,  2.11s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t5h.cif1'\n",
      "  1.A.7MG.931 0.120\n",
      "  1.A.7MG.1045 0.123\n",
      "  1.A.7MG.1725 0.120\n",
      "  1.B.7MG.657 0.126\n",
      "  1.B.7MG.1107 0.131\n",
      "  1.B.7MG.1138 0.135\n",
      "  1.C.7MG.42 0.129\n",
      "    total number of nucleotides: 2884\n",
      "    total number of amino acids: 5369\n",
      "[w] arginine [1..I.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Q.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Q.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Q.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..Q.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.175.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.83.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.156.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.162.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..a.ARG.132.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.26.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.124.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..c.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.347.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..f.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..i.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..k.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..k.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..l.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..m.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..o.ARG.4.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..r.ARG.198.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..r.ARG.246.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..r.ARG.323.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.38.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..t.ARG.87.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..v.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..v.ARG.80.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..v.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..w.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..x.ARG.307.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1349\n",
      "    total number of multiplets: 185\n",
      "    total number of helices: 89\n",
      "    total number of stems: 188\n",
      "    total number of isolated WC/wobble pairs: 72\n",
      "    total number of non-pairing interactions: 3972\n",
      "    total number of atom-base capping interactions: 206\n",
      "    total number of splayed-apart dinucleotides: 456\n",
      "                        consolidated into units: 267\n",
      "    total number of hairpin loops: 52\n",
      "    total number of bulges: 44\n",
      "    total number of internal loops: 63\n",
      "    total number of junctions: 28\n",
      "    total number of non-loop single-stranded segments: 104\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t5h.cif1', hairpin loops (#9/#40) have 2 interactions [#22, #23 ...]\n",
      "    total number of kissing loops: 5\n",
      "    total number of A-minor (types I and II) motifs: 48\n",
      "    total number of eXtended A-minor (type X) motifs: 52\n",
      "    total number of U-turns: 20\n",
      "    total number of ribose zippers: 35\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 1306\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 942/1074 [1:11:27<22:02, 10.02s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4z4d.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of amino acids: 803\n",
      "[w] arginine [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.506.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 943/1074 [1:11:29<15:59,  7.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5hk0.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 426\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 944/1074 [1:11:29<11:15,  5.20s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6hma.cif1'\n",
      "    total number of nucleotides: 2949\n",
      "    total number of amino acids: 3039\n",
      "[w] arginine [1..C.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.43.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.243.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.106.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.54.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.62.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.152.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.163.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.170.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.2.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.100.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.45.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.51.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.56.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.115.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.14.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.96.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..M.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.23.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.90.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.91.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..O.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.67.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..P.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.86.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..R.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..S.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.9.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..T.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.49.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..U.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..V.ARG.37.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.44.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.52.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..W.ARG.60.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.10.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Z.ARG.7.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..1.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.22.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.24.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..4.ARG.19.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 1439\n",
      "    total number of multiplets: 208\n",
      "    total number of helices: 85\n",
      "    total number of stems: 181\n",
      "    total number of isolated WC/wobble pairs: 69\n",
      "    total number of non-pairing interactions: 3737\n",
      "    total number of atom-base capping interactions: 161\n",
      "    total number of splayed-apart dinucleotides: 415\n",
      "                        consolidated into units: 267\n",
      "    total number of hairpin loops: 71\n",
      "    total number of bulges: 36\n",
      "    total number of internal loops: 78\n",
      "    total number of junctions: 36\n",
      "    total number of non-loop single-stranded segments: 20\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/6hma.cif1', hairpin loops (#1/#3) have 2 interactions [#10, #11 ...]\n",
      "    total number of kissing loops: 7\n",
      "    total number of A-minor (types I and II) motifs: 75\n",
      "    total number of eXtended A-minor (type X) motifs: 61\n",
      "    total number of U-turns: 24\n",
      "    total number of ribose zippers: 31\n",
      "    total number of kink turns: 6\n",
      "    total number of phosphate interactions: 971\n",
      "\n",
      "Time used: 00:00:00:29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 945/1074 [1:11:57<26:22, 12.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5wzh.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 533\n",
      "[w] arginine [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.329.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.468.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.629.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.290.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.329.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.468.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.629.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 6\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 946/1074 [1:11:58<18:46,  8.80s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lqt.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 947/1074 [1:11:59<13:15,  6.27s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v90.cif1'\n",
      "    total number of nucleotides: 4613\n",
      "    total number of amino acids: 6946\n",
      "    total number of base pairs: 2295\n",
      "    total number of multiplets: 349\n",
      "    total number of helices: 138\n",
      "    total number of stems: 283\n",
      "    total number of isolated WC/wobble pairs: 95\n",
      "    total number of non-pairing interactions: 6637\n",
      "    total number of atom-base capping interactions: 307\n",
      "    total number of splayed-apart dinucleotides: 624\n",
      "                        consolidated into units: 387\n",
      "    total number of hairpin loops: 113\n",
      "[w]  sheared G-A pair [5]: asymmetric internal loop: nts=9; [3,2]; linked by [#90,#-27]\n",
      "    total number of bulges: 65\n",
      "    total number of internal loops: 115\n",
      "    total number of junctions: 55\n",
      "    total number of non-loop single-stranded segments: 31\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v90.cif1', hairpin loops (#37/#39) have 2 interactions [#106, #107 ...]\n",
      "    total number of kissing loops: 8\n",
      "    total number of A-minor (types I and II) motifs: 124\n",
      "    total number of eXtended A-minor (type X) motifs: 88\n",
      "    total number of U-turns: 50\n",
      "    total number of ribose zippers: 59\n",
      "    total number of kink turns: 11\n",
      "    total number of phosphate interactions: 1714\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:57\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 948/1074 [1:12:56<45:17, 21.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ato.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of amino acids: 168\n",
      "    total number of base pairs: 12\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 949/1074 [1:12:56<31:36, 15.17s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4kre.cif1'\n",
      "    total number of nucleotides: 11\n",
      "    total number of amino acids: 799\n",
      "[w] arginine [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.633.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.633.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 88%|████████▊ | 950/1074 [1:12:57<22:28, 10.87s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3avw.cif1'\n",
      "    total number of nucleotides: 21\n",
      "    total number of amino acids: 1203\n",
      "[w] arginine [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.564.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.573.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.612.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.618.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.710.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.711.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.752.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.914.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.937.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.997.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1088.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1197.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1249.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 9\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 17\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▊ | 951/1074 [1:12:58<16:32,  8.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1sdr.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 58\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▊ | 952/1074 [1:12:59<11:36,  5.71s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1av6.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 290\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▊ | 953/1074 [1:12:59<08:14,  4.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2voo.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 356\n",
      "[w] arginine [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.57.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.57.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 954/1074 [1:12:59<05:53,  2.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3mei.cif1'\n",
      "    total number of nucleotides: 46\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 955/1074 [1:12:59<04:12,  2.12s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ev3.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 196\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 956/1074 [1:13:00<03:06,  1.58s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xgj.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 1741\n",
      "[w] Residue [1..A.ARG.470.] -- missing planar atom ' NH2'\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 957/1074 [1:13:01<03:06,  1.60s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3oxe.cif1'\n",
      "  1.A.CCC.88 0.128\n",
      "  1.B.CCC.88 0.128\n",
      "    total number of nucleotides: 176\n",
      "    total number of amino acids: 2\n",
      "    total number of base pairs: 77\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 237\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 23\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of A-minor (types I and II) motifs: 7\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 5\n",
      "    total number of phosphate interactions: 48\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 958/1074 [1:13:02<02:39,  1.37s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5czz.cif1'\n",
      "    total number of nucleotides: 108\n",
      "    total number of amino acids: 1043\n",
      "[w] arginine [1..A.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.480.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.497.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.47.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.140.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.208.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.338.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.480.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.497.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.504.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.617.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.630.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.665.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.774.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.792.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.913.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.980.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1002.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 49\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 3\n",
      "    total number of stems: 5\n",
      "    total number of non-pairing interactions: 112\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 47\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 959/1074 [1:13:04<02:41,  1.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3vnv.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 1203\n",
      "[w] arginine [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.13.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.42.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.119.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.134.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.151.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.292.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.359.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.401.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.518.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.564.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.573.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.612.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.618.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.662.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.710.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.711.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.752.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.761.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.803.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.847.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.885.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.914.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.937.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.939.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.997.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1066.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1088.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1107.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1125.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1183.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1193.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1197.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1210.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1241.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1249.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 15\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 960/1074 [1:13:05<02:48,  1.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5u0q.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 26\n",
      "    total number of atom-base capping interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 89%|████████▉ | 961/1074 [1:13:06<02:09,  1.15s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2nug.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 434\n",
      "    total number of base pairs: 20\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 46\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 19\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|████████▉ | 962/1074 [1:13:06<01:45,  1.06it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t3k.cif1'\n",
      "    total number of nucleotides: 66\n",
      "    total number of base pairs: 29\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 72\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 3\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|████████▉ | 963/1074 [1:13:06<01:20,  1.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ptx.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of amino acids: 2086\n",
      "[w] arginine [1..A.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.179.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.179.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 38\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 10\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 43\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|████████▉ | 964/1074 [1:13:08<01:48,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2hvy.cif1'\n",
      "    total number of nucleotides: 58\n",
      "    total number of amino acids: 568\n",
      "    total number of base pairs: 22\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 62\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 5\n",
      "                        consolidated into units: 4\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|████████▉ | 965/1074 [1:13:09<01:42,  1.07it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2val.cif1'\n",
      "  1.A.A.7 0.136\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 14\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 31\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 1\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|████████▉ | 966/1074 [1:13:09<01:18,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4wtk.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 553\n",
      "[w] arginine [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.48.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.114.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.200.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.465.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.498.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.505.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 3\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|█████████ | 967/1074 [1:13:10<01:13,  1.46it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2e9r.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..X.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..X.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|█████████ | 968/1074 [1:13:10<01:07,  1.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2et8.cif1'\r\n",
      "    total number of nucleotides: 42\r\n",
      "    total number of base pairs: 18\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 3\r\n",
      "    total number of isolated WC/wobble pairs: 2\r\n",
      "    total number of non-pairing interactions: 38\r\n",
      "    total number of splayed-apart dinucleotides: 5\r\n",
      "                        consolidated into units: 4\r\n",
      "    total number of internal loops: 4\r\n",
      "    total number of phosphate interactions: 5\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|█████████ | 969/1074 [1:13:10<00:51,  2.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6g3b.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 450\n",
      "[w] arginine [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.33.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 21\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|█████████ | 970/1074 [1:13:11<00:53,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5e81.cif1'\n",
      "  1.13.U.37 0.121\n",
      "  1.13.U.229 0.122\n",
      "  1.13.U.365 0.124\n",
      "  1.13.U.427 0.120\n",
      "  1.13.U.497 0.122\n",
      "  1.13.U.560 0.122\n",
      "  1.13.U.598 0.124\n",
      "  1.13.U.723 0.124\n",
      "  1.13.U.884 0.122\n",
      "1..1K.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1K.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2K.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1L.PSU.39. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..1L.PSU.55. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "1..2L.PSU.56. has pseudo-uridine C1'...C5 glycosidic bond\n",
      "    total number of nucleotides: 9463\n",
      "    total number of amino acids: 11204\n",
      "[w] arginine [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..8I.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..71.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..3E.ARG.209.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..6E.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..2I.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..8I.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AI.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BI.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..71.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..11.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..21.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..41.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..78.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..98.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B8.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C8.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I8.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..Q8.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..32.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..62.ARG.143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BA.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..79.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..19.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..29.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..49.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.16.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.65.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..35.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..55.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..85.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..95.ARG.21.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E5.ARG.32.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4673\n",
      "    total number of multiplets: 692\n",
      "    total number of helices: 276\n",
      "    total number of stems: 588\n",
      "    total number of isolated WC/wobble pairs: 185\n",
      "    total number of non-pairing interactions: 13580\n",
      "    total number of atom-base capping interactions: 688\n",
      "    total number of splayed-apart dinucleotides: 1206\n",
      "                        consolidated into units: 766\n",
      "    total number of hairpin loops: 224\n",
      "    total number of bulges: 122\n",
      "    total number of internal loops: 239\n",
      "    total number of junctions: 113\n",
      "    total number of non-loop single-stranded segments: 93\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5e81.cif1', hairpin loops (#39/#41) have 2 interactions [#117, #118 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/5e81.cif1', hairpin loops (#153/#155) have 2 interactions [#415, #416 ...]\n",
      "    total number of kissing loops: 16\n",
      "    total number of A-minor (types I and II) motifs: 250\n",
      "    total number of eXtended A-minor (type X) motifs: 189\n",
      "    total number of U-turns: 105\n",
      "    total number of ribose zippers: 131\n",
      "    total number of kink turns: 18\n",
      "    total number of phosphate interactions: 3448\n",
      "[w]  number-of-residues=508 -- in shortened form\n",
      "[w]  number-of-residues=508 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:02:59\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 90%|█████████ | 971/1074 [1:16:11<1:33:22, 54.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v8b.cif1'\n",
      "  1.AA.U.97 0.122\n",
      "  1.AA.U.1495 0.124\n",
      "  1.AB.U.17 0.122\n",
      "  1.CA.U.1495 0.121\n",
      "  1.CB.U.39 0.128\n",
      "  1.CB.U.42 0.120\n",
      "  1.CB.U.58 0.120\n",
      "  1.CB.U.65 0.121\n",
      "  1.CB.U.76 0.121\n",
      "    total number of nucleotides: 9577\n",
      "    total number of amino acids: 11527\n",
      "[w] arginine [1..AG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AG.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AJ.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AT.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..AW.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BD.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BD.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BE.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..BE.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AG.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AJ.ARG.155.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AT.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..AW.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BD.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BE.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BE.ARG.79.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BG.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..BO.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B0.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B0.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B1.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B1.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B1.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B3.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B3.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B8.ARG.30.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.3.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CG.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CT.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..CW.ARG.89.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DD.ARG.239.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DE.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DG.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.41.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.61.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.102.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DO.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..DP.ARG.59.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D0.ARG.88.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D1.ARG.64.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D3.ARG.32.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D4.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D6.ARG.37.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4788\n",
      "    total number of multiplets: 687\n",
      "    total number of helices: 279\n",
      "    total number of stems: 599\n",
      "    total number of isolated WC/wobble pairs: 178\n",
      "    total number of non-pairing interactions: 13756\n",
      "    total number of atom-base capping interactions: 650\n",
      "    total number of splayed-apart dinucleotides: 1258\n",
      "                        consolidated into units: 792\n",
      "    total number of hairpin loops: 239\n",
      "    total number of bulges: 133\n",
      "    total number of internal loops: 226\n",
      "    total number of junctions: 120\n",
      "    total number of non-loop single-stranded segments: 58\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v8b.cif1', hairpin loops (#44/#46) have 2 interactions [#122, #123 ...]\n",
      "[i] -- kissing-loops in '/home/mikhail/bioinformatics/data/NR_3.0/cif/4v8b.cif1', hairpin loops (#164/#166) have 2 interactions [#422, #423 ...]\n",
      "    total number of kissing loops: 17\n",
      "    total number of A-minor (types I and II) motifs: 253\n",
      "    total number of eXtended A-minor (type X) motifs: 196\n",
      "    total number of U-turns: 99\n",
      "    total number of ribose zippers: 133\n",
      "    total number of kink turns: 23\n",
      "    total number of phosphate interactions: 3229\n",
      "[w]  number-of-residues=462 -- in shortened form\n",
      "[w]  number-of-residues=462 -- in shortened form\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:03:44\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 972/1074 [1:19:55<2:59:02, 105.32s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5mfx.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 434\n",
      "    total number of non-pairing interactions: 3\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 973/1074 [1:19:56<2:04:17, 73.84s/it] "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/433d.cif1'\r\n",
      "    total number of nucleotides: 28\r\n",
      "    total number of base pairs: 14\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 27\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 974/1074 [1:19:56<1:26:13, 51.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5gxi.cif1'\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 667\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.684.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.120.] -- missing planar atom ' NH2'\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.TRP.513.] -- missing planar atom ' CD2'\n",
      "[w] Residue [1..A.ARG.684.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 975/1074 [1:19:56<1:00:04, 36.41s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qpx.cif1'\n",
      "    total number of nucleotides: 15\n",
      "    total number of amino acids: 495\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 976/1074 [1:19:57<41:52, 25.64s/it]  "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1c9s.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of amino acids: 1551\n",
      "[w] arginine [1..I.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.58.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 42\n",
      "    total number of atom-base capping interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 977/1074 [1:19:58<29:38, 18.34s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lr4.cif1'\n",
      "    total number of nucleotides: 40\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 50\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 978/1074 [1:19:59<20:43, 12.96s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5v1k.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of base pairs: 12\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 20\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 979/1074 [1:19:59<14:27,  9.13s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1g2j.cif1'\n",
      "1..A.PYY.4. has unusual GLYCOSIDIC bond!\n",
      "    total number of nucleotides: 8\n",
      "[w] Residue 1..A.PYY.4. missing planar atom [ N3 ]\n",
      "[w] Residue 1..A.PYY.4. missing planar atom [ N1 ]\n",
      "\t[i] 1..A.PYY.4.: replace missing base ring atom [ N3 ] with [C3] (d=0.08)\n",
      "\t[i] 1..A.PYY.4.: replace missing base ring atom [ N1 ] with [C1] (d=0.02)\n",
      "    total number of non-pairing interactions: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████ | 980/1074 [1:19:59<10:06,  6.45s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5h1l.cif1'\n",
      "    total number of nucleotides: 7\n",
      "    total number of amino acids: 682\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.273.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.284.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.335.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.527.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.682.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.719.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████▏| 981/1074 [1:20:00<07:17,  4.70s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3qjj.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 483\n",
      "    total number of base pairs: 2\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 91%|█████████▏| 982/1074 [1:20:00<05:20,  3.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/406d.cif1'\n",
      "    total number of nucleotides: 34\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 36\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 983/1074 [1:20:01<03:50,  2.53s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5fj0.cif1'\n",
      "    total number of nucleotides: 57\n",
      "    total number of base pairs: 19\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 66\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 984/1074 [1:20:01<02:52,  1.92s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5xc6.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 902\n",
      "[w] arginine [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.254.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.617.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.202.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.217.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.236.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.463.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.538.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.568.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.617.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 985/1074 [1:20:02<02:18,  1.56s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4oau.cif1'\n",
      "  1.A.A.1 0.143\n",
      "  1.A.A.2 0.147\n",
      "    total number of nucleotides: 4\n",
      "    total number of amino acids: 692\n",
      "[w] arginine [1..C.ARG.257.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.257.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 986/1074 [1:20:02<01:53,  1.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2o3v.cif1'\n",
      "    total number of nucleotides: 43\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 44\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 1\n",
      "    total number of internal loops: 4\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 987/1074 [1:20:03<01:25,  1.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4knq.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 122\n",
      "[w] arginine [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.53.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 14\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 988/1074 [1:20:03<01:03,  1.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/438d.cif1'\r\n",
      "    total number of nucleotides: 16\r\n",
      "    total number of base pairs: 8\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 14\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 989/1074 [1:20:03<00:48,  1.76it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3add.cif1'\n",
      "    total number of nucleotides: 178\n",
      "    total number of amino acids: 490\n",
      "[w] arginine [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.195.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.111.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.112.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 76\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 210\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 15\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 3\n",
      "    total number of phosphate interactions: 36\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 990/1074 [1:20:04<00:46,  1.79it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5ay2.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 24\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 53\n",
      "    total number of internal loops: 4\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 991/1074 [1:20:04<00:46,  1.78it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5t16.cif1'\n",
      "    total number of nucleotides: 136\n",
      "    total number of amino acids: 2006\n",
      "    total number of base pairs: 60\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 133\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 4\n",
      "    total number of phosphate interactions: 43\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 992/1074 [1:20:05<01:01,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3dil.cif1'\n",
      "    total number of nucleotides: 174\n",
      "    total number of amino acids: 1\n",
      "[w]  chainId 'A' with nts=174 and aas=1\n",
      "    total number of base pairs: 88\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 3\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 225\n",
      "    total number of atom-base capping interactions: 9\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 5\n",
      "    total number of bulges: 3\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 5\n",
      "    total number of U-turns: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 92%|█████████▏| 993/1074 [1:20:06<00:59,  1.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1n35.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 1264\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 994/1074 [1:20:08<01:24,  1.06s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4e78.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 538\n",
      "[w] arginine [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.81.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.98.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.109.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.168.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.250.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.259.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.277.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.337.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.345.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.394.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.484.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.503.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.508.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.514.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 8\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 995/1074 [1:20:09<01:15,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1n78.cif1'\n",
      "    total number of nucleotides: 152\n",
      "    total number of amino acids: 936\n",
      "[w] arginine [1..A.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.104.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.127.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.319.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.349.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 69\n",
      "    total number of multiplets: 8\n",
      "    total number of helices: 4\n",
      "    total number of stems: 8\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 178\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 6\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of kissing loops: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 2\n",
      "    total number of phosphate interactions: 39\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 996/1074 [1:20:10<01:21,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1yls.cif1'\n",
      "    total number of nucleotides: 98\n",
      "    total number of base pairs: 46\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 4\n",
      "    total number of stems: 10\n",
      "    total number of non-pairing interactions: 108\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 13\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 997/1074 [1:20:10<01:02,  1.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3szx.cif1'\n",
      "  1.A.U.11 0.122\n",
      "  1.B.U.2 0.121\n",
      "    total number of nucleotides: 38\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 1\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 40\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 998/1074 [1:20:10<00:49,  1.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5a0t.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 905\n",
      "    total number of non-pairing interactions: 7\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 999/1074 [1:20:11<00:52,  1.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4z0c.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of amino acids: 1418\n",
      "[w] arginine [1..A.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.260.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.330.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.260.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.330.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.604.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.103.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.260.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.330.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.207.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.260.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.330.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.604.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 18\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 12\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 1000/1074 [1:20:13<01:22,  1.12s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4pdb.cif1'\n",
      "    total number of nucleotides: 38\n",
      "    total number of amino acids: 129\n",
      "    total number of base pairs: 22\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 1001/1074 [1:20:14<01:09,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4oji.cif1'\n",
      "    total number of nucleotides: 52\n",
      "    total number of base pairs: 26\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 57\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 2\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 1002/1074 [1:20:14<01:00,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1mzp.cif1'\n",
      "    total number of nucleotides: 55\n",
      "    total number of amino acids: 217\n",
      "    total number of base pairs: 27\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 3\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 66\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of U-turns: 2\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 22\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 1003/1074 [1:20:15<00:52,  1.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qtq.cif1'\n",
      "    total number of nucleotides: 75\n",
      "    total number of amino acids: 530\n",
      "    total number of base pairs: 33\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 81\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 23\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 93%|█████████▎| 1004/1074 [1:20:16<00:53,  1.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q3z.cif1'\n",
      "    total number of nucleotides: 154\n",
      "    total number of base pairs: 77\n",
      "    total number of multiplets: 10\n",
      "    total number of helices: 6\n",
      "    total number of stems: 10\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 194\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 2\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of kink turns: 2\n",
      "    total number of phosphate interactions: 31\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▎| 1005/1074 [1:20:17<00:57,  1.19it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5k77.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 1745\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 5\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 10\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▎| 1006/1074 [1:20:18<01:03,  1.07it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1mdg.cif1'\r\n",
      "    total number of nucleotides: 6\r\n",
      "    total number of non-pairing interactions: 5\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 4\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1007/1074 [1:20:18<00:46,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4aob.cif1'\n",
      "    total number of nucleotides: 95\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 47\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 4\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 4\n",
      "    total number of non-pairing interactions: 120\n",
      "    total number of atom-base capping interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 7\n",
      "    total number of hairpin loops: 3\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 20\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1008/1074 [1:20:19<00:45,  1.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ig8.cif1'\n",
      "    total number of nucleotides: 37\n",
      "    total number of amino acids: 338\n",
      "    total number of base pairs: 18\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 36\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1009/1074 [1:20:19<00:39,  1.66it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6gvy.cif1'\n",
      "    total number of nucleotides: 12\n",
      "    total number of amino acids: 475\n",
      "[w] arginine [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.75.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 5\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 12\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1010/1074 [1:20:20<00:35,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f5h.cif1'\n",
      "    total number of nucleotides: 42\n",
      "    total number of amino acids: 300\n",
      "[w] arginine [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.297.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.229.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.280.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.297.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.219.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.251.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.280.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 17\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 51\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1011/1074 [1:20:20<00:34,  1.80it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4lvw.cif1'\n",
      "    total number of nucleotides: 91\n",
      "[w] Residue 1..A.7DG.101. missing planar atom [ N7 ]\n",
      "\t[i] 1..A.7DG.101.: replace missing base ring atom [ N7 ] with [C7] (d=0.02)\n",
      "[w] Residue 1..A.7DG.102. missing planar atom [ N7 ]\n",
      "\t[i] 1..A.7DG.102.: replace missing base ring atom [ N7 ] with [C7] (d=0.03)\n",
      "    total number of base pairs: 48\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 2\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 114\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 4\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of ribose zippers: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1012/1074 [1:20:20<00:27,  2.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4fxd.cif1'\n",
      "    total number of nucleotides: 32\n",
      "    total number of amino acids: 1655\n",
      "[w] arginine [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.490.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.490.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 30\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1013/1074 [1:20:23<01:00,  1.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4z31.cif1'\n",
      "    total number of nucleotides: 58\n",
      "    total number of amino acids: 588\n",
      "[w] arginine [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.248.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.148.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.349.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.351.] -- missing planar atom ' NH2'\n",
      "[w] Residue [1..B.ARG.187.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.216.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.248.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.349.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 25\n",
      "    total number of helices: 4\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 56\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 6\n",
      "    total number of phosphate interactions: 21\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 94%|█████████▍| 1014/1074 [1:20:23<00:51,  1.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5f9f.cif1'\n",
      "    total number of nucleotides: 144\n",
      "    total number of amino acids: 4028\n",
      "[w] arginine [1..A.ARG.502.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.664.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.502.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..G.ARG.664.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..K.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..K.ARG.664.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.502.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.664.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.488.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.502.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.664.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.316.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..K.ARG.664.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 65\n",
      "    total number of helices: 6\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 149\n",
      "    total number of atom-base capping interactions: 12\n",
      "    total number of hairpin loops: 6\n",
      "    total number of U-turns: 6\n",
      "    total number of phosphate interactions: 70\n",
      "\n",
      "Time used: 00:00:00:04\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▍| 1015/1074 [1:20:27<01:42,  1.75s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2r7x.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of amino acids: 2146\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.190.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▍| 1016/1074 [1:20:29<01:52,  1.95s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5g4u.cif1'\n",
      "    total number of nucleotides: 114\n",
      "    total number of amino acids: 711\n",
      "    total number of base pairs: 55\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 4\n",
      "    total number of stems: 9\n",
      "    total number of non-pairing interactions: 136\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 6\n",
      "    total number of internal loops: 6\n",
      "    total number of kink turns: 6\n",
      "    total number of phosphate interactions: 25\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▍| 1017/1074 [1:20:31<01:41,  1.79s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2xlk.cif1'\n",
      "    total number of nucleotides: 30\n",
      "    total number of amino acids: 369\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 29\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 17\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▍| 1018/1074 [1:20:31<01:19,  1.42s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1n1h.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 1264\n",
      "    total number of base pairs: 2\n",
      "    total number of helices: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:02\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▍| 1019/1074 [1:20:33<01:29,  1.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1gtn.cif1'\n",
      "    total number of nucleotides: 44\n",
      "    total number of amino acids: 1552\n",
      "[w] arginine [1..I.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.58.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 11\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▍| 1020/1074 [1:20:35<01:21,  1.52s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2pn3.cif1'\r\n",
      "    total number of nucleotides: 44\r\n",
      "    total number of base pairs: 19\r\n",
      "    total number of helices: 2\r\n",
      "    total number of stems: 3\r\n",
      "    total number of isolated WC/wobble pairs: 1\r\n",
      "    total number of non-pairing interactions: 50\r\n",
      "    total number of splayed-apart dinucleotides: 4\r\n",
      "                        consolidated into units: 2\r\n",
      "    total number of bulges: 2\r\n",
      "    total number of internal loops: 1\r\n",
      "    total number of non-loop single-stranded segments: 2\r\n",
      "    total number of phosphate interactions: 2\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▌| 1021/1074 [1:20:35<00:58,  1.10s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qu6.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 99\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of atom-base capping interactions: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▌| 1022/1074 [1:20:35<00:44,  1.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6fqr.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 318\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▌| 1023/1074 [1:20:35<00:34,  1.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3agv.cif1'\n",
      "    total number of nucleotides: 43\n",
      "    total number of amino acids: 385\n",
      "[w] arginine [1..A.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.355.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.255.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.344.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.355.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 19\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 43\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 2\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▌| 1024/1074 [1:20:36<00:32,  1.55it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4xbf.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 799\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 95%|█████████▌| 1025/1074 [1:20:37<00:39,  1.24it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4p5j.cif1'\n",
      "    total number of nucleotides: 84\n",
      "    total number of base pairs: 37\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 98\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 7\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 5\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 28\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1026/1074 [1:20:38<00:36,  1.31it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4far.cif1'\n",
      "  1.A.EPE.454 0.154\n",
      "    total number of nucleotides: 395\n",
      "[w] Residue 1..A.EPE.454. missing planar atom [ C4 ]\n",
      "[w] Residue 1..A.EPE.454. missing planar atom [ N3 ]\n",
      "    total number of base pairs: 193\n",
      "    total number of multiplets: 20\n",
      "    total number of helices: 12\n",
      "    total number of stems: 24\n",
      "    total number of isolated WC/wobble pairs: 7\n",
      "    total number of non-pairing interactions: 535\n",
      "    total number of atom-base capping interactions: 19\n",
      "    total number of splayed-apart dinucleotides: 53\n",
      "                        consolidated into units: 33\n",
      "    total number of hairpin loops: 10\n",
      "    total number of bulges: 4\n",
      "    total number of internal loops: 6\n",
      "    total number of junctions: 4\n",
      "    total number of non-loop single-stranded segments: 13\n",
      "    total number of A-minor (types I and II) motifs: 9\n",
      "    total number of eXtended A-minor (type X) motifs: 4\n",
      "    total number of U-turns: 4\n",
      "    total number of ribose zippers: 8\n",
      "    total number of phosphate interactions: 86\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1027/1074 [1:20:39<00:37,  1.26it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6hc5.cif1'\r\n",
      "    total number of nucleotides: 18\r\n",
      "    total number of base pairs: 7\r\n",
      "    total number of helices: 1\r\n",
      "    total number of stems: 1\r\n",
      "    total number of non-pairing interactions: 17\r\n",
      "    total number of hairpin loops: 1\r\n",
      "    total number of phosphate interactions: 3\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1028/1074 [1:20:39<00:27,  1.67it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lta.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 698\n",
      "[w] arginine [1..A.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.464.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.567.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.410.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.464.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.567.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 5\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1029/1074 [1:20:39<00:27,  1.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4boc.cif1'\n",
      "    total number of nucleotides: 57\n",
      "    total number of amino acids: 989\n",
      "    total number of base pairs: 26\n",
      "    total number of helices: 3\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 60\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 13\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1030/1074 [1:20:41<00:33,  1.33it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5u30.cif1'\n",
      "    total number of nucleotides: 143\n",
      "    total number of amino acids: 1085\n",
      "[w] arginine [1..A.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.297.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.11.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.50.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.77.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.169.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.180.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.237.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.297.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.369.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.432.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.507.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.515.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.534.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.565.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.646.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.657.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.666.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.720.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.724.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.734.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.738.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.766.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.786.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.808.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.815.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.904.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1041.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1043.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1079.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1087.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.1092.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 58\n",
      "    total number of helices: 5\n",
      "    total number of stems: 7\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 147\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 12\n",
      "                        consolidated into units: 10\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 9\n",
      "    total number of phosphate interactions: 63\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1031/1074 [1:20:42<00:42,  1.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r9x.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 537\n",
      "    total number of base pairs: 14\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 28\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1032/1074 [1:20:43<00:34,  1.21it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4bwm.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 538\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▌| 1033/1074 [1:20:43<00:30,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jaj.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of amino acids: 653\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 26\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 18\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▋| 1034/1074 [1:20:44<00:29,  1.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4hos.cif1'\n",
      "    total number of nucleotides: 3\n",
      "    total number of amino acids: 482\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▋| 1035/1074 [1:20:44<00:25,  1.55it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4qcl.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 865\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 96%|█████████▋| 1036/1074 [1:20:45<00:27,  1.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ftf.cif1'\n",
      "    total number of nucleotides: 45\n",
      "    total number of amino acids: 247\n",
      "    total number of base pairs: 21\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 49\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 5\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1037/1074 [1:20:45<00:21,  1.72it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2a8v.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 354\n",
      "    total number of non-pairing interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1038/1074 [1:20:46<00:19,  1.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4iqs.cif1'\n",
      "    total number of nucleotides: 48\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 47\n",
      "    total number of atom-base capping interactions: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1039/1074 [1:20:46<00:15,  2.32it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4yco.cif1'\n",
      "    total number of nucleotides: 215\n",
      "    total number of amino acids: 931\n",
      "[w] arginine [1..A.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.295.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.35.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.58.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.295.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 98\n",
      "    total number of multiplets: 16\n",
      "    total number of helices: 6\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 243\n",
      "    total number of atom-base capping interactions: 11\n",
      "    total number of splayed-apart dinucleotides: 24\n",
      "                        consolidated into units: 18\n",
      "    total number of hairpin loops: 9\n",
      "    total number of junctions: 3\n",
      "    total number of non-loop single-stranded segments: 8\n",
      "    total number of kissing loops: 3\n",
      "    total number of U-turns: 6\n",
      "    total number of phosphate interactions: 58\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1040/1074 [1:20:47<00:22,  1.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4gha.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 488\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1041/1074 [1:20:48<00:18,  1.78it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3koa.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 476\n",
      "[w] arginine [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.17.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 10\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1042/1074 [1:20:48<00:16,  2.00it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1knz.cif1'\n",
      "    total number of nucleotides: 20\n",
      "    total number of amino acids: 1168\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 16\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1043/1074 [1:20:48<00:15,  1.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3q0s.cif1'\n",
      "    total number of nucleotides: 8\n",
      "    total number of amino acids: 343\n",
      "    total number of non-pairing interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1044/1074 [1:20:49<00:13,  2.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6jcx.cif1'\n",
      "    total number of nucleotides: 49\n",
      "    total number of amino acids: 3104\n",
      "[w] arginine [1..A.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.230.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.390.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.613.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.6.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.142.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.69.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.181.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.230.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.390.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.482.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.613.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.668.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.673.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.797.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.830.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.855.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.925.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.958.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1044.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.1067.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.84.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.113.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.211.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.334.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.578.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.703.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.841.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.887.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.940.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1041.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1143.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.1268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.31.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.46.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.153.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.188.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.200.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 17\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 37\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 4\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 18\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1045/1074 [1:20:52<00:37,  1.28s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/6d2z.cif1'\n",
      "    total number of nucleotides: 2\n",
      "    total number of amino acids: 188\n",
      "[w] arginine [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1046/1074 [1:20:52<00:29,  1.04s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5h9f.cif1'\n",
      "    total number of nucleotides: 139\n",
      "    total number of amino acids: 3252\n",
      "[w] arginine [1..A.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.421.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.177.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.182.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.421.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.101.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.110.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.118.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.135.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.76.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.244.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..F.ARG.82.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..G.ARG.358.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..H.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.20.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.165.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.268.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.8.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.33.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.92.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..J.ARG.96.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 52\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 8\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 109\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 21\n",
      "                        consolidated into units: 18\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 7\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 53\n",
      "[w]  cross-paired segments in separate chains, be *careful* with .dbn\n",
      "\n",
      "Time used: 00:00:00:03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 97%|█████████▋| 1047/1074 [1:20:55<00:41,  1.54s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1kd4.cif1'\n",
      "    total number of nucleotides: 22\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 26\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1048/1074 [1:20:55<00:29,  1.14s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nnc.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 175\n",
      "[w] arginine [1..A.ARG.138.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.138.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 4\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1049/1074 [1:20:56<00:21,  1.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4m4o.cif1'\n",
      "    total number of nucleotides: 59\n",
      "    total number of amino acids: 129\n",
      "[w] arginine [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.128.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 28\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 4\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 71\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 5\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 3\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of eXtended A-minor (type X) motifs: 1\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1050/1074 [1:20:56<00:17,  1.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nlf.cif1'\n",
      "    total number of nucleotides: 27\n",
      "    total number of base pairs: 12\n",
      "    total number of multiplets: 1\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 27\n",
      "    total number of atom-base capping interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "    total number of hairpin loops: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1051/1074 [1:20:56<00:12,  1.79it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5lr3.cif1'\n",
      "  1.A.CBV.2 0.129\n",
      "  1.B.CBV.2 0.128\n",
      "    total number of nucleotides: 20\n",
      "    total number of base pairs: 10\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1052/1074 [1:20:56<00:10,  2.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3d0u.cif1'\n",
      "    total number of nucleotides: 161\n",
      "    total number of amino acids: 1\n",
      "[w]  chainId 'A' with nts=161 and aas=1\n",
      "    total number of base pairs: 80\n",
      "    total number of multiplets: 6\n",
      "    total number of helices: 3\n",
      "    total number of stems: 9\n",
      "    total number of isolated WC/wobble pairs: 3\n",
      "    total number of non-pairing interactions: 205\n",
      "    total number of atom-base capping interactions: 6\n",
      "    total number of splayed-apart dinucleotides: 14\n",
      "                        consolidated into units: 9\n",
      "    total number of hairpin loops: 5\n",
      "    total number of bulges: 3\n",
      "    total number of internal loops: 3\n",
      "    total number of junctions: 2\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of kissing loops: 1\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of kink turns: 1\n",
      "    total number of phosphate interactions: 24\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1053/1074 [1:20:57<00:09,  2.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3ncu.cif1'\n",
      "    total number of nucleotides: 24\n",
      "    total number of amino acids: 240\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of phosphate interactions: 7\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1054/1074 [1:20:57<00:08,  2.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2a43.cif1'\n",
      "    total number of nucleotides: 26\n",
      "    total number of base pairs: 14\n",
      "    total number of multiplets: 4\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 27\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 2\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 3\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1055/1074 [1:20:58<00:08,  2.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5y85.cif1'\n",
      "    total number of nucleotides: 136\n",
      "    total number of base pairs: 70\n",
      "    total number of multiplets: 7\n",
      "    total number of helices: 4\n",
      "    total number of stems: 12\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 170\n",
      "    total number of atom-base capping interactions: 7\n",
      "    total number of splayed-apart dinucleotides: 20\n",
      "                        consolidated into units: 14\n",
      "    total number of hairpin loops: 4\n",
      "    total number of bulges: 2\n",
      "    total number of internal loops: 4\n",
      "    total number of junctions: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 29\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1056/1074 [1:20:59<00:13,  1.33it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5h1k.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 1364\n",
      "[w] arginine [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.141.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.719.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.18.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.40.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.55.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.120.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.190.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.192.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.252.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.258.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.318.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.335.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.719.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2\n",
      "    total number of helices: 1\n",
      "    total number of isolated WC/wobble pairs: 2\n",
      "    total number of non-pairing interactions: 13\n",
      "    total number of splayed-apart dinucleotides: 8\n",
      "                        consolidated into units: 6\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 13\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 98%|█████████▊| 1057/1074 [1:21:01<00:16,  1.05it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4nfq.cif1'\n",
      "    total number of nucleotides: 48\n",
      "[w] Residue 1..A.7AT.5. missing planar atom [ N7 ]\n",
      "[w] Residue 1..A.7AT.5. missing planar atom [ C8 ]\n",
      "\t[i] 1..A.7AT.5.: replace missing base ring atom [ N7 ] with [C7] (d=0.21)\n",
      "\t[i] 1..A.7AT.5.: replace missing base ring atom [ C8 ] with [N8] (d=0.25)\n",
      "[w] Residue 1..B.7AT.5. missing planar atom [ N7 ]\n",
      "[w] Residue 1..B.7AT.5. missing planar atom [ C8 ]\n",
      "\t[i] 1..B.7AT.5.: replace missing base ring atom [ N7 ] with [C7] (d=0.20)\n",
      "\t[i] 1..B.7AT.5.: replace missing base ring atom [ C8 ] with [N8] (d=0.23)\n",
      "[w] Residue 1..C.7AT.5. missing planar atom [ N7 ]\n",
      "[w] Residue 1..C.7AT.5. missing planar atom [ C8 ]\n",
      "\t[i] 1..C.7AT.5.: replace missing base ring atom [ N7 ] with [C7] (d=0.21)\n",
      "    total number of base pairs: 16\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 50\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▊| 1058/1074 [1:21:01<00:11,  1.37it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4o8j.cif1'\n",
      "  1.B.EPE.402 0.199\n",
      "[i] taken as a pyrimidine ... 1.B.EPE.402\n",
      "    total number of nucleotides: 13\n",
      "    total number of amino acids: 682\n",
      "[w] Residue 1..B.EPE.402. missing planar atom [ C4 ]\n",
      "[w] Residue 1..B.EPE.402. missing planar atom [ N3 ]\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▊| 1059/1074 [1:21:01<00:10,  1.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3e5c.cif1'\n",
      "    total number of nucleotides: 54\n",
      "    total number of amino acids: 1\n",
      "    total number of base pairs: 25\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 3\n",
      "    total number of stems: 4\n",
      "    total number of non-pairing interactions: 65\n",
      "    total number of atom-base capping interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 2\n",
      "    total number of hairpin loops: 2\n",
      "    total number of internal loops: 1\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 2\n",
      "    total number of phosphate interactions: 10\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▊| 1060/1074 [1:21:02<00:07,  1.76it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4m7a.cif1'\n",
      "    total number of nucleotides: 10\n",
      "    total number of amino acids: 1026\n",
      "[w] arginine [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..C.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..E.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..I.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..L.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..N.ARG.72.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..C.ARG.15.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..E.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..I.ARG.63.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..L.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..N.ARG.72.] -- NH1/NH2 naming swapped\n",
      "    total number of base pairs: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 2\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1061/1074 [1:21:02<00:06,  1.93it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5e3h.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 642\n",
      "    total number of base pairs: 11\n",
      "    total number of helices: 1\n",
      "    total number of stems: 1\n",
      "    total number of non-pairing interactions: 23\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 8\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1062/1074 [1:21:03<00:06,  1.91it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3sux.cif1'\n",
      "  1.X.CCC.101 0.130\n",
      "    total number of nucleotides: 101\n",
      "    total number of base pairs: 39\n",
      "    total number of multiplets: 3\n",
      "    total number of helices: 2\n",
      "    total number of stems: 5\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 120\n",
      "    total number of atom-base capping interactions: 5\n",
      "    total number of splayed-apart dinucleotides: 11\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of bulges: 2\n",
      "    total number of junctions: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of A-minor (types I and II) motifs: 1\n",
      "    total number of eXtended A-minor (type X) motifs: 2\n",
      "    total number of U-turns: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1063/1074 [1:21:03<00:04,  2.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3nna.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 175\n",
      "    total number of non-pairing interactions: 3\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1064/1074 [1:21:03<00:03,  2.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2jea.cif1'\n",
      "    total number of nucleotides: 5\n",
      "    total number of amino acids: 715\n",
      "[w] arginine [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.28.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..A.ARG.112.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.19.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.108.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..B.ARG.215.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1065/1074 [1:21:03<00:03,  2.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1h2c.cif1'\r\n",
      "    total number of nucleotides: 3\r\n",
      "    total number of amino acids: 125\r\n",
      "    total number of non-pairing interactions: 2\r\n",
      "    total number of splayed-apart dinucleotides: 1\r\n",
      "\r\n",
      "boundary for lvector(): [1 to 0]\r\n",
      "    total number of non-loop single-stranded segments: 1\r\n",
      "    total number of phosphate interactions: 1\r\n",
      "\r\n",
      "Time used: 00:00:00:00\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1066/1074 [1:21:04<00:02,  3.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3k5z.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 394\n",
      "    total number of non-pairing interactions: 2\n",
      "    total number of splayed-apart dinucleotides: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 2\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1067/1074 [1:21:04<00:02,  3.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/4ill.cif1'\n",
      "    total number of nucleotides: 36\n",
      "    total number of amino acids: 556\n",
      "    total number of base pairs: 6\n",
      "    total number of helices: 2\n",
      "    total number of stems: 2\n",
      "    total number of non-pairing interactions: 22\n",
      "    total number of atom-base capping interactions: 3\n",
      "    total number of splayed-apart dinucleotides: 18\n",
      "                        consolidated into units: 6\n",
      "    total number of hairpin loops: 2\n",
      "    total number of non-loop single-stranded segments: 4\n",
      "    total number of phosphate interactions: 19\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      " 99%|█████████▉| 1068/1074 [1:21:04<00:02,  2.63it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1qbp.cif1'\n",
      "    total number of nucleotides: 84\n",
      "    total number of base pairs: 42\n",
      "    total number of helices: 1\n",
      "    total number of stems: 6\n",
      "    total number of non-pairing interactions: 95\n",
      "    total number of internal loops: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "100%|█████████▉| 1069/1074 [1:21:05<00:01,  3.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3dd2.cif1'\n",
      "    total number of nucleotides: 25\n",
      "    total number of amino acids: 289\n",
      "    total number of base pairs: 11\n",
      "    total number of multiplets: 2\n",
      "    total number of helices: 1\n",
      "    total number of stems: 2\n",
      "    total number of isolated WC/wobble pairs: 1\n",
      "    total number of non-pairing interactions: 24\n",
      "    total number of atom-base capping interactions: 1\n",
      "    total number of splayed-apart dinucleotides: 6\n",
      "                        consolidated into units: 3\n",
      "    total number of hairpin loops: 1\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of phosphate interactions: 9\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "100%|█████████▉| 1070/1074 [1:21:05<00:01,  3.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/3r1e.cif1'\n",
      "    total number of nucleotides: 16\n",
      "    total number of base pairs: 8\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 16\n",
      "    total number of internal loops: 2\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "100%|█████████▉| 1071/1074 [1:21:05<00:00,  3.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/5jrc.cif1'\n",
      "    total number of nucleotides: 6\n",
      "    total number of amino acids: 743\n",
      "    total number of non-pairing interactions: 4\n",
      "    total number of splayed-apart dinucleotides: 2\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 3\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "100%|█████████▉| 1072/1074 [1:21:06<00:00,  3.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/2vnu.cif1'\n",
      "    total number of nucleotides: 9\n",
      "    total number of amino acids: 694\n",
      "[w] arginine [1..D.ARG.440.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.688.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.792.] -- NH1/NH2 naming swapped\n",
      "[w] arginine [1..D.ARG.806.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.440.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.621.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.688.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.792.] -- NH1/NH2 naming swapped\n",
      "[w] Residue [1..D.ARG.806.] -- NH1/NH2 naming swapped\n",
      "    total number of non-pairing interactions: 8\n",
      "    total number of splayed-apart dinucleotides: 3\n",
      "                        consolidated into units: 1\n",
      "\n",
      "boundary for lvector(): [1 to 0]\n",
      "    total number of non-loop single-stranded segments: 1\n",
      "    total number of phosphate interactions: 6\n",
      "\n",
      "Time used: 00:00:00:01\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "100%|█████████▉| 1073/1074 [1:21:06<00:00,  2.33it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Processing file '/home/mikhail/bioinformatics/data/NR_3.0/cif/1csl.cif1'\n",
      "    total number of nucleotides: 28\n",
      "    total number of base pairs: 12\n",
      "    total number of helices: 1\n",
      "    total number of stems: 3\n",
      "    total number of non-pairing interactions: 25\n",
      "    total number of splayed-apart dinucleotides: 4\n",
      "                        consolidated into units: 2\n",
      "    total number of bulges: 1\n",
      "    total number of internal loops: 1\n",
      "    total number of non-loop single-stranded segments: 3\n",
      "    total number of phosphate interactions: 1\n",
      "\n",
      "Time used: 00:00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1074/1074 [1:21:06<00:00,  2.81it/s]\n"
     ]
    }
   ],
   "source": [
    "for fname in tqdm(os.listdir(CIF_DIR)):\n",
    "    pdb_id, _ = os.path.splitext(fname)\n",
    "    \n",
    "    input_path = os.path.join(CIF_DIR, fname)\n",
    "    output_path = os.path.join(JSON_DIR, '{}.json'.format(pdb_id))\n",
    "    \n",
    "    ! x3dna-dssr --input=$input_path --format=mmcif --idstr=long --u-turn --more \\\n",
    "    --non-pair --po4 --output=$output_path --json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
